---
title: "Load Data"
output: html_document
editor_options: 
  chunk_output_type: console
creative_commons: CC BY
---

# Load Resources and Data

## Load resources
```{r load_resources,eval=T}
load("mgi_go_annotations.rdata")
load("symbols_M23.rdata")
```

## Load and label
```{r loadCounts,eval=T}

#load(url("https://s3.msi.umn.edu/zarkowe0/featureCounts_Gencode-M16_q4_Mon_Jan_22_2018_1051.rdata"))
#load("../data/featureCounts_Gencode-M16_q4_Mon_Jan_22_2018_1051.rdata")
#load("symbols_M16.rdata")
#load(url("https://s3.msi.umn.edu/gearhart/mgi91.rdata"))

#download.file("https://s3.msi.umn.edu/zarkowe0/featureCounts_Gencode-M19_q4_Fri_Nov_02_2018_1100.rdata",destfile = "featureCounts_Gencode-M19_q4_Fri_Nov_02_2018_1100.rdata")
#load("featureCounts_Gencode-M19_q4_Fri_Nov_02_2018_1100.rdata")
load("featureCounts_Gencode-M23_q4_Sun_Nov_03_2019_1857.rdata")
load("symbols_M23.rdata")

coldata<-as.data.frame(apply(rev_stranded_unfiltered$counts,2,sum))
colnames(coldata)<-"total_counts"
dim(coldata)

unstranded_coldata<-as.data.frame(apply(unstranded_counts_unfiltered$counts,2,sum))  
colnames(unstranded_coldata)<-"total_counts"
dim(unstranded_coldata)

#unstranded_coldata$filename<-rownames(unstranded_coldata) %>% 
#  gsub("X\\.home\\.bardwell\\.gearh006\\.dmrt1\\.ctv\\.mm10\\.GRCm38\\.","",.)  %>% 
#  gsub("\\.repair","",.)

coldata<-rbind(coldata,unstranded_coldata)
counts<-cbind(rev_stranded_counts$counts,unstranded_counts$counts)



coldata$filename<-rownames(coldata) %>% 
gsub("Project_035\\.GRCm38\\.","",.) %>% 
gsub("Project_037\\.GRCm38\\.","",.) %>% 
gsub("Project_038\\.GRCm38\\.","",.) %>% 
gsub("Bardwell_Project_041\\.GRCm38\\.","",.) %>% 
gsub("Project_006\\.GRCm38\\.","",.) %>% 
gsub("Bardwell_Project_033\\.GRCm38\\.","",.) %>% 
gsub("Bardwell_Project_035\\.GRCm38\\.","",.) %>% 
gsub("Genewiz_Project_30144243513\\.GRCm38\\.","",.) %>% 
gsub("\\.repair","",.)
  

#  gsub("X\\.home\\.bardwell\\.gearh006\\.Project_035\\.GRCm38.","",.) %>% 
#  gsub("\\.\\.\\.Project_035\\.GRCm38\\.","",.) %>% 
#  gsub("\\.\\.\\.Project_037\\.GRCm38\\.","",.) %>% 
#  gsub("\\.\\.\\.Project_038\\.GRCm38\\.","",.) %>% 
#  gsub("\\.\\.\\.Bardwell_Project_041\\.GRCm38\\.","",.) %>% 
#  gsub("X\\.home\\.bardwell\\.gearh006\\.Project_033p3\\.GRCm38\\.","",.)  %>% 
#  gsub("X\\.home\\.bardwell\\.gearh006\\.genewiz\\.GRCm38\\.","",.)  %>% 
 

#unstranded_coldata<-as.data.frame(apply(unstranded_counts$counts,2,sum))  
#colnames(unstranded_coldata)<-"total_counts"
#dim(unstranded_coldata)

#unstranded_coldata$filename<-rownames(unstranded_coldata) %>% 
#  gsub("X\\.home\\.bardwell\\.gearh006\\.dmrt1\\.ctv\\.mm10\\.GRCm38\\.","",.)  %>% 
#  gsub("\\.repair","",.)

#coldata<-rbind(coldata,unstranded_coldata)
#counts<-cbind(rev_stranded_counts$counts,unstranded_counts$counts)

dds<-DESeqDataSetFromMatrix(counts,colData = coldata,design = ~1)
rownames(colData(dds)) <- colData(dds)$filename %>%
  gsub("\\.Aligned\\.out\\.bam","",.) %>% 
  gsub("\\.Aligned\\.sortedByCoord\\.out\\.bam","",.) %>% 
  gsub("_scaled\\.bam","",.) %>% 
  gsub("\\.sort","",.) %>% 
  gsub("cag_","ctv_",.) %>% 
  gsub("Bardwell_","",.) %>% 
  gsub("\\.bam","",.)
rownames(dds)<-substr(rownames(dds),1,18)
mcols(dds)$basepairs<-rev_stranded_counts$annotation$Length
#mcols(dds)$mgi_symbol<-mgi[match(rownames(dds),mgi$ensembl_gene_id),]$mgi_symbol
mcols(dds)$symbol<-symbols[match(rownames(dds),symbols$gene_id),]$gene_name

#as.data.frame(colData(dds)

#colData(dds)$userid<-sapply(strsplit(sample_table[colData(dds)$filename],","),function(x) x[1])
#colData(dds)$trial <-sapply(strsplit(sample_table[colData(dds)$filename],","),function(x) x[2])
#colData(dds)$gene <-factor(sapply(strsplit(sample_table[colData(dds)$filename],","),function(x) x[3]))
#colData(dds)$time <-factor(sapply(strsplit(sample_table[colData(dds)$filename],","),function(x) x[4]))
#colData(dds)$batch <-factor(sapply(strsplit(sample_table[colData(dds)$filename],","),function(x) x[5]))

rownames(sample_table)<-rownames(sample_table) %>%   gsub("\\.Aligned\\.out\\.bam","",.) 

colData(dds)$userid<-sample_table[rownames(colData(dds)),"user"]
colData(dds)$trial<-sample_table[rownames(colData(dds)),"batch"]
colData(dds)$gene<-factor(sample_table[rownames(colData(dds)),"gene"])
colData(dds)$time<-sample_table[rownames(colData(dds)),"time"]
colData(dds)$batch<-sample_table[rownames(colData(dds)),"project"]
#View(as.data.frame(colData(dds)))
#as.data.frame(colData(dds))

dds<-estimateSizeFactors(dds)
plotPCA(normTransform(dds),intgroup=c("time")) + theme_bw() + ggtitle("PCA on All RNA-Seq Samples")


#Write Supplemental Raw Count Table
temp<-counts(dds[,colData(dds)$batch!="P006"]) %>% as.data.frame() %>%
  dplyr::rename("S8--_S9cf+_CTV_Sf1Cre_6257_Index14_S14" = "S8.._S9cf._CTV_Sf1Cre_6257_Index14_S14",
"S8--_S9gfcf_CTV_Sf1Cre_4520_Index6_S6" = "S8.._S9gfcf_CTV_Sf1Cre_4520_Index6_S6",
"S8--_S9gfcf_CTV_Sf1Cre_4669_Index9_S9" = "S8.._S9gfcf_CTV_Sf1Cre_4669_Index9_S9",
"S8--_S9gfcf_CTV_Sf1Cre_4675_Index16_S16" = "S8.._S9gfcf_CTV_Sf1Cre_4675_Index16_S16",
"S8+-_S9gfcf_CTV_Sf1Cre_6921_Index15_S15" = "S8.._S9gfcf_CTV_Sf1Cre_6921_Index15_S15",
"S8++_S9gfcf_CTV_Sf1Cre_6922_Index3_S3" = "S8.._S9gfcf_CTV_Sf1Cre_6922_Index3_S3",
"S8--_S9gfcf_noCTV_Sf1Cre_4525_Index7_S7" = "S8.._S9gfcf_noCTV_Sf1Cre_4525_Index7_S7",
"S8--_S9gf+_CTV_Sf1Cre_5463_Index2_S2" = "S8.._S9gf._CTV_Sf1Cre_5463_Index2_S2",
"S8+-_S9gf+_CTV_Sf1Cre_5617_Index11_S11" = "S8.._S9gf._CTV_Sf1Cre_5617_Index11_S11",
"S8--_S9gf+_CTV_Sf1Cre_5620_Index12_S12" = "S8.._S9gf._CTV_Sf1Cre_5620_Index12_S12",
"S8+-_S9gf+_CTV_Sf1Cre_6170_Index13_S13" = "S8.._S9gf._CTV_Sf1Cre_6170_Index13_S13",
"S8+-_S9gfgf_CTV_Sf1Cre_4519_Index5_S5"= "S8.._S9gfgf_CTV_Sf1Cre_4519_Index5_S5",
"S8--_S9gfgf_CTV_Sf1Cre_4668_Index8_S8" = "S8.._S9gfgf_CTV_Sf1Cre_4668_Index8_S8",
"S8--_S9gfgf_CTV_Sf1Cre_4684_Index10_S10"= "S8.._S9gfgf_CTV_Sf1Cre_4684_Index10_S10",
"S8--_S9gfgf_CTV_Sf1Cre_6927_Index4_S4" = "S8.._S9gfgf_CTV_Sf1Cre_6927_Index4_S4",
"S8--_S9gfgf_noCTV_Sf1Cre_4523_Index1_S1" = "S8.._S9gfgf_noCTV_Sf1Cre_4523_Index1_S1")
dim(temp)
as.data.frame(colnames(temp))

readxl::excel_sheets("../2021 Manuscript/Supplementary_Table_1_temp.xlsx")
sup_table1<-as.data.frame(readxl::read_excel("../2021 Manuscript/Supplementary_Table_1_temp.xlsx",sheet="RNA-Seq"))
colnames(temp) %in% sup_table1$`Temp name (delete this column)`
colnames(temp)<-sup_table1[match(colnames(temp),sup_table1$`Temp name (delete this column)`),"Sample Name"]
temp$gene_name<-symbols[match(rownames(temp),symbols$gene_id),"gene_name"]
dim(temp)
temp<-temp[sort(rownames(temp)),c("gene_name",sup_table1$`Sample Name`)]
temp<-rownames_to_column(temp,var="Ensembl GeneID")
#temp<-temp[with(temp,order("Ensembl GeneID")),]

write.csv(temp,"Lindeman_et_al_RNAseq_counts.csv",quote=F,row.names=F)

```


## Motifs
```{r import_motifs,eval=T}

#probably don't need these because I scan genome for Figure 2 #MDG
length(dmrt1_pwm<-rtracklayer::import("../data/dmrt1_invitro_pwm_score85.bed"))
seqlevelsStyle(dmrt1_pwm)<-"NCBI"
length(sox9_pwm<-rtracklayer::import("sox9_pwm_score90.bed"))
seqlevelsStyle(sox9_pwm)<-"NCBI"
#length(lrh_pwm<-rtracklayer::import("../data/lrh_pwm_score85.bed"))
#seqlevelsStyle(lrh_pwm)<-"NCBI"
length(klf4_pwm<-rtracklayer::import("../../R_RESOURCES/klf4_pwm_score85.bed"))
seqlevelsStyle(klf4_pwm)<-"NCBI"
length(foxl2_pwm<-rtracklayer::import("foxl2_pwm_score95.bed"))
seqlevelsStyle(foxl2_pwm)<-"NCBI"

#Add Invitro DMRT1 and JASPAR SOX9 motifs

#download.file("https://s3.msi.umn.edu/zarkowe0/murphy_2007_site_selection.fa",destfile="murphy_2007_site_selection.fa")
dmrt1_invitro_pfm<-consensusMatrix(subseq(readDNAStringSet("murphy_2007_site_selection.fa"),3,15))[1:4,]
dmrt1_invitro_motif<-new("pfm",mat=t(t(dmrt1_invitro_pfm[1:4,])*1/colSums(dmrt1_invitro_pfm[1:4,])), name="Dmrt1_In_Vitro")

#load jaspar motif
#download.file("http://jaspar.genereg.net/api/v1/matrix/MA0077.1.pfm",destfile = "MA0077.1.pfm")
sox9_jaspar_pfm<-as.matrix(read.table("MA0077.1.pfm",skip=1))
sox9_jaspar_pfm <- apply (sox9_jaspar_pfm, c (1, 2), function (x) {(as.integer(x))})
rownames(sox9_jaspar_pfm)<-c("A","C","G","T")
sox9_jaspar_motif<-new("pfm",mat=t(t(sox9_jaspar_pfm[1:4,])*1/colSums(sox9_jaspar_pfm[1:4,])), name="Sox9_Jaspar_motif")

#download.file("http://jaspar.genereg.net/api/v1/matrix/MA0033.2.pfm",destfile = "MA0033.2.pfm")
foxl2_pfm<-as.matrix(read.table("MA0033.2.pfm",skip=1))
foxl2_pfm <- apply (foxl2_pfm, c (1, 2), function (x) {(as.integer(x))})
rownames(foxl2_pfm)<-c("A","C","G","T")
colnames(foxl2_pfm)<-NULL
foxl2_jaspar_motif<-new("pfm",mat=t(t(foxl2_pfm[1:4,])*1/colSums(foxl2_pfm[1:4,])), name="Foxl2 MA0033.2")

#download.file("http://jaspar.genereg.net/api/v1/matrix/MA1121.1.pfm",destfile = "MA1121.1.pfm")
tead2_pfm<-as.matrix(read.table("MA1121.1.pfm",skip=1))
tead2_pfm <- apply (tead2_pfm, c (1, 2), function (x) {(as.integer(x))})
rownames(tead2_pfm)<-c("A","C","G","T")
colnames(tead2_pfm)<-NULL
tead2_jaspar_motif<-new("pfm",mat=t(t(tead2_pfm[1:4,])*1/colSums(tead2_pfm[1:4,])), name="Tead2 MA1121.1")

#download.file("http://jaspar.genereg.net/api/v1/matrix/MA0258.2.pfm",destfile = "MA0258.2.pfm")
esr2_pfm<-as.matrix(read.table("MA0258.2.pfm",skip=1))
esr2_pfm <- apply (esr2_pfm, c (1, 2), function (x) {(as.integer(x))})
rownames(esr2_pfm)<-c("A","C","G","T")
colnames(esr2_pfm)<-NULL
esr2_jaspar_motif<-new("pfm",mat=t(t(esr2_pfm[1:4,])*1/colSums(esr2_pfm[1:4,])), name="ESR2 MA0258.2")


#download.file("http://jaspar.genereg.net/api/v1/matrix/MA0505.1.pfm",destfile = "MA0505.1.pfm")
nr5a2_pfm<-as.matrix(read.table("MA0505.1.pfm",skip=1))
nr5a2_pfm <- apply (nr5a2_pfm, c (1, 2), function (x) {(as.integer(x))})
rownames(nr5a2_pfm)<-c("A","C","G","T")
colnames(nr5a2_pfm)<-NULL
nr5a2_jaspar_motif<-new("pfm",mat=t(t(nr5a2_pfm[1:4,])*1/colSums(nr5a2_pfm[1:4,])), name="NR5A2 MA0505.1")


```

## Load ChIP & ATAC Peaklists
```{r load_ChIP_ATAC_peaklist,eval=T}
#For Figure 1E
length(ovary8w_ctvdmrt1_dmrt1<-narrowPeakToGRanges(url("https://s3.msi.umn.edu/zarkowe0/ovary82_ctvdmrt1_Dmrt1_Project032_122716.fastq.macs_peaks.narrowPeak")))
length(ovary_ctvdmrt1_sox9<-narrowPeakToGRanges(url("https://s3.msi.umn.edu/zarkowe0/ovary8w_ctvdmrt1_Sox9_Project032_122716.fastq.dedup.unique.macs_peaks.narrowPeak")))
#export(reduce(ovary8w_ctvdmrt1_dmrt1),"ovary8w_ctvdmrt1_dmrt1.bed")
#export(reduce(ovary_ctvdmrt1_sox9),"ovary_ctvdmrt1_sox9.bed")
length(ovary8w_ctvdmrt1_dmrt1<-keepSeqlevels(ovary8w_ctvdmrt1_dmrt1,c(1:19,"X"),pruning.mode = "coarse"))
length(ovary_ctvdmrt1_sox9<-keepSeqlevels(ovary_ctvdmrt1_sox9,c(1:19,"X"),pruning.mode = "coarse"))


#Export for Sup Table
length(temp<- ovary8w_ctvdmrt1_dmrt1)
seqlevelsStyle(temp)<-"UCSC"
temp<-sortSeqlevels(temp)
temp<-sort(temp)
head(names(temp)<-paste0("cagDmrt1gonadDMRT1ChIP_",formatC(1:length(temp),width=5,format="d",flag="0")))
mcols(temp)<-mcols(temp)[,c("score","summit")]
export(temp,"ovary8w_ctvdmrt1_dmrt1.bed")
write.csv(as.data.frame(temp),"ovary8w_ctvdmrt1_dmrt1_peaklist.csv",row.names = T,quote=F)

#Export for Sup Table
length(temp<- ovary_ctvdmrt1_sox9)
seqlevelsStyle(temp)<-"UCSC"
temp<-sortSeqlevels(temp)
temp<-sort(temp)
head(names(temp)<-paste0("cagDmrt1gonadSOX9ChIP_",formatC(1:length(temp),width=5,format="d",flag="0")))
mcols(temp)<-mcols(temp)[,c("score","summit")]
export(temp,"ovary_ctvdmrt1_sox9.bed")
write.csv(as.data.frame(temp),"ovary_ctvdmrt1_sox9_peaklist.csv",row.names = T,quote=F)
 
 
rmsk<-rtracklayer::import("../../R_RESOURCES/mm10_rmsk_TE.gtf")
seqlevelsStyle(rmsk)<-"NCBI"
rmsk<-keepSeqlevels(rmsk,value=c(1:19,"X","Y"),pruning.mode = "coarse")
#Exclude Mitochondrial-genome like regions
mito_homologs<-narrowPeakToGRanges("../data/mitochondrial_homologs_peaks.narrowPeak")

# Open regions in granulosa Cells
length(peak_p23_atac_XX_rep1<-broadPeakToGRanges("../data/gran-1-112315_S1.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_p23_atac_XX_rep2<-broadPeakToGRanges("../data/gran-2-112315_S1.noMT.dedup.unique.macs_peaks.broadPeak"))

length(peak_p23_atac_XX <- reduce(c(peak_p23_atac_XX_rep1,peak_p23_atac_XX_rep2)))
length(peak_p23_atac_XX<-peak_p23_atac_XX[peak_p23_atac_XX %over% peak_p23_atac_XX_rep1 & 
                                               peak_p23_atac_XX %over% peak_p23_atac_XX_rep2 &
                                               !peak_p23_atac_XX %over% el_mm10])




#Open regions in Sertoli Cells
length(peak_p7_atac_XY_rep1<-broadPeakToGRanges("../data/sertoli-101415_S1.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_p7_atac_XY_rep2<-broadPeakToGRanges("../data/Serolti-112315_S1.noMT.dedup.unique.macs_peaks.broadPeak"))


length(peak_p7_atac_XY <- reduce(c(peak_p7_atac_XY_rep1,peak_p7_atac_XY_rep2)))
length(peak_p7_atac_XY<-peak_p7_atac_XY[peak_p7_atac_XY %over% peak_p7_atac_XY_rep1 & 
                                               peak_p7_atac_XY %over% peak_p7_atac_XY_rep2 &
                                               !peak_p7_atac_XY %over% el_mm10])


rtracklayer::export(peak_p23_atac_XX,paste0("peak_p23_atac_XX_",ts,".bed"))
#rtracklayer::export(peak_p23_atac_XX_rep1,paste0("peak_p23_atac_XX_rep1",ts,".bed"))
#rtracklayer::export(peak_p23_atac_XX_rep2,paste0("peak_p23_atac_XX_rep2",ts,".bed"))

rtracklayer::export(peak_p7_atac_XY,paste0("peak_p7_atac_XY_",ts,".bed"))
#rtracklayer::export(peak_p7_atac_XY_rep1,paste0("peak_p7_atac_XY_rep1",ts,".bed"))
#rtracklayer::export(peak_p7_atac_XY_rep2,paste0("peak_p7_atac_XY_rep2",ts,".bed"))



```


```{r load_ChIP_ATAC_peaklist,eval=T}
# This code doesn't find the intersection of replicates
#(peak_e105_atac_XX_files <- list.files("../data","broadPeak$",full.names=T) %>% 
#    str_subset("futtner",negate=F) %>% 
#    str_subset("BipotentialFemale",negate=F) )
#lengths(peak_e105_atac_XX<-GRangesList(lapply(peak_e105_atac_XX_files,broadPeakToGRanges)))
#length(peak_e105_atac_XX<-reduce(unlist(peak_e105_atac_XX)))
#length(peak_e105_atac_XX<-peak_e105_atac_XX[!peak_e105_atac_XX %over% el_mm10])
#
#(peak_e105_atac_XY_files <- list.files("../data","broadPeak$",full.names=T) %>% 
#    str_subset("futtner",negate=F) %>% 
#    str_subset("BipotentialMale",negate=F) )
#lengths(peak_e105_atac_XY<-GRangesList(lapply(peak_e105_atac_XY_files,broadPeakToGRanges)))
#length(peak_e105_atac_XY<-reduce(unlist(peak_e105_atac_XY)))
#length(peak_e105_atac_XY<-peak_e105_atac_XY[!peak_e105_atac_XY %over% el_mm10])

#(peak_e135_atac_XX_files <- list.files("../data","broadPeak$",full.names=T) %>% 
#    str_subset("futtner",negate=F) %>% 
#    str_subset("Granulosa",negate=F) )
#lengths(peak_e135_atac_XX<-GRangesList(lapply(peak_e135_atac_XX_files,broadPeakToGRanges)))
#length(peak_e135_atac_XX<-reduce(unlist(peak_e135_atac_XX)))
#length(peak_e135_atac_XX<-peak_e135_atac_XX[!peak_e135_atac_XX %over% el_mm10])
#
#(peak_e135_atac_XY_files <- list.files("../data","broadPeak$",full.names=T) %>% 
#    str_subset("futtner",negate=F) %>% 
#    str_subset("Sertoli",negate=F) )
#lengths(peak_e135_atac_XY<-GRangesList(lapply(peak_e135_atac_XY_files,broadPeakToGRanges)))
#length(peak_e135_atac_XY<-reduce(unlist(peak_e135_atac_XY)))
#length(peak_e135_atac_XY<-peak_e135_atac_XY[!peak_e135_atac_XY %over% el_mm10])


#Figure 0b - Futtner ATAC-seq
#length(peak_e105_atac_XX_dm19 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialFemaleDM19_E105_broad_nomodel_041919_peaks.broadPeak"))
#length(peak_e105_atac_XX_dm24 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialFemaleDM24_E105_broad_nomodel_041919_peaks.broadPeak"))
#length(peak_e105_atac_XY_dm21 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialMale_DM21_E105_broad_nomodel_041919_peaks.broadPeak"))
#length(peak_e105_atac_XY_dm22 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialMale_DM22_E105_broad_nomodel_041919_peaks.broadPeak"))

length(peak_e105_atac_XX_dm19 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialFemaleDM_19_E105_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_e105_atac_XX_dm24 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialFemaleDM_24_E105_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_e105_atac_XY_dm21 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialMale_DM21_E105_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_e105_atac_XY_dm22 <- broadPeakToGRanges("../data/futtner_ATAC_BipotentialMale_DM22_E105_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))


#length(peak_e135_atac_XX_dm12 <- broadPeakToGRanges("../data/futtner_ATAC_Granulosa_DM12_E135_broad_nomodel_041919_peaks.broadPeak"))
#length(peak_e135_atac_XX_dm13 <- broadPeakToGRanges("../data/futtner_ATAC_Granulosa_DM13_E135_broad_nomodel_041919_peaks.broadPeak"))
#length(peak_e135_atac_XY_dm15 <- broadPeakToGRanges("../data/futtner_ATAC_Sertoli_DM15_E135_broad_nomodel_041919_peaks.broadPeak"))
#length(peak_e135_atac_XY_dm16 <- broadPeakToGRanges("../data/futtner_ATAC_Sertoli_DM16_E135_broad_nomodel_041919_peaks.broadPeak"))

length(peak_e135_atac_XX_dm12 <- broadPeakToGRanges("../data/futtner_ATAC_Granulosa_DM12_E135_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_e135_atac_XX_dm13 <- broadPeakToGRanges("../data/futtner_ATAC_Granulosa_DM13_E135_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_e135_atac_XY_dm15 <- broadPeakToGRanges("../data/futtner_ATAC_Sertoli_DM15_E135_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_e135_atac_XY_dm16 <- broadPeakToGRanges("../data/futtner_ATAC_Sertoli_DM16_E135_R1.50bp_5prime.noMT.dedup.unique.macs_peaks.broadPeak"))




length(peak_e105_atac_XX <- reduce(c(peak_e105_atac_XX_dm19,peak_e105_atac_XX_dm24)))
length(peak_e105_atac_XX<-peak_e105_atac_XX[peak_e105_atac_XX %over% peak_e105_atac_XX_dm19 & 
                                               peak_e105_atac_XX %over% peak_e105_atac_XX_dm24 &
                                               !peak_e105_atac_XX %over% el_mm10])

length(peak_e105_atac_XY <- reduce(c(peak_e105_atac_XY_dm21,peak_e105_atac_XY_dm22)))
length(peak_e105_atac_XY <-peak_e105_atac_XY[peak_e105_atac_XY %over% peak_e105_atac_XY_dm21 & 
                                             peak_e105_atac_XY %over% peak_e105_atac_XY_dm22 &
                                             !peak_e105_atac_XY %over% el_mm10])

length(peak_e135_atac_XX <- reduce(c(peak_e135_atac_XX_dm12,peak_e135_atac_XX_dm13)))
length(peak_e135_atac_XX<-peak_e135_atac_XX[peak_e135_atac_XX %over% peak_e135_atac_XX_dm12 & 
                                               peak_e135_atac_XX %over% peak_e135_atac_XX_dm13 &
                                               !peak_e135_atac_XX %over% el_mm10])

length(peak_e135_atac_XY <- reduce(c(peak_e135_atac_XY_dm15,peak_e135_atac_XY_dm16)))
length(peak_e135_atac_XY<-peak_e135_atac_XY[peak_e135_atac_XY %over% peak_e135_atac_XY_dm15 & 
                                               peak_e135_atac_XY %over% peak_e135_atac_XY_dm16 &
                                               !peak_e135_atac_XY %over% el_mm10])


```


```{r load_ChIP_ATAC_peaklist,eval=T}
#ChIP Peak lists

#DMRT1
length(sertoli_dmrt1_rep1<-narrowPeakToGRanges("../data/covaris_6c_sertoli_dmrt1ChIP.R1.fastq.dedup.unique.macs_peaks.narrowPeak",name="rep1"))
export(sertoli_dmrt1_rep1,"sertoli_dmrt1_rep1.bed")
length(sertoli_dmrt1_rep2<-narrowPeakToGRanges("../data/DM1.dedup.unique.macs_peaks.narrowPeak",name="rep2"))
export(sertoli_dmrt1_rep2,"sertoli_dmrt1_rep2.bed")

length(sertoli_dmrt1_cat <- sort(c(sertoli_dmrt1_rep1,sertoli_dmrt1_rep2)))
length(sertoli_dmrt1<-reduce(sertoli_dmrt1_cat,with.revmap=T))
sertoli_dmrt1_revmap<-sertoli_dmrt1$revmap
x<-relist(mcols(sertoli_dmrt1_cat)[unlist(sertoli_dmrt1_revmap), ], sertoli_dmrt1_revmap)
sertoli_dmrt1$score<-unlist(lapply(x,function(a) round(mean(a$score),1)))
sertoli_dmrt1$summit<-unlist(lapply(x,function(a) round(mean(a$summit),0)))

length(sertoli_dmrt1_discard <-sertoli_dmrt1[sertoli_dmrt1 %over% el_mm10])

length(sertoli_dmrt1_non_reproducible <-sertoli_dmrt1[!(sertoli_dmrt1 %over% sertoli_dmrt1_rep1 &
                                       sertoli_dmrt1 %over% sertoli_dmrt1_rep2) &
                                      !sertoli_dmrt1 %over% el_mm10])

length(sertoli_dmrt1<-sertoli_dmrt1[sertoli_dmrt1 %over% sertoli_dmrt1_rep1 &
                                    sertoli_dmrt1 %over% sertoli_dmrt1_rep2 &
                                      !sertoli_dmrt1 %over% el_mm10])
seqlevelsStyle(sertoli_dmrt1)<-"UCSC"
sertoli_dmrt1<-sortSeqlevels(sertoli_dmrt1)
sertoli_dmrt1<-sort(sertoli_dmrt1)
head(names(sertoli_dmrt1)<-paste0("sertDMRT1_",formatC(1:length(sertoli_dmrt1),width=5,format="d",flag="0")))
sertoli_dmrt1$detected<-"1 and 2"
mcols(sertoli_dmrt1)<-mcols(sertoli_dmrt1)[,c("score","summit","detected")]
#export(sertoli_dmrt1,"sertoli_dmrt1_peaklist.bed")
write.csv(as.data.frame(sertoli_dmrt1),"sertoli_dmrt1_peaklist.csv",row.names = T,quote=F)
seqlevelsStyle(sertoli_dmrt1)<-"NCBI"

table(sertoli_dmrt1_non_reproducible$detected<-case_when(
  sertoli_dmrt1_non_reproducible %over% sertoli_dmrt1_rep1 & sertoli_dmrt1_non_reproducible %over% sertoli_dmrt1_rep2 ~ "repo",
  sertoli_dmrt1_non_reproducible %over% sertoli_dmrt1_rep1  ~ "1",
  sertoli_dmrt1_non_reproducible %over% sertoli_dmrt1_rep2 ~ "2",
  !sertoli_dmrt1_non_reproducible %over% sertoli_dmrt1_rep1 & !sertoli_dmrt1_non_reproducible %over% sertoli_dmrt1_rep2 ~ "huh?",
  ))

seqlevelsStyle(sertoli_dmrt1_non_reproducible)<-"UCSC"
sertoli_dmrt1_non_reproducible<-sortSeqlevels(sertoli_dmrt1_non_reproducible)
sertoli_dmrt1_non_reproducible<-sort(sertoli_dmrt1_non_reproducible)
head(names(sertoli_dmrt1_non_reproducible)<-paste0("sertdmrt1_",formatC((length(sertoli_dmrt1)+1):(length(sertoli_dmrt1)+length(sertoli_dmrt1_non_reproducible)),width=5,format="d",flag="0")))
mcols(sertoli_dmrt1_non_reproducible)<-mcols(sertoli_dmrt1_non_reproducible)[,c("score","summit","detected")]
write.csv(as.data.frame(sertoli_dmrt1_non_reproducible),"sertoli_dmrt1_peaklist_non_reproducible.csv",row.names = T,quote=F)


#SOX9
length(sertoli_sox9_rep1<-narrowPeakToGRanges("../data/Sox91.dedup.unique.macs_peaks.narrowPeak",name="rep1"))
export(sertoli_sox9_rep1,"sertoli_sox9_rep1.bed")
length(sertoli_sox9_rep2<-narrowPeakToGRanges("../data/Sox92.dedup.unique.macs_peaks.narrowPeak",name="rep2"))
export(sertoli_sox9_rep2,"sertoli_sox9_rep2.bed")

length(sertoli_sox9_cat <- sort(c(sertoli_sox9_rep1,sertoli_sox9_rep2)))
length(sertoli_sox9<-reduce(sertoli_sox9_cat,with.revmap=T))
sertoli_sox9_revmap<-sertoli_sox9$revmap
x<-relist(mcols(sertoli_sox9_cat)[unlist(sertoli_sox9_revmap), ], sertoli_sox9_revmap)
sertoli_sox9$score<-unlist(lapply(x,function(a) round(mean(a$score),1)))
sertoli_sox9$summit<-unlist(lapply(x,function(a) round(mean(a$summit),0)))

length(sertoli_sox9_discard <-sertoli_sox9[sertoli_sox9 %over% el_mm10])

length(sertoli_sox9_non_reproducible <-sertoli_sox9[!(sertoli_sox9 %over% sertoli_sox9_rep1 &
                                       sertoli_sox9 %over% sertoli_sox9_rep2) &
                                      !sertoli_sox9 %over% el_mm10])

length(sertoli_sox9<-sertoli_sox9[sertoli_sox9 %over% sertoli_sox9_rep1 &
                                    sertoli_sox9 %over% sertoli_sox9_rep2 &
                                      !sertoli_sox9 %over% el_mm10])
seqlevelsStyle(sertoli_sox9)<-"UCSC"
sertoli_sox9<-sortSeqlevels(sertoli_sox9)
sertoli_sox9<-sort(sertoli_sox9)
head(names(sertoli_sox9)<-paste0("sertSOX9_",formatC(1:length(sertoli_sox9),width=5,format="d",flag="0")))
sertoli_sox9$detected<-"1 and 2"
mcols(sertoli_sox9)<-mcols(sertoli_sox9)[,c("score","summit","detected")]

#export(sertoli_sox9,"sertoli_sox9_peaklist.bed")
write.csv(as.data.frame(sertoli_sox9),"sertoli_sox9_peaklist.csv",row.names = T,quote=F)
seqlevelsStyle(sertoli_sox9)<-"NCBI"

table(sertoli_sox9_non_reproducible$detected<-case_when(
  sertoli_sox9_non_reproducible %over% sertoli_sox9_rep1 & sertoli_sox9_non_reproducible %over% sertoli_sox9_rep2 ~ "repo",
  sertoli_sox9_non_reproducible %over% sertoli_sox9_rep1  ~ "1",
  sertoli_sox9_non_reproducible %over% sertoli_sox9_rep2 ~ "2",
  !sertoli_sox9_non_reproducible %over% sertoli_sox9_rep1 & !sertoli_sox9_non_reproducible %over% sertoli_sox9_rep2 ~ "huh?",
  ))

seqlevelsStyle(sertoli_sox9_non_reproducible)<-"UCSC"
sertoli_sox9_non_reproducible<-sortSeqlevels(sertoli_sox9_non_reproducible)
sertoli_sox9_non_reproducible<-sort(sertoli_sox9_non_reproducible)
head(names(sertoli_sox9_non_reproducible)<-paste0("sertSOX9_",formatC((length(sertoli_sox9)+1):(length(sertoli_sox9)+length(sertoli_sox9_non_reproducible)),width=5,format="d",flag="0")))
mcols(sertoli_sox9_non_reproducible)<-mcols(sertoli_sox9_non_reproducible)[,c("score","summit","detected")]
write.csv(as.data.frame(sertoli_sox9_non_reproducible),"sertoli_sox9_peaklist_non_reproducible.csv",row.names = T,quote=F)


#FOXL2
#length(ovary_foxl2<-narrowPeakToGRanges("../data/ovaryav574_CAGATC_042219_peaks.narrowPeak"))
length(ovary_foxl2<-narrowPeakToGRanges("../data/foxl2ChIP_wt_ovary_070519g_peaks.narrowPeak")) 
length(ovary_foxl2<-ovary_foxl2[!ovary_foxl2 %over% el_mm10])
length(ovary_foxl2<-keepSeqlevels(ovary_foxl2,c(1:19,"X"),pruning.mode="coarse"))

seqlevelsStyle(ovary_foxl2)<-"UCSC"
ovary_foxl2<-sortSeqlevels(ovary_foxl2)
ovary_foxl2<-sort(ovary_foxl2)
head(names(ovary_foxl2)<-paste0("ovaryFOXL2_",formatC(1:length(ovary_foxl2),width=5,format="d",flag="0")))
mcols(ovary_foxl2)<-mcols(ovary_foxl2)[,c("score","summit")]
export(ovary_foxl2,"ovary_foxl2_peaklist.bed")
write.csv(as.data.frame(ovary_foxl2),"ovary_foxl2_peaklist.csv",row.names = T,quote=F)
seqlevelsStyle(ovary_foxl2)<-"NCBI"



#ESR2
length(ovary_esr2<-narrowPeakToGRanges("../data/ESR22_GCCAAT_042319_peaks.narrowPeak"))
length(ovary_esr2<-ovary_esr2[!ovary_esr2 %over% el_mm10])
length(ovary_esr2<-keepSeqlevels(ovary_esr2,c(1:19,"X"),pruning.mode="coarse"))

seqlevelsStyle(ovary_esr2)<-"UCSC"
ovary_esr2<-sortSeqlevels(ovary_esr2)
ovary_esr2<-sort(ovary_esr2)
head(names(ovary_esr2)<-paste0("ovaryESR2_",formatC(1:length(ovary_esr2),width=5,format="d",flag="0")))
mcols(ovary_esr2)<-mcols(ovary_esr2)[,c("score","summit")]
export(ovary_esr2,"ovary_esr2_peaklist.bed")
write.csv(as.data.frame(ovary_esr2),"ovary_esr2_peaklist.csv",row.names = T,quote=F)

```

# Load in Called Dots
```{r}
nrow(sert_dots<-read.table("../HiC/sert.bedpe"))
colnames(sert_dots)<-c("chrom1","start1","end1","chrom2","start2","end2",
                        "cstart1","cstart2","c_label","c_size","count","la_exp.donut.value","la_exp.vertical.value",
                        "la_exp.horizontal.value","la_exp.lowleft.value","la_exp.donut.qval","la_exp.vertical.qval",
                        "la_exp.horizontal.qval","la_exp.lowleft.qval")
head(rownames(sert_dots)<-paste0("sertdots_",formatC(1:nrow(sert_dots),width=5,format="d",flag="0")))
#sert_dots_anchor1<-GRanges(seqnames=sert_dots$chrom1,IRanges(start=sert_dots$start1,end=sert_dots$end1))
#names(sert_dots_anchor1)<-rownames(sert_dots)
#head(sert_dots_anchor1)
#sert_dots_anchor2<-GRanges(seqnames=sert_dots$chrom2,IRanges(start=sert_dots$start2,end=sert_dots$end2))
#names(sert_dots_anchor2)<-rownames(sert_dots)
#head(sert_dots_anchor2)
write.csv(rownames_to_column(as.data.frame(sert_dots),var="Contact Name"),"sert_dots.csv",row.names = F,quote=F)

#View(sert_dots)

nrow(gran_dots<-read.table("../HiC/gran.bedpe"))
colnames(gran_dots)<-c("chrom1","start1","end1","chrom2","start2","end2",
                        "cstart1","cstart2","c_label","c_size","count","la_exp.donut.value","la_exp.vertical.value",
                        "la_exp.horizontal.value","la_exp.lowleft.value","la_exp.donut.qval","la_exp.vertical.qval",
                        "la_exp.horizontal.qval","la_exp.lowleft.qval")
head(rownames(gran_dots)<-paste0("grandots_",formatC(1:nrow(gran_dots),width=5,format="d",flag="0")))
#sert_dots_anchor1<-GRanges(seqnames=sert_dots$chrom1,IRanges(start=sert_dots$start1,end=sert_dots$end1))
#names(sert_dots_anchor1)<-rownames(sert_dots)
#head(sert_dots_anchor1)
#sert_dots_anchor2<-GRanges(seqnames=sert_dots$chrom2,IRanges(start=sert_dots$start2,end=sert_dots$end2))
#names(sert_dots_anchor2)<-rownames(sert_dots)
#head(sert_dots_anchor2)
write.csv(rownames_to_column(as.data.frame(gran_dots),var="Contact Name"),"gran_dots.csv",row.names = F,quote=F)

nrow(sert_dots)
nrow(gran_dots)
```

## Load In Vitro data
```{r load_invitro_data,eval=T}

length(peak_vitro_atac_cagSox9control_rep1<-broadPeakToGRanges("../data/gran_7d_cagSox9_noCRE_ATAC_rep1_N707_GW052319_R1_001.50bp_5prime_trimmed.fq.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagSox9control_rep2<-broadPeakToGRanges("../data/gran_7d_cagSox9_noCRE_ATAC_rep2_N708_GW052319_R1_001.50bp_5prime_trimmed.fq.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagSox9control_rep3<-broadPeakToGRanges("../data/gran_7d_cagSox9_noCRE_ATAC_rep3_N709_GW052319_R1_001.50bp_5prime_trimmed.fq.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagSox9control <- reduce(c(peak_vitro_atac_cagSox9control_rep1,peak_vitro_atac_cagSox9control_rep2,peak_vitro_atac_cagSox9control_rep3)))
length(peak_vitro_atac_cagSox9control<-peak_vitro_atac_cagSox9control[peak_vitro_atac_cagSox9control %over% peak_vitro_atac_cagSox9control_rep1 & 
                                               peak_vitro_atac_cagSox9control %over% peak_vitro_atac_cagSox9control_rep2 &
                                               peak_vitro_atac_cagSox9control %over% peak_vitro_atac_cagSox9control_rep3 &
                                               !peak_vitro_atac_cagSox9control %over% el_mm10])



rtracklayer::export(peak_vitro_atac_cagSox9control,"peak_vitro_atac_cagSox9control.bed")

length(peak_vitro_atac_cagSox9_rep1<-broadPeakToGRanges("../data/gran_7d_cagSox9_CRE_ATAC_rep1_N710_GW052319_R1_001.50bp_5prime_trimmed.fq.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagSox9_rep2<-broadPeakToGRanges("../data/gran_7d_cagSox9_CRE_ATAC_rep2_N711_GW052319_R1_001.50bp_5prime_trimmed.fq.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagSox9_rep3<-broadPeakToGRanges("../data/gran_7d_cagSox9_CRE_ATAC_rep3_N712_GW052319_R1_001.50bp_5prime_trimmed.fq.noMT.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagSox9 <- reduce(c(peak_vitro_atac_cagSox9_rep1,peak_vitro_atac_cagSox9_rep2,peak_vitro_atac_cagSox9_rep3)))
length(peak_vitro_atac_cagSox9<-peak_vitro_atac_cagSox9[peak_vitro_atac_cagSox9 %over% peak_vitro_atac_cagSox9_rep1 & 
                                               peak_vitro_atac_cagSox9 %over% peak_vitro_atac_cagSox9_rep2 &
                                               peak_vitro_atac_cagSox9 %over% peak_vitro_atac_cagSox9_rep3 &
                                               !peak_vitro_atac_cagSox9 %over% el_mm10])
rtracklayer::export(peak_vitro_atac_cagSox9,"peak_vitro_atac_cagSox9.bed")

#length(peak_vitro_atac_cagDmrt1control <-broadPeakToGRanges("../data/control_P037_ATACseq.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1control_rep1<-broadPeakToGRanges("../data/gran_p7control_n1_P037_ATACseq_N703_galore.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1control_rep2<-broadPeakToGRanges("../data/gran_noCre_d7_P037_ATACseq_N705_galore.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1control_rep3<-broadPeakToGRanges("../data/gran_noCAGDmrt1_d7_P037_ATACseq_N707_galore.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1control <- reduce(c(peak_vitro_atac_cagDmrt1control_rep1,peak_vitro_atac_cagDmrt1control_rep2,peak_vitro_atac_cagDmrt1control_rep3)))
length(peak_vitro_atac_cagDmrt1control<-peak_vitro_atac_cagDmrt1control[peak_vitro_atac_cagDmrt1control %over% peak_vitro_atac_cagDmrt1control_rep1 & 
                                               peak_vitro_atac_cagDmrt1control %over% peak_vitro_atac_cagDmrt1control_rep2 &
                                               peak_vitro_atac_cagDmrt1control %over% peak_vitro_atac_cagDmrt1control_rep3 &
                                               !peak_vitro_atac_cagDmrt1control %over% el_mm10])
rtracklayer::export(peak_vitro_atac_cagDmrt1control,"peak_vitro_atac_cagDmrt1control.bed")

#length(peak_vitro_atac_cagDmrt1<-broadPeakToGRanges("../data/cagDmrt1_P037_ATACseq.macs_peaks.broadPeak"))
#Use intersection of all three triplicates
length(peak_vitro_atac_cagDmrt1_rep1<-broadPeakToGRanges("../data/gran_p7CAGDmrt1_n2_P037_ATACseq_N702_galore.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1_rep2<-broadPeakToGRanges("../data/gran_plusCre_d7_P037_ATACseq_N706_galore.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1_rep3<-broadPeakToGRanges("../data/gran_plusCAGDmrt1_d7_P037_ATACseq_N708_galore.dedup.unique.macs_peaks.broadPeak"))
length(peak_vitro_atac_cagDmrt1 <- reduce(c(peak_vitro_atac_cagDmrt1_rep1,peak_vitro_atac_cagDmrt1_rep2,peak_vitro_atac_cagDmrt1_rep3)))
length(peak_vitro_atac_cagDmrt1<-peak_vitro_atac_cagDmrt1[peak_vitro_atac_cagDmrt1 %over% peak_vitro_atac_cagDmrt1_rep1 & 
                                               peak_vitro_atac_cagDmrt1 %over% peak_vitro_atac_cagDmrt1_rep2 &
                                               peak_vitro_atac_cagDmrt1 %over% peak_vitro_atac_cagDmrt1_rep3 &
                                               !peak_vitro_atac_cagDmrt1 %over% el_mm10])
rtracklayer::export(peak_vitro_atac_cagDmrt1,"peak_vitro_atac_cagDmrt1.bed")

#In vitro ChIP
 length(gran_ctvdmrt1_dmrt1<-narrowPeakToGRanges(url("https://s3.msi.umn.edu/zarkowe0/granulosa_ctvdmrt1_Dmrt1_Project032_122716.fastq.macs_peaks.narrowPeak")))
 length(gran_ctvdmrt1_sox9<-narrowPeakToGRanges(url("https://s3.msi.umn.edu/zarkowe0/granulosa_ctvdmrt1_Sox9_Project032_122716.fastq.dedup.unique.macs_peaks.narrowPeak")))
#length(gran_control_sox9<-narrowPeakToGRanges(url("https://s3.msi.umn.edu/zarkowe0/granulosa_control_Sox9_Project032_122716.fastq.macs_peaks.narrowPeak"))) # no sox9
  length(gran_ctvsox9_sox9<-narrowPeakToGRanges(url("https://s3.msi.umn.edu/zarkowe0/sox9_chip_from_sox9_expressing_granulsa_cells_idx9_Project033_041317.fastq.dedup.unique.macs_peaks.narrowPeak")))

length(gran_ctvdmrt1_dmrt1<-keepSeqlevels(gran_ctvdmrt1_dmrt1,c(1:19,"X"),pruning.mode = "coarse"))
length(gran_ctvdmrt1_sox9<-keepSeqlevels(gran_ctvdmrt1_sox9,c(1:19,"X"),pruning.mode = "coarse"))
length(gran_ctvsox9_sox9<-keepSeqlevels(gran_ctvsox9_sox9,c(1:19,"X"),pruning.mode = "coarse"))
  
# export(reduce(gran_ctvdmrt1_dmrt1),"gran_ctvdmrt1_dmrt1.bed")
#  export(reduce(gran_ctvdmrt1_sox9),"gran_ctvdmrt1_sox9.bed")
#  export(reduce(gran_ctvsox9_sox9),"gran_ctvsox9_sox9.bed")
  
  
length(temp<- gran_ctvdmrt1_dmrt1)
seqlevelsStyle(temp)<-"UCSC"
temp<-sortSeqlevels(temp)
temp<-sort(temp)
head(names(temp)<-paste0("cagDmrt1granulosaDMRT1ChIP_",formatC(1:length(temp),width=5,format="d",flag="0")))
mcols(temp)<-mcols(temp)[,c("score","summit")]
export(temp,"gran_ctvdmrt1_dmrt1_peaklist.bed")
write.csv(as.data.frame(temp),"gran_ctvdmrt1_dmrt1_peaklist.csv",row.names = T,quote=F)

length(temp<- gran_ctvdmrt1_sox9)
seqlevelsStyle(temp)<-"UCSC"
temp<-sortSeqlevels(temp)
temp<-sort(temp)
head(names(temp)<-paste0("cagDmrt1granulosaSOX9ChIP_",formatC(1:length(temp),width=5,format="d",flag="0")))
mcols(temp)<-mcols(temp)[,c("score","summit")]
export(temp,"gran_ctvdmrt1_sox9_peaklist.bed")
write.csv(as.data.frame(temp),"gran_ctvdmrt1_sox9_peaklist.csv",row.names = T,quote=F)

length(temp<- gran_ctvsox9_sox9)
seqlevelsStyle(temp)<-"UCSC"
temp<-sortSeqlevels(temp)
temp<-sort(temp)
head(names(temp)<-paste0("cagSox9granulosaSOX9ChIP_",formatC(1:length(temp),width=5,format="d",flag="0")))
mcols(temp)<-mcols(temp)[,c("score","summit")]
export(temp,"gran_ctvsox9_sox9_peaklist.bed")
write.csv(as.data.frame(temp),"gran_ctvsox9_sox9_peaklist.csv",row.names = T,quote=F)

```

