---
title: "Load Data"
output: html_document
editor_options: 
  chunk_output_type: console
---
# Figure 1


## Use Fresh Sertoli / granulosa data to define somatic genes that are differentially expressed

```{r somatic_gonad,eval=T}
#use differential expression between granulosa and Sertoli to define genes of interest
as.data.frame(colData(dds_soma<-dds[,str_detect(colData(dds)$time,"fresh")]))
#plotPCA(normTransform(dds_soma),intgroup=c("gene")) + theme_bw()
dds_soma$gene <- droplevels(dds_soma$gene)
design(dds_soma)<- ~ gene 
dds_soma<-DESeq(dds_soma)

#fpkm table
#get fpkm
dim(f_soma<-as.data.frame(log2(fpkm(dds_soma)+0.1)))

#colnames(f_gonad)<-paste(colData(dds_gonad)$time,colData(dds_gonad)$gene,colData(dds_gonad)$sox,1:nrow(colData(dds_gonad)),sep="_")
#f_gonad$mgi_symbol<-mcols(dds_gonad)$mgi_symbol
#View(f_gonad)

  f_soma_mean <-f_soma %>% 
    tibble::rownames_to_column(var="ensembl") %>%
    tidyr::gather(sample,fpkm,-ensembl) %>% 
    dplyr::mutate(sex=colData(dds_soma)[sample,]$gene) %>% 
    dplyr::mutate(sex=gsub("control","granulosa",sex)) %>% 
    dplyr::group_by(sex,ensembl) %>% 
      dplyr::summarize(mean_fpkm=round(mean(fpkm),3)) %>% 
      dplyr::select(sex,ensembl,mean_fpkm) %>% 
      ungroup() %>%
  tidyr::spread(sex,mean_fpkm) %>% as.data.frame()

  rownames(f_soma_mean) <- f_soma_mean$ensembl
  f_soma_mean<-f_soma_mean[,-1]
  
f_soma_mean$symbol<-mcols(dds_soma)[match(rownames(f_soma_mean),rownames(dds_soma)),]$symbol
length(high_soma_fpkm<-rownames(f_soma_mean[apply(f_soma_mean,1,max) > log2(2.5),]))
table(symbols[symbols$gene_id %in% high_soma_fpkm,"chr"])


#DEGs
resultsNames(dds_soma)
(summary(results_soma<-results(dds_soma)))
results_soma$mgi_symbol<-mcols(dds_soma)$symbol
#results_soma<-results_soma[!(is.na(results_soma$log2FoldChange) | is.na(results_soma$padj)),]
results_soma<-results_soma[with(results_soma,order(padj)),]

results_soma$f_sertoli<-f_soma_mean[match(rownames(results_soma),rownames(f_soma_mean)),]$sertoli
results_soma$f_granulosa<-f_soma_mean[match(rownames(results_soma),rownames(f_soma_mean)),]$granulosa

#View(as.data.frame(results_soma))


write.csv(results_soma,file="granulosa_vs_Sertoli_comparison.csv",quote=F)



#define a list of Sertoli Up genes for later
length(sertoli_up<-rownames(subset(results_soma,log2FoldChange > 2 & padj < 0.05 & f_sertoli > 2.5 )))
table(symbols[symbols$gene_id %in% sertoli_up,"chr"])
length(granulosa_up<-rownames(subset(results_soma,log2FoldChange < -2 & padj < 0.05 &  f_granulosa > 2.5)))
table(symbols[symbols$gene_id %in% granulosa_up,"chr"])

#define transcripts fore each set
sertoli_transcripts<-transcripts[transcripts$gene_id %in% sertoli_up]
granulosa_transcripts<-transcripts[transcripts$gene_id %in% granulosa_up]

#View(as.data.frame(results_soma[sertoli_up,]))


nrow(temp<-as.data.frame(results_soma))

#subset to significant genes (###use fpkm cutoff instead of baseMean?)
nrow(results_soma<-subset(results_soma,abs(log2FoldChange) > 2 & padj < 0.05 & baseMean > 100 & (f_sertoli > 2.5 | f_granulosa > 2.5)))
table(symbols[symbols$gene_id %in% rownames(results_soma),"chr"])

table(temp$results_soma<- factor(case_when (
  temp$log2FoldChange > 2 & temp$padj < 0.05 & temp$baseMean > 100 ~ "Sertoli",
  temp$log2FoldChange < -2 & temp$padj < 0.05 & temp$baseMean > 100 ~ "Granulosa",
  TRUE ~ "unchanged"
  ),levels=c("Granulosa","unchanged","Sertoli")))
  

(df<-temp[temp$mgi_symbol %in% c("Dmrt1","Nr5a2","Esr2","Foxl2","Sox8","Sox9","Cyp19a1","Cytip","Hsd17b3","Wnt5b", "Sdc3", "Dhh", "Cst9", "Rspo2","Bmpr1b"),])

temp %>% 
  dplyr::filter(!is.na(log2FoldChange) & !is.na(padj)) %>% 
ggplot(aes(x=log2FoldChange,y=(-1*log10(padj)),color=results_soma)) + geom_point() + theme_bw() + scale_color_manual(values=c("red","darkgray","blue")) +
  annotate("text",x=df$log2FoldChange,y=(-1*log10(df$padj)),label=df$mgi_symbol)

```

## Create a DDS object of the whole gonad samples
```{r }
as.data.frame(colData(dds_gonad<-dds[,str_detect(colData(dds)$time,"ovary|testis") & !str_detect(colData(dds)$filename,"DMEf|WTf")]))
colData(dds_gonad)$sox8<-as.numeric(substr(colData(dds_gonad)$gene,2,2))
colData(dds_gonad)$sox9<-as.numeric(substr(colData(dds_gonad)$gene,3,3))
colData(dds_gonad)$sox<-colData(dds_gonad)$sox8+colData(dds_gonad)$sox9
colData(dds_gonad)$gene<-as.factor(sapply(str_split(colData(dds_gonad)$gene,"_"), function(x) x[2]))
colData(dds_gonad)$batch<-as.factor(colData(dds_gonad)$batch)
as.data.frame(colData(dds_gonad))
plotPCA(normTransform(dds_gonad),intgroup=c("userid","time","gene","sox")) + 
  theme_bw() + ggtitle("Gonad Data - Full Gene Set ")

#subset to ~soma
#dim(dds_gonad)
#dim(dds_gonad_subset<-dds_gonad[rownames(dds_gonad) %in% rownames(results_soma),])
#as.data.frame(colData(dds_gonad_subset))
#plotPCA(normTransform(dds_gonad_subset),intgroup=c("time","gene","sox")) + 
#  theme_bw() + ggtitle("Gonad Data - Somatic Subset")
#pheatmap(f_gonad_subset)
```



## Figure 1a - UpsetR
```{r upset,eval=T}


length(vivo_atac<-reduce(c(peak_e105_atac_XX,peak_e105_atac_XY,peak_e135_atac_XX,peak_e135_atac_XY,peak_p23_atac_XX,peak_p7_atac_XY)))

length(vivo_atac<-keepSeqlevels(vivo_atac,c(1:19,"X"),pruning.mode="coarse"))  #drop Y because not all samples have a Y
length(vivo_atac<-suppressWarnings(vivo_atac[!vivo_atac %over% mito_homologs]))

head(names(vivo_atac)<-paste0("vivo_atac",formatC(1:length(vivo_atac),width=6,format="d",flag="0")))


summary(width(vivo_atac))


(fls.k27ac_p3_Sert <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>%
   grep("k27",.,value=T) %>%
   grep("wt",.,value=T) %>%
   grep("GW02",.,value=T) %>%
   grep("P3",.,value=T ))

#system.time(vivo_atac_k27ac_counts<- summarizeOverlaps(vivo_atac,BamFileList(fls.k27ac_p3_Sert,yieldSize = 1e5),mode =Union, ignore.strand=T))
#as.data.frame(apply(assays(vivo_atac_k27ac_counts)$counts,2,sum))
#save(vivo_atac_k27ac_counts,file="vivo_atac_k27ac_counts.rdata")
load("vivo_atac_k27ac_counts.rdata")


temp<-as.data.frame(assays(vivo_atac_k27ac_counts)$counts)
colnames(temp)<-c("wt_idx13","wt_idx14")
head(temp)
colSums(temp)
temp<-sweep(temp,2,colSums(temp)/1e6,"/")
temp$width<-width(vivo_atac)
temp2<-data.frame(p3_XY_k27ac=(temp$wt_idx13 + temp$wt_idx14)/(0.002*temp$width))
head(temp2)
mcols(vivo_atac)<-log2(temp2+0.5)


vivo_atac$e105_XX<-as.numeric(vivo_atac %over% peak_e105_atac_XX)
vivo_atac$e105_XY<-as.numeric(vivo_atac %over% peak_e105_atac_XY)
vivo_atac$e135_XX<-as.numeric(vivo_atac %over% peak_e135_atac_XX)
vivo_atac$e135_XY<-as.numeric(vivo_atac %over% peak_e135_atac_XY)
vivo_atac$p23_XX<-as.numeric(vivo_atac %over% peak_p23_atac_XX)
vivo_atac$p7_XY<-as.numeric(vivo_atac %over% peak_p7_atac_XY)

colSums(as.data.frame(mcols(vivo_atac)))

upset(as.data.frame(mcols(vivo_atac)), nsets = 4, number.angles = 30, point.size = 3.5, line.size = 2,
    mainbar.y.label = "Peak Intersections", main.bar.color = "black",
    sets=c("p7_XY","p23_XX","e135_XY","e135_XX","e105_XY","e105_XX"), sets.x.label = "Numebr of Peaks", keep.order = TRUE,
    queries=list(list(query = intersects, params = list("p7_XY", "e135_XY"),color = "blue", active = T)),text.scale = c(1.3, 1.3, 1, 1, 2, 0.75),
     boxplot.summary = c("p3_XY_k27ac"))

```




## Figure 0b  - Combine Embryonic Data with WT vs Sertoli DARs
### Map Genes to DARs
### Define g_regions and s_regions for DARs close to granulosa expressed genes and Sertoli expressed genes

```{r}
#Find Differentially bound ATAC-able regions

#(fls1.vivo <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("/N70"))
(fls1.vivo <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("GW052319") %>% str_subset("fresh"))
(fls2.vivo <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("futtner"))
#(fls3.vitro <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("ATAC") %>% str_subset("gran") %>% str_subset("P043"))
(fls3.vitro <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("GW052319") %>% str_subset("cagSox9"))
(fls4.vitro <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% 
   grep("ATAC",.,value=T) %>% 
   grep("marked",.,value=T,invert=T) %>% 
   grep("P037",.,value=T) %>% 
   grep("gran",.,value=T ))

  #system.time(atac_counts<- summarizeOverlaps(vivo_atac,BamFileList(c(fls1.vivo,fls2.vivo),yieldSize = 1e5),mode =Union, ignore.strand=T))
#system.time(atac_counts<- summarizeOverlaps(vivo_atac,BamFileList(c(fls1.vivo,fls2.vivo,fls3.vitro,fls4.vitro),yieldSize = 1e5),mode =Union, ignore.strand=T))
#apply(assays(atac_counts)$counts,2,sum)
#save(atac_counts,file="atac_counts.rdata")
load("atac_counts.rdata")

#mcols(vivo_atac)<-assays(atac_counts)$counts
#colnames(mcols(vivo_atac))<-colnames(assays(atac_counts)$counts) %>% gsub("\\.R1_trimmed\\.fastq\\.marked\\.bam","",.)
atac_dds<-DESeqDataSet(atac_counts,design = ~1)
mcols(atac_dds)$basepairs<-width(vivo_atac)
colData(atac_dds)$sample<-colnames(assays(atac_counts)$counts) %>% gsub("\\.R1_trimmed\\.fastq\\.marked\\.bam","",.) %>% 
  gsub("futtner_ATAC_","",.) %>% gsub("_R1_trimmed\\.bam","",.) %>% 
  gsub("\\.marked\\.bam","",.) %>%  gsub("\\.bam","",.) 

colData(atac_dds)$sex<-factor(c(rep("Female",3),rep("Male",3),rep("Female",2),rep("Male",2),rep("Female",2),rep("Male",2),rep("Female",12)))
colData(atac_dds)$age<-factor(c(rep("postnatal",6),rep("E105",4),rep("E135",4),rep("postnatal",12)),levels=c("E105","E135","postnatal"))
colData(atac_dds)$cag<-factor(c(rep("na",14),rep("Sox9",3),rep("na",5),"Dmrt1","na",rep("Dmrt1",2)),levels=c("na","Sox9","Dmrt1"))
colData(atac_dds)$source <-factor(c(rep("vivo",14),rep("vitro",12)),levels=c("vivo","vitro"))

#colData(atac_dds)$sex<-factor(c(rep("Male",2),rep("Female",4),rep("Male",2),rep("Female",2),rep("Male",2),rep("Female",12)))
#colData(atac_dds)$age<-factor(c(rep("postnatal",4),rep("E105",4),rep("E135",4),rep("postnatal",12)),levels=c("E105","E135","postnatal"))
#colData(atac_dds)$cag<-factor(c(rep("na",12),rep("Sox9",3),rep("na",5),"Dmrt1","na",rep("Dmrt1",2)),levels=c("na","Sox9","Dmrt1"))

colData(atac_dds)$group<-factor(paste0(colData(atac_dds)$age,"_",colData(atac_dds)$sex,"_",colData(atac_dds)$source,"_",colData(atac_dds)$cag),
                        levels=c("E105_Female_vivo_na","E105_Male_vivo_na","E135_Female_vivo_na","E135_Male_vivo_na","postnatal_Female_vivo_na","postnatal_Male_vivo_na","postnatal_Female_vitro_na","postnatal_Female_vitro_Sox9","postnatal_Female_vitro_Dmrt1"))

as.data.frame(colData(atac_dds))

#Move to Figure 5
#(df<-plotPCA(normTransform(atac_dds),intgroup=c("group"),returnData=T,ntop=500))
#
#plotPCA(normTransform(atac_dds),intgroup=c("group"),ntop=500) + theme_bw() +
#  scale_color_manual(values = c(cbPalette,"red")) + geom_text(x=df$PC1,y=df$PC2,label=df$group) 


```

## Make an  in vivo version of this PCA plot for Figure 1 (full version moved to figure 5)
```{r}
as.data.frame(colData(vivo_dds<-atac_dds[,colData(atac_dds)$source=="vivo"]))


as.data.frame(colData(vivo_dds))
#table(as.character(seqnames(vivo_atac)) %in% as.character(c(1:19,"X")))
#dim(vivo_dds<-vivo_dds[as.character(seqnames(vivo_atac)) %in% as.character(c(1:19,"X")),])
(df<-plotPCA(normTransform(vivo_dds),intgroup=c("sex","age"),returnData=T,ntop=500))

#df$sample<-df$sample %>% gsub("N702","Sertoli_N702_p7",.) %>%
#   gsub("N703","Sertoli_N703_p7",.) %>% 
#   gsub("N704","Granulosa_N704_p23",.) %>% 
#   gsub("N705","Granulosa_N705_p23",.)


plotPCA(normTransform(vivo_dds),intgroup=c("sex","age"),ntop=500) + theme_bw() +
scale_color_manual(values = cbPalette[c(1:6)]) + geom_text(x=df$PC1,y=df$PC2,label=paste0(df$sex,"_",df$age)) 



```

#Use contrast to find differentially expressed regions between granulosa and Sertoli
```{r}
f_atac_vivo <- fpkm(vivo_dds,robust=TRUE)

f_mean_atac_vivo <- f_atac_vivo %>% as.data.frame() %>% 
  tibble::rownames_to_column(var="region") %>% 
  tidyr::gather(sample,fpkm,-region) %>%
  dplyr::mutate(group=colData(vivo_dds)[sample,]$group) %>% 
  dplyr::group_by(region,group) %>% 
    dplyr::summarize(mean_fpkm=round(log2(mean(fpkm)+0.01),3)) %>% 
    ungroup() %>% 
  dplyr::select(region,group,mean_fpkm) %>% 
  tidyr::spread(group,mean_fpkm) %>%  as.data.frame() 

rownames(f_mean_atac_vivo)<-f_mean_atac_vivo$region
f_mean_atac_vivo<-f_mean_atac_vivo[,-1]
colnames(f_mean_atac_vivo)<- colnames(f_mean_atac_vivo) %>% 
  gsub("_vivo_na","",.) %>% 
  gsub("postnatal_Female","p23_Female",.) %>% 
  gsub("postnatal_Male","p7_Male",.)
  
head(f_mean_atac_vivo)


design(vivo_dds)<-~group
vivo_dds$group<-droplevels(vivo_dds$group)
vivo_dds<-DESeq(vivo_dds)

summary(res_vivo_postnatal<-results(vivo_dds,contrast=c("group","postnatal_Male_vivo_na","postnatal_Female_vivo_na"),alpha=0.05))

vivo_atac
#Add metadata to vivo_atac gr
stopifnot(all.equal(rownames(f_mean_atac_vivo),names(vivo_atac)))
mcols(vivo_atac)<-f_mean_atac_vivo

#design(vivo_dds)<-~sex
#vivo_dds<-DESeq(vivo_dds)
#vivo_res_raw<-results(vivo_dds,alpha=0.05)
#vivo_res<-as.data.frame(vivo_res_raw)
#vivo_res$ihw<-ihw(pvalue ~ baseMean,  data = vivo_res, alpha = 0.05)@df$adj_pvalue

stopifnot(all.equal(rownames(res_vivo_postnatal),names(vivo_atac)))
vivo_atac$lfc<-round(res_vivo_postnatal$log2FoldChange,3)
vivo_atac$padj<-res_vivo_postnatal$padj

#are they called in the Sert or Granulosa Datasets
table(vivo_atac$sert_call <- vivo_atac %over% peak_p7_atac_XY)
table(vivo_atac$gran_call <- vivo_atac %over% peak_p23_atac_XX)

#vivo_atac$padj<-formatC(res_vivo_postnatal$padj, format = "e", digits = 3)
#export(peak_p7_atac_XY,"peak_p7_atac_XY.bed")
#export(peak_p23_atac_XX,"peak_p23_atac_XX.bed")

## CHANGED THIS TO USE LOG2 OF THE MEAN INSTEAD OF UNTRANSFORMED INDIVIDUAL SAMPLES
#classify
#table(vivo_atac$class<-case_when (
#  vivo_atac$padj < 0.05 & vivo_atac$postnatal_Male_vivo_na > 4 & vivo_atac$postnatal_Female_vivo_na < 1 ~ "Sertoli", #3.3,1 vs 4.3,2.8
#  vivo_atac$padj < 0.05 & vivo_atac$postnatal_Male_vivo_na < 1 &  vivo_atac$postnatal_Female_vivo_na > 4 ~ "Granulosa", # vs 2.8,4.3
#  abs(vivo_atac$lfc) < log2(1.25) & vivo_atac$postnatal_Male_vivo_na > 4  & vivo_atac$postnatal_Female_vivo_na > 4  ~ "Constitutive",
#  TRUE ~ "unclassified"
#))

table(vivo_atac$class<-case_when (
  vivo_atac$sert_call & vivo_atac$padj < 0.05 & vivo_atac$lfc > 1 & !vivo_atac$gran_call ~ "Sertoli", #3.3,1 vs 4.3,2.8
  vivo_atac$gran_call & vivo_atac$padj < 0.05 & vivo_atac$lfc < -1 &  !vivo_atac$sert_call ~ "Granulosa", # vs 2.8,4.3
  abs(vivo_atac$lfc) < log2(1.25) & vivo_atac$sert_call & vivo_atac$gran_call ~ "Constitutive",
  TRUE ~ "unclassified"
))

head(as.data.frame(vivo_atac))
#View(as.data.frame(vivo_atac))
vivo_atac_subset<-vivo_atac[vivo_atac$class!="unclassified"]
#switch to using the single nearest gene
vivo_atac_subset<-annotate_peaks(vivo_atac_subset)  
table(vivo_atac_subset$dimorph)
#View(as.data.frame(vivo_atac_subset[vivo_atac_subset$dimorph=="Gran;Sert"]))
#vivo_atac_subset["vivo_atac029624"]$dimorph<-"Sert"
#vivo_atac_subset["vivo_atac054417"]$dimorph<-"Gran"
#table(vivo_atac_subset$dimorph)

length(vivo_atac_subset[vivo_atac_subset$class=="Granulosa" & vivo_atac_subset$dimorph=="Gran"])
length(vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$dimorph=="Gran"])
length(vivo_atac_subset[vivo_atac_subset$class=="Granulosa" & vivo_atac_subset$dimorph=="Sert"])
length(vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$dimorph=="Sert"])

#View(as.data.frame(vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$dimorph=="Gran"]))
#View(as.data.frame(vivo_atac_subset[vivo_atac_subset$class=="Granulosa" & vivo_atac_subset$dimorph=="Gran"]))


nr5a2_peaks<-vivo_atac[vivo_atac %over% swl("chr1:136803128-137029332")]
nr5a2_peaks[nr5a2_peaks %over% sertoli_dmrt1]
rspo2_peaks<-vivo_atac[vivo_atac %over% swl("chr15:42,953,041-43,193,844")]
rspo2_peaks[rspo2_peaks %over% sertoli_dmrt1]
(esr2_peaks<-vivo_atac[vivo_atac %over% swl("chr12:76,130,887-76,158,648")])

#phastcons
#vivo_atac_subset<-add_mean_phastcons_score(vivo_atac_subset)

#export(vivo_atac_subset,"vivo_atac_subset_gw070919.bed")

#Which ones overlap what sites?
table(vivo_atac_subset$dmrt1<-vivo_atac_subset %over% sertoli_dmrt1)
table(vivo_atac_subset$sox9<-vivo_atac_subset %over% testis_sox9)
table(vivo_atac_subset$foxl2<-vivo_atac_subset %over% ovary_foxl2)

table(vivo_atac_subset$peak<-case_when (
  vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 ~ "DSF",
  vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 ~ "DS",
  vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 ~ "D",
  !vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 ~ "SF",
  vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 ~ "DF",
  !vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 ~ "S",
  !vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 ~ "F",
  !vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 ~ "N",
  TRUE ~ "unclassified"
))


roi<-swl(c("chr6:119,522,263-119,531,826", # Wnt5b
           "chr12:76,130,814-76,134,590", #Esr2
          "chr1:136,912,616-136,917,861", #Nr5a2
              "chr4:130,800,511-130,803,585",#Sdc3
          "chr15:98,890,974-98,893,055", #Dhh
          "chr2:148,832,886-148,834,489", #Cst9
          "chr13:64,063,297-64,065,858", #HSD17b3
            "chr15:43,134,055-43,138,423" #Rspo2
                  ))
#roi<-c("vivo_atac029963","vivo_atac075082","vivo_atac078850","vivo_atac036384","vivo_atac007173")
#roi<-c("vivo_atac024838","vivo_atac061641","vivo_atac064754","vivo_atac005988","vivo_atac030065","vivo_atac081920")
#poi<-roi[roi %in% names(vivo_atac_subset)]
(df<-vivo_atac_subset[vivo_atac_subset %over% roi])
#df$label<-c("Esr2","Cytip","B2m","Nr5a2","Hsd17b3","Sdc3")


#13      64074603        64075302

length(vivo_atac_subset)
#  chr12:75,884,786-76,384,786 chr1:136,590,477-137,092,477 chr2:148,505,257-149,005,257

#t(log2(f_atac_vivo[roi,]))
  
mcols(vivo_atac_subset) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=peak)) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)])) +
  annotate("text",x=df$lfc,y=(-1*log10(df$padj)),label=df$genes)

#optionally color by Gran vs Sert gene expression
mcols(vivo_atac_subset) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=dimorph)) + geom_point() + theme_bw()  + scale_color_manual(values=c("red","purple","gray","blue")) +
  annotate("text",x=df$lfc,y=(-1*log10(df$padj)),label=df$genes)


```
## Use meme to identify motifs under Dmrt1, Foxl2 and Sox9 ChIP peaks
```{r}
length(ovary_foxl2)
length(testis_sox9)
length(sertoli_dmrt1_top<-topify(sertoli_dmrt1))

ovary_foxl2.center<-GRanges(seqnames=seqnames(ovary_foxl2),IRanges(start=start(ovary_foxl2)+ovary_foxl2$summit,width=1),strand="*")+75
ovary_foxl2.dna<-getSeq(mm10,sample(ovary_foxl2.center,length(testis_sox9)))
names(ovary_foxl2.dna)<-paste0("ovary_foxl2_",1:length(ovary_foxl2.dna))
export(ovary_foxl2.dna,"ovary_foxl2_sampled_151bp.fasta")

testis_sox9.center<-GRanges(seqnames=seqnames(testis_sox9),IRanges(start=start(testis_sox9)+testis_sox9$summit,width=1),strand="*")+75
testis_sox9.dna<-getSeq(mm10,testis_sox9.center)
names(testis_sox9.dna)<-paste0("testis_sox9_",1:length(testis_sox9.dna))
export(testis_sox9.dna,"testis_sox9_151bp.fasta")

sertoli_dmrt1_top.center<-GRanges(seqnames=seqnames(sertoli_dmrt1_top),IRanges(start=start(sertoli_dmrt1_top)+sertoli_dmrt1_top$summit,width=1),strand="*")+75
sertoli_dmrt1_top.dna<-getSeq(mm10,sample(sertoli_dmrt1_top.center,length(testis_sox9)))
names(sertoli_dmrt1_top.dna)<-paste0("sertoli_dmrt1_top_",1:length(sertoli_dmrt1_top.dna))
export(sertoli_dmrt1_top.dna,"sertoli_dmrt1_top_sampled_151bp.fasta")

```

#Import Meme on ChIPs
```{r}
testis_sox9.meme<-TFBSTools:::parseMEMEOutput412("testis_sox9_151bp.txt")
x<-lengths(testis_sox9.meme_dna<-relist(getSeq(readDNAStringSet("testis_sox9_151bp.fasta"),unlist(testis_sox9.meme$motifList)),testis_sox9.meme$motifList))
names(testis_sox9.meme_dna)<-paste0("Motif ",1:length(testis_sox9.meme$motifList),", p-value: ",testis_sox9.meme$motifEvalues," ( ",x," sites )")
#ggplot() + geom_logo(lapply(testis_sox9.meme_dna,as.character),seq_type='dna') + theme_logo() +
#  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 ChIP Peaks")

testis_sox9.pfm<-consensusMatrix(reverseComplement(testis_sox9.meme_dna[[1]]))
testis_sox9.motif<-new("pfm",mat=t(t(testis_sox9.pfm[1:4,])*1/colSums(testis_sox9.pfm[1:4,])), name="Testis Sox9 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(sox9_jaspar_motif,testis_sox9.motif),revcomp=c(F,F)))

system.time(testis_sox9.sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=testis_sox9.pfm[1:4,],score="85%")))
sapply(testis_sox9.sites,length)
testis_sox9.sites<-sort(unlist(GRangesList(testis_sox9.sites)))

#foxl2
ovary_foxl2.meme<-TFBSTools:::parseMEMEOutput412("ovary_foxl2_sampled_151bp.txt")
x<-lengths(ovary_foxl2.meme_dna<-relist(getSeq(readDNAStringSet("ovary_foxl2_sampled_151bp.fasta"),unlist(ovary_foxl2.meme$motifList)),ovary_foxl2.meme$motifList))
names(ovary_foxl2.meme_dna)<-paste0("Motif ",1:length(ovary_foxl2.meme$motifList),", p-value: ",ovary_foxl2.meme$motifEvalues," ( ",x," sites )")
#ggplot() + geom_logo(lapply(ovary_foxl2.meme_dna,as.character),seq_type='dna') + theme_logo() +
#  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Foxl2 ChIP Peaks")

ovary_foxl2.pfm<-consensusMatrix(reverseComplement(ovary_foxl2.meme_dna[[1]]))
ovary_foxl2.motif<-new("pfm",mat=t(t(ovary_foxl2.pfm[1:4,])*1/colSums(ovary_foxl2.pfm[1:4,])), name="Ovary FOXL2 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(foxl2_jaspar_motif,ovary_foxl2.motif),revcomp=c(F,F)))


sertoli_dmrt1.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dmrt1_top_sampled_151bp.txt")
x<-lengths(sertoli_dmrt1.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dmrt1_top_sampled_151bp.fasta"),unlist(sertoli_dmrt1.meme$motifList)),sertoli_dmrt1.meme$motifList))
names(sertoli_dmrt1.meme_dna)<-paste0("Motif ",1:length(sertoli_dmrt1.meme$motifList),", p-value: ",sertoli_dmrt1.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dmrt1.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Dmrt1 ChIP Peaks")

sertoli_dmrt1.pfm<-consensusMatrix(reverseComplement(sertoli_dmrt1.meme_dna[[1]]))
sertoli_dmrt1.motif<-new("pfm",mat=t(t(sertoli_dmrt1.pfm[1:4,])*1/colSums(sertoli_dmrt1.pfm[1:4,])), name="Sertoli Dmrt1 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(dmrt1_invitro_motif,sertoli_dmrt1.motif),revcomp=c(F,F)))

system.time(sertoli_dmrt1.sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=sertoli_dmrt1.pfm[1:4,],score="85%")))
sapply(sertoli_dmrt1.sites,length)
sertoli_dmrt1.sites<-sort(unlist(GRangesList(sertoli_dmrt1.sites)))


#show similarity of Dmrt1 and Sox9 Motifs
plotMotifLogoStack(DNAmotifAlignment(list(sox9_jaspar_motif,testis_sox9.motif,dmrt1_invitro_motif,sertoli_dmrt1.motif),revcomp=c(F,F,F,F)))

```

## what motif is under Sertoli DARs without Dmrt1/Sox9 motifs
```{r}

# Dmrt
length(sertoli_dar_Dmrt1<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$peak =="D"])
sertoli_dar_Dmrt1<-sertoli_dar_Dmrt1[with(sertoli_dar_Dmrt1,order(padj)),]
sertoli_dar_Dmrt1$top<-c(rep("top",1000),rep("not_top",length(sertoli_dar_Dmrt1)-1000))

mcols(sertoli_dar_Dmrt1) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=top)) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)]))

sertoli_dar_Dmrt1<-peak_center(sertoli_dar_Dmrt1[1:1000])
sertoli_dar_Dmrt1.dna<-getSeq(mm10,sertoli_dar_Dmrt1+100)
names(sertoli_dar_Dmrt1.dna)<-paste0("sertoli_dar_Dmrt1_",1:length(sertoli_dar_Dmrt1.dna))
export(sertoli_dar_Dmrt1.dna,"sertoli_dar_Dmrt1_201bp.fasta")

sertoli_dar_Dmrt1.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dar_Dmrt1_201bp.txt")
x<-lengths(sertoli_dar_Dmrt1.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dar_Dmrt1_201bp.fasta"),unlist(sertoli_dar_Dmrt1.meme$motifList)),sertoli_dar_Dmrt1.meme$motifList))
names(sertoli_dar_Dmrt1.meme_dna)<-paste0("Motif ",1:length(sertoli_dar_Dmrt1.meme$motifList),", p-value: ",sertoli_dar_Dmrt1.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dar_Dmrt1.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Dmrt1 Figure 1")


# Sox9
length(sertoli_dar_Sox9<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$peak =="S"])
sertoli_dar_Sox9<-sertoli_dar_Sox9[with(sertoli_dar_Sox9,order(padj)),]
# sertoli_dar_Sox9$top<-c(rep("top",1000),rep("not_top",length(sertoli_dar_Sox9)-1000))

mcols(sertoli_dar_Sox9) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)))) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)]))

#sertoli_dar_Sox9<-peak_center(sertoli_dar_Sox9[1:1000])
sertoli_dar_Sox9.dna<-getSeq(mm10,sertoli_dar_Sox9+100)
names(sertoli_dar_Sox9.dna)<-paste0("sertoli_dar_Sox9_",1:length(sertoli_dar_Sox9.dna))
export(sertoli_dar_Sox9.dna,"sertoli_dar_Sox9_201bp.fasta")

sertoli_dar_Sox9.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dar_Sox9_201bp.txt")
x<-lengths(sertoli_dar_Sox9.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dar_Sox9_201bp.fasta"),unlist(sertoli_dar_Sox9.meme$motifList)),sertoli_dar_Sox9.meme$motifList))
names(sertoli_dar_Sox9.meme_dna)<-paste0("Motif ",1:length(sertoli_dar_Sox9.meme$motifList),", p-value: ",sertoli_dar_Sox9.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dar_Sox9.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 Figure 1")



length(sertoli_dar_wo_DS<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$peak =="N"])
sertoli_dar_wo_DS<-sertoli_dar_wo_DS[with(sertoli_dar_wo_DS,order(padj)),]
sertoli_dar_wo_DS$top<-c(rep("top",1000),rep("not_top",length(sertoli_dar_wo_DS)-1000))

mcols(sertoli_dar_wo_DS) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=top)) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)]))

sertoli_dar_wo_DS<-peak_center(sertoli_dar_wo_DS[1:1000])
sertoli_dar_wo_DS.dna<-getSeq(mm10,sertoli_dar_wo_DS+100)
names(sertoli_dar_wo_DS.dna)<-paste0("sertoli_dar_wo_DS_",1:length(sertoli_dar_wo_DS.dna))
export(sertoli_dar_wo_DS.dna,"sertoli_dar_wo_DS_201bp.fasta")

sertoli_dar_wo_DS.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dar_wo_DS_201bp.txt")
x<-lengths(sertoli_dar_wo_DS.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dar_wo_DS_201bp.fasta"),unlist(sertoli_dar_wo_DS.meme$motifList)),sertoli_dar_wo_DS.meme$motifList))
names(sertoli_dar_wo_DS.meme_dna)<-paste0("Motif ",1:length(sertoli_dar_wo_DS.meme$motifList),", p-value: ",sertoli_dar_wo_DS.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dar_wo_DS.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("wo Dmrt1/Sox9 binding")

sertoli_dar_wo_DS.pfm<-consensusMatrix(reverseComplement(sertoli_dar_wo_DS.meme_dna[[3]]))
sertoli_dar_wo_DS.motif<-new("pfm",mat=t(t(sertoli_dar_wo_DS.pfm[1:4,])*1/colSums(sertoli_dar_wo_DS.pfm[1:4,])), name="sertoli_dar_wo_DS")

plotMotifLogoStack(DNAmotifAlignment(list(tead2_jaspar_motif,sertoli_dar_wo_DS.motif),revcomp=c(F,F)))

#Scan for TEA motif

system.time(sertoli_dar_wo_DS.sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=sertoli_dar_wo_DS.pfm[1:4,],score="90%")))
sapply(sertoli_dar_wo_DS.sites,length)
sertoli_dar_wo_DS.sites<-sort(unlist(GRangesList(sertoli_dar_wo_DS.sites)))


```



#If a DAR is within 1kp of a gene is it mostly likely to be a sex-specific gene?


# invert thinking - find all atac peaks near differentially expressed genes
```{r mean_distance,eval=F}

cutoff<-list(c(0,1000),c(1000,10000),c(10000,100000))

stratify_distance<-function(co,d) {table(vivo_atac_subset[queryHits(d)[mcols(d)$distance >= co[1] & mcols(d)$distance < co[2]]]$class)}
#stratify_distance(c(0,100000),s)

#find nearest peaks for Sertoli Specific Genes
s<-distanceToNearest(vivo_atac_subset,c(sertoli_transcripts),ignore.strand=TRUE)
lst.s<-lapply(cutoff,stratify_distance,s)
indx.s <- sapply(lst.s, length)
res.s <- as.data.frame(do.call(rbind,lapply(lst.s, `length<-`,
                          max(indx.s))))
colnames(res.s) <- names(lst.s[[which.max(indx.s)]])
res.s$window<-as.character(cutoff)
res.s
  g1<-res.s %>% gather(celltype,n,-window) %>% 
    ggplot( aes(x=window, y=n, fill=celltype)) + 
    scale_fill_manual(values=c("gray","magenta","blue")) +
    geom_bar(stat="identity",position="dodge") + coord_flip() + 
    theme_bw() + theme(legend.position = "none") + ylim(c(0,1500)) +
    ggtitle("Binned Distances to Sertoli Specific Transcripts")
 
#find nearest peaks for Granulsa Specific Genes
g<-distanceToNearest(vivo_atac_subset,c(granulosa_transcripts),ignore.strand=TRUE)
lst.g<-lapply(cutoff,stratify_distance,g)
indx.g <- sapply(lst.g, length)
 res.g <- as.data.frame(do.call(rbind,lapply(lst.g, `length<-`,
                          max(indx.g))))
colnames(res.g) <- names(lst.g[[which.max(indx.g)]])
res.g$window<-as.character(cutoff)
res.g

g2<-res.g %>% gather(celltype,n,-window) %>% 
  ggplot( aes(x=window, y=n, fill=celltype)) + 
  scale_fill_manual(values=c("gray","magenta","blue")) +
  geom_bar(stat="identity",position="dodge") + 
  coord_flip() + scale_y_reverse(limits=c(1500,0)) + 
  theme_bw() + ggtitle("Binned Distances to Granulosa Specific Transcripts")
 
#plot back-to-back  
legend <- get_legend(g2)


ggdraw(plot_grid(plot_grid(g2 + theme(legend.position = "none"), g1 + theme(legend.position = "none"), ncol=2, align='h'),
                 plot_grid(NULL, legend, ncol=1),
                 rel_widths=c(1, 0.2)))

```
### make Enriched Heatmap plots of Differentially ATAC-able regions

```{r}

cutoff<-1000


s<-distanceToNearest(vivo_atac_subset,sertoli_transcripts,ignore.strand=TRUE)
vivo_atac_subset$sertoli_distance<- NA 
vivo_atac_subset$nearest_sertoli_symbol<-NA
vivo_atac_subset$nearest_sertoli_gene_id<-NA
vivo_atac_subset[queryHits(s)]$sertoli_distance<-mcols(s)$distance 
vivo_atac_subset[queryHits(s)]$nearest_sertoli_symbol<-sertoli_transcripts[subjectHits(s)]$symbol
vivo_atac_subset[queryHits(s)]$nearest_sertoli_gene_id<-sertoli_transcripts[subjectHits(s)]$gene_id

g<-distanceToNearest(vivo_atac_subset,c(granulosa_transcripts),ignore.strand=TRUE)
vivo_atac_subset$granulosa_distance<- NA 
vivo_atac_subset$nearest_granulosa_symbol<-NA
vivo_atac_subset$nearest_granulosa_gene_id<-NA
vivo_atac_subset[queryHits(g)]$granulosa_distance<-mcols(g)$distance 
vivo_atac_subset[queryHits(g)]$nearest_granulosa_symbol<-granulosa_transcripts[subjectHits(g)]$symbol
vivo_atac_subset[queryHits(g)]$nearest_granulosa_gene_id<-granulosa_transcripts[subjectHits(g)]$gene_id

head(vivo_atac_subset)
#add fpkm for both gehes
vivo_atac_subset$sertoli_gene_sert_fpkm<-f_soma_mean[vivo_atac_subset$nearest_sertoli_gene_id,]$sertoli
vivo_atac_subset$sertoli_gene_gran_fpkm<-f_soma_mean[vivo_atac_subset$nearest_sertoli_gene_id,]$granulosa
vivo_atac_subset$sertoli_gene_stat<-results_soma[vivo_atac_subset$nearest_sertoli_gene_id,"stat"]
vivo_atac_subset$gran_gene_sert_fpkm<-f_soma_mean[vivo_atac_subset$nearest_granulosa_gene_id,]$sertoli
vivo_atac_subset$gran_gene_gran_fpkm<-f_soma_mean[vivo_atac_subset$nearest_granulosa_gene_id,]$granulosa
vivo_atac_subset$gran_gene_stat<-results_soma[vivo_atac_subset$nearest_granulosa_gene_id,"stat"]

length(s_regions<-vivo_atac_subset[queryHits(s)[mcols(s)$distance < cutoff]])

s_regions$sert_fpkm<-s_regions$sertoli_gene_sert_fpkm
s_regions$gran_fpkm<-s_regions$sertoli_gene_gran_fpkm
s_regions$stat<-s_regions$sertoli_gene_stat

length(g_regions<-vivo_atac_subset[queryHits(g)[mcols(g)$distance < cutoff]])

g_regions$sert_fpkm<-g_regions$gran_gene_sert_fpkm
g_regions$gran_fpkm<-g_regions$gran_gene_gran_fpkm
g_regions$stat<-g_regions$gran_gene_stat

head(g_regions)

#View(as.data.frame(g_regions[g_regions %over% s_regions]))
sum(g_regions %over% s_regions)

g_regions$deg<-"deg_gran"
s_regions$deg<-"deg_sert"
t_regions<-c(g_regions,s_regions)
t_regions<-t_regions[!t_regions %in% g_regions[g_regions %over% s_regions]]

length(t_regions)
#View(as.data.frame((t_regions)))
head(sort(table(s_regions$nearest_sertoli_symbol),decreasing = TRUE),20)
head(sort(table(g_regions$nearest_granulosa_symbol),decreasing = TRUE),20)

```

```{r}

temp_1<-t_regions
temp_1<-temp_1[temp_1$class!="Granulosa"]  #only constitutive and Sertoli
temp_1$class<-paste0(temp_1$class,":",temp_1$deg) %>% gsub("oli","",.) %>% gsub("deg_","",.) %>% gsub("Constitutive","Con",.) %>% gsub("ulosa","", .)
table(temp_1$class)
temp_2<-peak_center(temp_1) 

df<-data.frame(gran=temp_1$gran_fpkm,sert=temp_1$sert_fpkm,stat=temp_1$stat)

#g_atac_dmrt1<-rtracklayer::import("../data/gran_p7CAGDmrt1_n2_P037_ATACseq_N702.bigWig",selection=temp_2+2000)
#g_atac_control<-rtracklayer::import("../data/gran_p7control_n1_P037_ATACseq_N703.bigWig",selection=temp_2+2000)
  
g_atac_dmrt1<-rtracklayer::import("../data/fresh_sertoli_noMT_dedup_unique_merge.bigWig",selection=temp_2+2000) #sertoli
g_atac_control<-rtracklayer::import("../data/fresh_gran_noMT_dedup_unique_merge.bigWig",selection=temp_2+2000) #granulosa
g_p2<-rtracklayer::import("../data/sox9_Index12_DSG_PFA_062315_TACAAG.dedup.unique.bigWig",selection=temp_2+2000)
#g_p3<-rtracklayer::import("../data/ovaryav574_CAGATC.dedup.unique.bigWig",selection=temp_2+2000)

g_p4<-rtracklayer::import("../data/k27ac_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p5<-rtracklayer::import("../data/k27ac_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

seqlevelsStyle(g_atac_dmrt1)<-"UCSC"
seqlevelsStyle(g_atac_control)<-"UCSC"
seqlevelsStyle(g_p2)<-"UCSC"
#seqlevelsStyle(g_p3)<-"UCSC"
seqlevelsStyle(g_p4)<-"UCSC"
seqlevelsStyle(g_p5)<-"UCSC"

mat_dmrt1=normalizeToMatrix(sertoli_dmrt1.sites, temp_2, value_column = "score", mean_mode = "absolute",
   extend = 2000, w = 50, smooth = FALSE)
#mat_dmrt1_open=normalizeToMatrix(group_a_sites, temp_2, value_column = "score", mean_mode = "absolute",
#   extend = 2000, w = 50, smooth = FALSE)
mat_sox9=normalizeToMatrix(testis_sox9.sites, temp_2, value_column = "score", mean_mode = "absolute",
   extend = 2000, w = 50, smooth = FALSE)
#mat_sox9_open=normalizeToMatrix(group_c_sites, temp_2, value_column = "score", mean_mode = "absolute",
#   extend = 2000, w = 50, smooth = FALSE)
mat_tea=normalizeToMatrix(sertoli_dar_wo_DS.sites, temp_2, value_column = "score", mean_mode = "absolute",
   extend = 2000, w = 50, smooth = FALSE)
#mat_foxl2=normalizeToMatrix(foxl2_pwm, temp_2, value_column = "score", mean_mode = "absolute",
#   extend = 2000, w = 50, smooth = FALSE)


seqlevelsStyle(temp_2)<-"UCSC"
g_p1<-rtracklayer::import("../data/covaris_6c_sertoli_dmrt1ChIP.R1.fastq.dedup.unique.ucsc.bigWig",selection=temp_2+2000)



mat_p1 = normalizeToMatrix(g_p1, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p2 = normalizeToMatrix(g_p2, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
#mat_p3 = normalizeToMatrix(g_p3, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p4 = normalizeToMatrix(g_p4, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p5 = normalizeToMatrix(g_p5, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_ad = normalizeToMatrix(g_atac_dmrt1, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_ac = normalizeToMatrix(g_atac_control, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)




col_fun_p1= colorRamp2(quantile(mat_p1, c(0, 0.99)), c("white", "darkblue"))
col_fun_p2= colorRamp2(quantile(mat_p2, c(0, 0.99)), c("white", "green"))
#col_fun_p3= colorRamp2(quantile(mat_p3, c(0, 0.99)), c("white", "magenta"))
col_fun_p4= colorRamp2(quantile(mat_p4, c(0, 0.99)), c("white", "black"))
col_fun_p5= colorRamp2(quantile(mat_p5, c(0, 0.99)), c("white", "black"))
col_fun_ac= colorRamp2(quantile(mat_ac, c(0, 0.99)), c("white", "red"))
col_fun_ad= colorRamp2(quantile(mat_ad, c(0, 0.99)), c("white", "blue"))

#mysplit<-ifelse(temp$is.sig,"DeNovo","PreExisting")
#mysplit<-ifelse(g_regions$class,"DeNovo","PreExisting")

col=cbPalette[c(3,5,7,4,6,8)]
#col=c("red","green","blue")
  dmrt1_order<-row_order(  EnrichedHeatmap(mat_p1, col = col_fun_p1, name = "dmrt1",column_title = "Dmrt1 (Sertoli)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,18),yaxis = F)) ))
    
Heatmap(df$gran,name="FPKM Gran",column_title = "Granulosa Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20,row_order = as.integer(unlist(dmrt1_order)),cluster_rows = FALSE) + 
Heatmap(df$sert,name="FPKM Sert",column_title = "Sertoli Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20)  +
  
EnrichedHeatmap(mat_ac, col = col_fun_ac, name = "atac_control",column_title = "Granulosa",axis_name_rot = 90,split=temp_1$class,pos_line=F,column_title_rot=90,show_heatmap_legend = FALSE,
              top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,13),yaxis = T)) ) +
EnrichedHeatmap(mat_ad, col = col_fun_ad, name = "atac_dmrt1",column_title = "Sertoli",axis_name_rot = 90,split=temp_1$class,pos_line=F,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,25),yaxis = T)) ) +
EnrichedHeatmap(mat_p1, col = col_fun_p1, name = "dmrt1",column_title = "Dmrt1 (Sertoli)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,9),yaxis = F)) ) +
EnrichedHeatmap(mat_p2, col = col_fun_p2, name = "Sox9",column_title = "Sox9 (Sertoli)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0.2,0.8),yaxis = T)) ) +
#EnrichedHeatmap(mat_p3, col = col_fun_p3, name = "Foxl2",column_title = "Foxl2 (Ovary)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,1.5),yaxis = F)) ) +
EnrichedHeatmap(mat_p4, col = col_fun_p4, name = "k27nc",column_title = "k27 noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,10),yaxis = F)) ) +
EnrichedHeatmap(mat_p5, col = col_fun_p5, name = "k27dhh",column_title = "k27 dHHCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,8),yaxis = F)) ) +
EnrichedHeatmap(mat_dmrt1, col = c("white", "darkblue"),pos_line=F, name = "dmrt1_sites", axis_name_rot = 90,column_title = "Dmrt1 Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) +
EnrichedHeatmap(mat_sox9, col = c("white", "darkgreen"),pos_line=F, name = "sox9_sites", axis_name_rot = 90,column_title = "Sox9 Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) +
#EnrichedHeatmap(mat_sox9_open, col = c("white", "darkgreen"),pos_line=F, name = "group_c_sites", axis_name_rot = 90,column_title = "group c Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) +
EnrichedHeatmap(mat_tea, col = c("white", "darkblue"),pos_line=F, name = "tea_sites", axis_name_rot = 90,column_title = "TEA Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) 

    
#EnrichedHeatmap(mat_foxl2, col = c("white", "magenta"),pos_line=F, name = "foxl2_sites", axis_name_rot = 90,column_title = "Foxl2 Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = T)) ) 
  
  
  #Heatmap(df$gran,name="FPKM Gran",column_title = "Granulosa Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20) + 
  #Heatmap(df$sert,name="FPKM Sert",column_title = "Sertoli Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20)


#  EnrichedHeatmap(mat_p3, col = col_fun_p3, name = "7d Sox9 - Sox9",column_title = "7d Sox9 - Sox9",axis_name_rot = 90,pos_line=F,split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE) +
#  EnrichedHeatmap(mat_p2, col = col_fun_p2, name = "7d Dmrt1 - Sox9",column_title = "7d Dmrt1 - Sox9",pos_line=F,axis_name_rot = 90,split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE) +
#  EnrichedHeatmap(mat_dmrt1, col = c("white", "black"),pos_line=F, name = "dmrt1_sites", axis_name_rot = 90,column_title = "dmrt1_sites",split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE) +
#  EnrichedHeatmap(mat_sox9, col = c("white", "black"),pos_line=F, name = "sox9_sites", axis_name_rot = 90,column_title = "sox9_sites",split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE)
  
```

## Heatmap for Chromatin Only
```{r}
seqlevelsStyle(temp_2)<-"NCBI"
g_p1<-rtracklayer::import("../data/input_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p2<-rtracklayer::import("../data/input_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p3<-rtracklayer::import("../data/k4me1_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p4<-rtracklayer::import("../data/k4me1_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p5<-rtracklayer::import("../data/k27ac_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p6<-rtracklayer::import("../data/k27ac_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p7<-rtracklayer::import("../data/k4me3_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p8<-rtracklayer::import("../data/k4me3_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p9<-rtracklayer::import("../data/k27me3_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p10<-rtracklayer::import("../data/k27me3_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

mat_p1 = normalizeToMatrix(g_p1, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p2 = normalizeToMatrix(g_p2, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p3 = normalizeToMatrix(g_p3, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p4 = normalizeToMatrix(g_p4, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p5 = normalizeToMatrix(g_p5, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p6 = normalizeToMatrix(g_p6, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p7 = normalizeToMatrix(g_p7, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p8 = normalizeToMatrix(g_p8, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p9 = normalizeToMatrix(g_p9, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p10 = normalizeToMatrix(g_p10, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)

col_fun_p1= colorRamp2(quantile(mat_p1, c(0, 0.99)), c("white", "black"))
col_fun_p2= colorRamp2(quantile(mat_p2, c(0, 0.99)), c("white", "black"))
col_fun_p3= colorRamp2(quantile(mat_p3, c(0, 0.99)), c("white", "magenta"))
col_fun_p4= colorRamp2(quantile(mat_p4, c(0, 0.99)), c("white", "magenta"))
col_fun_p5= colorRamp2(quantile(mat_p5, c(0, 0.99)), c("white", "blue"))
col_fun_p6= colorRamp2(quantile(mat_p6, c(0, 0.99)), c("white", "blue"))
col_fun_p7= colorRamp2(quantile(mat_p7, c(0, 0.99)), c("white", "green"))
col_fun_p8= colorRamp2(quantile(mat_p8, c(0, 0.99)), c("white", "green"))
col_fun_p9= colorRamp2(quantile(mat_p9, c(0, 0.99)), c("white", "red"))
col_fun_p10= colorRamp2(quantile(mat_p10, c(0, 0.99)), c("white", "red"))

col=cbPalette[c(3,5,7,4,6,8)]
#ggplot(data.frame(x=rep(10,length(col)),y=letters[1:length(col)]),aes(x=x,fill=y)) + geom_bar() + scale_fill_manual(values=col) + theme_bw()

EnrichedHeatmap(mat_p1, col = col_fun_p1, name = "input_noCre",column_title = "input noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p2, col = col_fun_p2, name = "input_dhhCre",column_title = "input dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p3, col = col_fun_p3, name = "k4me1_noCre",column_title = "k4me1 noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p4, col = col_fun_p4, name = "k4me1_dhhCre",column_title = "k4me1 dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p5, col = col_fun_p5, name = "k27ac_noCre",column_title = "k27ac noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,7.5),yaxis = F)) ) +
EnrichedHeatmap(mat_p6, col = col_fun_p6, name = "k27ac_dhhCre",column_title = "k27ac dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,7.5),yaxis = F)) ) +
EnrichedHeatmap(mat_p7, col = col_fun_p7, name = "k4me3_noCre",column_title = "k4me3 noCre",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,12),yaxis = F)) ) +
EnrichedHeatmap(mat_p8, col = col_fun_p8, name = "k4me3_dhhCre",column_title = "k4me3_dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,12),yaxis = F)) ) +
EnrichedHeatmap(mat_p9, col = col_fun_p9, name = "k27me3_noCre",column_title = "k27me3 noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,2),yaxis = F)) ) +
EnrichedHeatmap(mat_p10, col = col_fun_p10, name = "k27me3_dhhCre",column_title = "k27me3_dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,2),yaxis = F)) ) 

```

# Tn5 footprinting
```{r}
bamfile<-"../data/sertoli_fresh_ATAC_rep1_N704_GW052319.dedup.unique.noMT.bam"
bamfile.labels <- gsub(".bam", "", basename(bamfile))
seqlevelsStyle(Mmusculus)<-"NCBI"
seqlevel<-c(1:19,"X","Y")
which<-as(seqinfo(Mmusculus)[seqlevel],"GRanges")
possibleTag <- combn(LETTERS, 2)
possibleTag <- c(paste0(possibleTag[1, ], possibleTag[2, ]),
                 paste0(possibleTag[2, ], possibleTag[1, ]))
bamTop100 <- scanBam(BamFile(bamfile, yieldSize = 10000),
                     param = ScanBamParam(tag=possibleTag))[[1]]$tag
(tags <- names(bamTop100)[lengths(bamTop100)>0])

gal <- readBamFile(bamfile, tag=tags, which=which, asMates=TRUE, bigFile=TRUE)

outPath <- "splited"
dir.create(outPath)
shiftedBamfile <- file.path(outPath, "shifted.bam")
gal1 <- shiftGAlignmentsList(gal, outbam=shiftedBamfile)

pt <- PTscore(gal1, transcripts)
plot(pt$log2meanCoverage, pt$PT_score, cex=0.5,pch=16,
     xlab="log2 mean coverage",
     ylab="Promoter vs Transcript")

nfr <- NFRscore(gal1, transcripts)
  plot(nfr$log2meanCoverage, nfr$NFR_score, 
     xlab="log2 mean coverage",cex=0.5,pch=16,
     ylab="Nucleosome Free Regions score",
     main="NFRscore for 200bp flanking TSSs",
     xlim=c(-10, 10), ylim=c(-8, 8))

  fa<-readDNAStringSet("murphy_2007_site_selection.fa")
fa<-subseq(fa,3,15)
dmrt1_pfm<-consensusMatrix(fa)[1:4,]
dmrt1_pfm<-t(t(dmrt1_pfm[1:4,])*1/colSums(dmrt1_pfm[1:4,]))
  
  sigs <- factorFootprints(gal1, pfm=dmrt1_pfm, 
                         genome=mm10,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)
  
```


