---
title: "Figure 1 - Sertoli vs Granulosa"
output: html_document
editor_options: 
  chunk_output_type: console
creative_commons: CC BY
---
# Figure 1 - Granulosa vs Sertolii

## Use Fresh Sertoli / granulosa data to define somatic genes that are differentially expressed

```{r somatic_gonad,eval=T}
#use differential expression between granulosa and Sertoli to define genes of interest
as.data.frame(colData(dds_soma<-dds[,str_detect(colData(dds)$time,"fresh")]))
#plotPCA(normTransform(dds_soma),intgroup=c("gene")) + theme_bw()
dds_soma$gene <- droplevels(dds_soma$gene)
design(dds_soma)<- ~ gene 
dds_soma<-DESeq(dds_soma)

#fpkm table
#get fpkm
dim(f_soma<-as.data.frame(log2(fpkm(dds_soma)+0.1)))


#colnames(f_gonad)<-paste(colData(dds_gonad)$time,colData(dds_gonad)$gene,colData(dds_gonad)$sox,1:nrow(colData(dds_gonad)),sep="_")
#f_gonad$mgi_symbol<-mcols(dds_gonad)$mgi_symbol
#View(f_gonad)

    f_soma_mean <-f_soma %>% 
      tibble::rownames_to_column(var="ensembl") %>%
      tidyr::gather(sample,fpkm,-ensembl) %>% 
      dplyr::mutate(sex=colData(dds_soma)[sample,]$gene) %>% 
      dplyr::mutate(sex=gsub("control","granulosa",sex)) %>% 
      dplyr::group_by(sex,ensembl) %>% 
        dplyr::summarize(mean_fpkm=round(mean(fpkm),3)) %>% 
        dplyr::select(sex,ensembl,mean_fpkm) %>% 
        ungroup() %>%
    tidyr::spread(sex,mean_fpkm) %>% as.data.frame()
  
    rownames(f_soma_mean) <- f_soma_mean$ensembl
    f_soma_mean<-f_soma_mean[,-1]
  
f_soma_mean$symbol<-mcols(dds_soma)[match(rownames(f_soma_mean),rownames(dds_soma)),]$symbol
length(high_soma_fpkm<-rownames(f_soma_mean[apply(f_soma_mean,1,max) > log2(2.5),]))
table(symbols[symbols$gene_id %in% high_soma_fpkm,"chr"])


#DEGs
resultsNames(dds_soma)
(summary(results_soma<-results(dds_soma)))
results_soma$mgi_symbol<-mcols(dds_soma)$symbol
#results_soma<-results_soma[!(is.na(results_soma$log2FoldChange) | is.na(results_soma$padj)),]
results_soma<-results_soma[with(results_soma,order(padj)),]

results_soma$f_sertoli<-f_soma_mean[match(rownames(results_soma),rownames(f_soma_mean)),]$sertoli
results_soma$f_granulosa<-f_soma_mean[match(rownames(results_soma),rownames(f_soma_mean)),]$granulosa

#View(as.data.frame(results_soma))


#write.csv(results_soma,file=paste0("granulosa_vs_Sertoli_comparison_",ts,".csv"),quote=F)



#define a list of Sertoli Up genes for later
length(sertoli_up<-rownames(subset(results_soma,log2FoldChange > 2 & padj < 0.05 & f_sertoli > 2.5 )))
table(symbols[symbols$gene_id %in% sertoli_up,"chr"])
length(granulosa_up<-rownames(subset(results_soma,log2FoldChange < -2 & padj < 0.05 &  f_granulosa > 2.5)))
table(symbols[symbols$gene_id %in% granulosa_up,"chr"])

#define transcripts fore each set
sertoli_transcripts<-transcripts[transcripts$gene_id %in% sertoli_up]
granulosa_transcripts<-transcripts[transcripts$gene_id %in% granulosa_up]

#View(as.data.frame(results_soma[sertoli_up,]))


nrow(temp<-as.data.frame(results_soma))

#subset to significant genes
nrow(results_soma<-subset(results_soma,abs(log2FoldChange) > 2 & padj < 0.05 & (f_sertoli > 2.5 | f_granulosa > 2.5)))
table(symbols[symbols$gene_id %in% rownames(results_soma),"chr"])

table(temp$results_soma<- factor(case_when (
  temp$log2FoldChange > 2 & temp$padj < 0.05 & temp$f_sertoli > 2.5 ~ "Sertoli",
  temp$log2FoldChange < -2 & temp$padj < 0.05 & temp$f_granulosa > 2.5 ~ "granulosa",
  TRUE ~ "unchanged"
  ),levels=c("granulosa","unchanged","Sertoli")))
  

(df<-temp[temp$mgi_symbol %in% c("Dmrt1","Nr5a2","Esr2","Foxl2","Sox8","Sox9","Cyp19a1","Cytip","Hsd17b3","Wnt5b", "Sdc3", "Dhh", "Cst9", "Rspo2","Bmpr1b","Ocln"),])

svglite::svglite(paste0("somatic_volcano_",ts,".svg"),width=4,height=3)
temp %>% 
  dplyr::filter(!is.na(log2FoldChange) & !is.na(padj)) %>% 
ggplot(aes(x=log2FoldChange,y=(-1*log10(padj)),color=results_soma)) + geom_point(size=0.7) + theme_bw() + scale_color_manual(values=c("red","darkgray","blue")) +
  annotate("text",x=df$log2FoldChange,y=(-1*log10(df$padj)),label=df$mgi_symbol)
nodev.off()


temp<-tibble::rownames_to_column(temp,var="Ensembl_GeneID")
temp<-temp[,c("Ensembl_GeneID","mgi_symbol","baseMean","log2FoldChange","padj","f_sertoli","f_granulosa","results_soma")]
colnames(temp)<-c("Ensembl_GeneID","Gene Symbol","Mean Counts","Log2FoldChange","BH Padj","FPKM Sertoli","FPKM Granulosa","Sex Bias")
head(temp)

write.csv(temp,"Lindeman_et_al_Sex_Biased_Somatic_Genes.csv",row.names=F,quote=F)
```


## UpsetR comparison of ATACseq Peaklist
```{r vivo_ATACseq_upsetR,eval=T}

length(vivo_atac<-reduce(c(peak_e105_atac_XX,peak_e105_atac_XY,peak_e135_atac_XX,peak_e135_atac_XY,peak_p23_atac_XX,peak_p7_atac_XY)))

length(vivo_atac<-keepSeqlevels(vivo_atac,c(1:19,"X"),pruning.mode="coarse"))  #drop Y because not all samples have a Y
length(vivo_atac<-suppressWarnings(vivo_atac[!vivo_atac %over% mito_homologs]))

head(names(vivo_atac)<-paste0("vivo_atac",formatC(1:length(vivo_atac),width=6,format="d",flag="0")))


summary(width(vivo_atac))


(fls.k27ac_p3_Sert <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>%
   grep("k27",.,value=T) %>%
   grep("wt",.,value=T) %>%
   grep("GW02",.,value=T) %>%
   grep("P3",.,value=T ))

#system.time(vivo_atac_k27ac_counts<- summarizeOverlaps(vivo_atac,BamFileList(fls.k27ac_p3_Sert,yieldSize = 1e5),mode =Union, ignore.strand=T))
#as.data.frame(apply(assays(vivo_atac_k27ac_counts)$counts,2,sum))
#save(vivo_atac_k27ac_counts,file="vivo_atac_k27ac_counts.rdata")
load("vivo_atac_k27ac_counts.rdata")


temp<-as.data.frame(assays(vivo_atac_k27ac_counts)$counts)
colnames(temp)<-c("wt_idx13","wt_idx14")
head(temp)
colSums(temp)
temp<-sweep(temp,2,colSums(temp)/1e6,"/")
temp$width<-width(vivo_atac)
temp2<-data.frame(p3_XY_k27ac=(temp$wt_idx13 + temp$wt_idx14)/(0.002*temp$width))
head(temp2)
mcols(vivo_atac)<-log2(temp2+0.5)


vivo_atac$e105_XX<-as.numeric(vivo_atac %over% peak_e105_atac_XX)
vivo_atac$e105_XY<-as.numeric(vivo_atac %over% peak_e105_atac_XY)
vivo_atac$e135_XX<-as.numeric(vivo_atac %over% peak_e135_atac_XX)
vivo_atac$e135_XY<-as.numeric(vivo_atac %over% peak_e135_atac_XY)
vivo_atac$p23_XX<-as.numeric(vivo_atac %over% peak_p23_atac_XX)
vivo_atac$p7_XY<-as.numeric(vivo_atac %over% peak_p7_atac_XY)

colSums(as.data.frame(mcols(vivo_atac)))

upset(as.data.frame(mcols(vivo_atac)), nsets = 4, number.angles = 30, point.size = 3.5, line.size = 2,
    mainbar.y.label = "Peak Intersections", main.bar.color = "black",
    sets=c("p7_XY","p23_XX","e135_XY","e135_XX","e105_XY","e105_XX"), sets.x.label = "Numebr of Peaks", keep.order = TRUE,
    queries=list(list(query = intersects, params = list("p7_XY", "e135_XY"),color = "blue", active = T)),text.scale = c(1.3, 1.3, 1, 1, 2, 0.75),
     boxplot.summary = c("p3_XY_k27ac"))

```




## Combine Embryonic Data with WT vs Sertoli DARs
### Map Genes to DARs
### Define g_regions and s_regions for DARs close to granulosa expressed genes and Sertoli expressed genes

```{r in_vivo_DARs_DDS,eval=T}
#Find Differentially bound ATAC-able regions

#(fls1.vivo <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("/N70"))
(fls1.vivo <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("GW052319") %>% str_subset("fresh"))
(fls2.vivo <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("futtner"))
#(fls3.vitro <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("ATAC") %>% str_subset("gran") %>% str_subset("P043"))
(fls3.vitro <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% str_subset("GW052319") %>% str_subset("cagSox9"))
(fls4.vitro <- list.files("../data", pattern=glob2rx("*.bam$"),full=TRUE) %>% 
   grep("ATAC",.,value=T) %>% 
   grep("marked",.,value=T,invert=T) %>% 
   grep("P037",.,value=T) %>% 
   grep("gran",.,value=T ))

  #system.time(atac_counts<- summarizeOverlaps(vivo_atac,BamFileList(c(fls1.vivo,fls2.vivo),yieldSize = 1e5),mode =Union, ignore.strand=T))
#system.time(atac_counts<- summarizeOverlaps(vivo_atac,BamFileList(c(fls1.vivo,fls2.vivo,fls3.vitro,fls4.vitro),yieldSize = 1e5),mode =Union, ignore.strand=T))
#apply(assays(atac_counts)$counts,2,sum)
#save(atac_counts,file="atac_counts.rdata")
load("atac_counts.rdata")

#mcols(vivo_atac)<-assays(atac_counts)$counts
#colnames(mcols(vivo_atac))<-colnames(assays(atac_counts)$counts) %>% gsub("\\.R1_trimmed\\.fastq\\.marked\\.bam","",.)
atac_dds<-DESeqDataSet(atac_counts,design = ~1)
mcols(atac_dds)$basepairs<-width(vivo_atac)
colData(atac_dds)$sample<-colnames(assays(atac_counts)$counts) %>% gsub("\\.R1_trimmed\\.fastq\\.marked\\.bam","",.) %>% 
  gsub("futtner_ATAC_","",.) %>% gsub("_R1_trimmed\\.bam","",.) %>% 
  gsub("\\.marked\\.bam","",.) %>%  gsub("\\.bam","",.) 

colData(atac_dds)$sex<-factor(c(rep("Female",3),rep("Male",3),rep("Female",2),rep("Male",2),rep("Female",2),rep("Male",2),rep("Female",12)))
colData(atac_dds)$age<-factor(c(rep("postnatal",6),rep("E105",4),rep("E135",4),rep("postnatal",12)),levels=c("E105","E135","postnatal"))
colData(atac_dds)$cag<-factor(c(rep("na",14),rep("Sox9",3),rep("na",5),"Dmrt1","na",rep("Dmrt1",2)),levels=c("na","Sox9","Dmrt1"))
colData(atac_dds)$source <-factor(c(rep("vivo",14),rep("vitro",12)),levels=c("vivo","vitro"))

#colData(atac_dds)$sex<-factor(c(rep("Male",2),rep("Female",4),rep("Male",2),rep("Female",2),rep("Male",2),rep("Female",12)))
#colData(atac_dds)$age<-factor(c(rep("postnatal",4),rep("E105",4),rep("E135",4),rep("postnatal",12)),levels=c("E105","E135","postnatal"))
#colData(atac_dds)$cag<-factor(c(rep("na",12),rep("Sox9",3),rep("na",5),"Dmrt1","na",rep("Dmrt1",2)),levels=c("na","Sox9","Dmrt1"))

colData(atac_dds)$group<-factor(paste0(colData(atac_dds)$age,"_",colData(atac_dds)$sex,"_",colData(atac_dds)$source,"_",colData(atac_dds)$cag),
                        levels=c("E105_Female_vivo_na","E105_Male_vivo_na","E135_Female_vivo_na","E135_Male_vivo_na","postnatal_Female_vivo_na","postnatal_Male_vivo_na","postnatal_Female_vitro_na","postnatal_Female_vitro_Sox9","postnatal_Female_vitro_Dmrt1"))

as.data.frame(colData(atac_dds))

#Move to Figure 5
#(df<-plotPCA(normTransform(atac_dds),intgroup=c("group"),returnData=T,ntop=500))
#
#plotPCA(normTransform(atac_dds),intgroup=c("group"),ntop=500) + theme_bw() +
#  scale_color_manual(values = c(cbPalette,"red")) + geom_text(x=df$PC1,y=df$PC2,label=df$group) 


```

## Make an  in vivo version of this PCA plot for Figure 1 (full version moved to figure 7)
```{r in_vivo_DARs_PCA,eval=T}
as.data.frame(colData(vivo_dds<-atac_dds[,colData(atac_dds)$source=="vivo"]))


as.data.frame(colData(vivo_dds))
#table(as.character(seqnames(vivo_atac)) %in% as.character(c(1:19,"X")))
#dim(vivo_dds<-vivo_dds[as.character(seqnames(vivo_atac)) %in% as.character(c(1:19,"X")),])
(df<-plotPCA(normTransform(vivo_dds),intgroup=c("sex","age"),returnData=T,ntop=500))

#df$sample<-df$sample %>% gsub("N702","Sertoli_N702_p7",.) %>%
#   gsub("N703","Sertoli_N703_p7",.) %>% 
#   gsub("N704","Granulosa_N704_p23",.) %>% 
#   gsub("N705","Granulosa_N705_p23",.)


plotPCA(normTransform(vivo_dds),intgroup=c("sex","age"),ntop=500) + theme_bw() +
scale_color_manual(values = cbPalette[c(1:6)]) + geom_text(x=df$PC1,y=df$PC2,label=paste0(df$sex,"_",df$age)) 



```

## Use contrast to find differentially atac-able regions between granulosa and Sertoli
```{r in_vivo_DARs_DESeq,eval=T}
f_atac_vivo <- fpkm(vivo_dds,robust=TRUE)

f_mean_atac_vivo <- f_atac_vivo %>% as.data.frame() %>% 
  tibble::rownames_to_column(var="region") %>% 
  tidyr::gather(sample,fpkm,-region) %>%
  dplyr::mutate(group=colData(vivo_dds)[sample,]$group) %>% 
  dplyr::group_by(region,group) %>% 
    dplyr::summarize(mean_fpkm=round(log2(mean(fpkm)+0.01),3)) %>% 
    ungroup() %>% 
  dplyr::select(region,group,mean_fpkm) %>% 
  tidyr::spread(group,mean_fpkm) %>%  as.data.frame() 

rownames(f_mean_atac_vivo)<-f_mean_atac_vivo$region
f_mean_atac_vivo<-f_mean_atac_vivo[,-1]
colnames(f_mean_atac_vivo)<- colnames(f_mean_atac_vivo) %>% 
  gsub("_vivo_na","",.) %>% 
  gsub("postnatal_Female","p23_Female",.) %>% 
  gsub("postnatal_Male","p7_Male",.)
  
head(f_mean_atac_vivo)

#Add metadata to vivo_atac gr
stopifnot(all.equal(rownames(f_mean_atac_vivo),names(vivo_atac)))
mcols(vivo_atac)<-f_mean_atac_vivo

# Differential Expression
design(vivo_dds)<-~group
vivo_dds$group<-droplevels(vivo_dds$group)
vivo_dds<-DESeq(vivo_dds)


#e13.5
summary(res_vivo_E135<-results(vivo_dds,contrast=c("group","E135_Male_vivo_na","E135_Female_vivo_na"),alpha=0.05))
temp_E135<-vivo_atac
stopifnot(all.equal(rownames(res_vivo_E135),names(temp_E135)))
temp_E135$lfc<-round(res_vivo_E135$log2FoldChange,3)
temp_E135$padj<-res_vivo_E135$padj
#are they called in the Sert or Granulosa Datasets
table(temp_E135$sert_call <- temp_E135 %over% peak_e135_atac_XY)
table(temp_E135$gran_call <- temp_E135 %over% peak_e135_atac_XX)

table(temp_E135$class<-case_when (
  temp_E135$sert_call & temp_E135$padj < 0.05 & temp_E135$lfc > 1 & !temp_E135$gran_call ~ "Sertoli", #3.3,1 vs 4.3,2.8
  temp_E135$gran_call & temp_E135$padj < 0.05 & temp_E135$lfc < -1 &  !temp_E135$sert_call ~ "Granulosa", # vs 2.8,4.3
  abs(temp_E135$lfc) < log2(1.25) & temp_E135$sert_call & temp_E135$gran_call ~ "Constitutive",
  TRUE ~ "unclassified"
))

volcano_E135<-mcols(temp_E135) %>% as.data.frame() %>% 
  dplyr::filter(class != "unclassified") %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=class)) + geom_point(size=0.75) + theme_bw()  +ggtitle("E135")
  
#postnatal
summary(res_vivo_postnatal<-results(vivo_dds,contrast=c("group","postnatal_Male_vivo_na","postnatal_Female_vivo_na"),alpha=0.05))

stopifnot(all.equal(rownames(res_vivo_postnatal),names(vivo_atac)))
vivo_atac$lfc<-round(res_vivo_postnatal$log2FoldChange,3)
vivo_atac$padj<-res_vivo_postnatal$padj

#are they called in the Sert or Granulosa Datasets
table(vivo_atac$sert_call <- vivo_atac %over% peak_p7_atac_XY)
table(vivo_atac$gran_call <- vivo_atac %over% peak_p23_atac_XX)

table(vivo_atac$class<-case_when (
  vivo_atac$sert_call & vivo_atac$padj < 0.05 & vivo_atac$lfc > 1 & !vivo_atac$gran_call ~ "Sertoli", #3.3,1 vs 4.3,2.8
  vivo_atac$gran_call & vivo_atac$padj < 0.05 & vivo_atac$lfc < -1 &  !vivo_atac$sert_call ~ "Granulosa", # vs 2.8,4.3
  abs(vivo_atac$lfc) < log2(1.25) & vivo_atac$sert_call & vivo_atac$gran_call ~ "Constitutive",
  TRUE ~ "unclassified"
))

volcano_postnatal<-mcols(vivo_atac) %>% as.data.frame() %>% 
  dplyr::filter(class != "unclassified") %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=class)) + geom_point(size=0.75) + theme_bw()+ ggtitle("postnatal")

grid.arrange(volcano_E135,volcano_postnatal,ncol=1)
```


## Add ChIP Overlap Data to vivo_atac

```{r in_vivo_DARs_ChIP_overlap,eval=T}


head(as.data.frame(vivo_atac))
#View(as.data.frame(vivo_atac))
vivo_atac_subset<-vivo_atac[vivo_atac$class!="unclassified"]
#switch to using the single nearest gene
vivo_atac_subset<-annotate_peaks(vivo_atac_subset)  
table(vivo_atac_subset$dimorph)
#View(as.data.frame(vivo_atac_subset[vivo_atac_subset$dimorph=="Gran;Sert"]))
#vivo_atac_subset["vivo_atac029624"]$dimorph<-"Sert"
#vivo_atac_subset["vivo_atac054417"]$dimorph<-"Gran"
#table(vivo_atac_subset$dimorph)

length(vivo_atac_subset[vivo_atac_subset$class=="Granulosa" & vivo_atac_subset$dimorph=="Gran"])
length(vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$dimorph=="Gran"])
length(vivo_atac_subset[vivo_atac_subset$class=="Granulosa" & vivo_atac_subset$dimorph=="Sert"])
length(vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$dimorph=="Sert"])

#View(as.data.frame(vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$dimorph=="Gran"]))
#View(as.data.frame(vivo_atac_subset[vivo_atac_subset$class=="Granulosa" & vivo_atac_subset$dimorph=="Gran"]))


nr5a2_peaks<-vivo_atac[vivo_atac %over% swl("chr1:136803128-137029332")]
nr5a2_peaks[nr5a2_peaks %over% sertoli_dmrt1]
rspo2_peaks<-vivo_atac[vivo_atac %over% swl("chr15:42,953,041-43,193,844")]
rspo2_peaks[rspo2_peaks %over% sertoli_dmrt1]
(esr2_peaks<-vivo_atac[vivo_atac %over% swl("chr12:76,130,887-76,158,648")])

#phastcons
#vivo_atac_subset<-add_mean_phastcons_score(vivo_atac_subset)

table(vivo_atac_subset$class)
export(vivo_atac_subset[vivo_atac_subset$class != "Constitutive"],paste0("vivo_atac_subset_",ts,".bed"))

#Which ones overlap what sites?
table(vivo_atac_subset$dmrt1<-vivo_atac_subset %over% sertoli_dmrt1)
table(vivo_atac_subset$sox9<-vivo_atac_subset %over% testis_sox9)
table(vivo_atac_subset$foxl2<-vivo_atac_subset %over% ovary_foxl2)
table(vivo_atac_subset$esr2<-vivo_atac_subset %over% ovary_esr2)

table(vivo_atac_subset$peak<-case_when (
  vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "DSFE", 
  vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "DSE", 
  vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "DE", 
  !vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "SFE", 
  vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "DFE", 
  !vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "SE",
  !vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "FE",
  !vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & vivo_atac_subset$esr2 ~ "E",
  vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "DS",
  vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "D",
  !vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "SF",
  vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "DF",
  !vivo_atac_subset$dmrt1 & vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "S",
  !vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "F",
  !vivo_atac_subset$dmrt1 & !vivo_atac_subset$sox9 & !vivo_atac_subset$foxl2 & !vivo_atac_subset$esr2 ~ "N",
  TRUE ~ "unclassified"
))

# Look at Ovary  peak lists
table(vivo_atac_subset$ovary_dmrt1<-vivo_atac_subset %over% ovary8w_ctvdmrt1_dmrt1)
table(vivo_atac_subset$ovary_sox9<-vivo_atac_subset %over% ovary_ctvdmrt1_sox9)

vivo_atac_subset$ovary_peak<-case_when (
  vivo_atac_subset$ovary_dmrt1 & vivo_atac_subset$ovary_sox9 ~ "DS",
  vivo_atac_subset$ovary_dmrt1 & !vivo_atac_subset$ovary_sox9 ~ "D",
  !vivo_atac_subset$ovary_dmrt1 & vivo_atac_subset$ovary_sox9  ~ "S",
  !vivo_atac_subset$ovary_dmrt1 & !vivo_atac_subset$ovary_sox9 ~ "N",
  TRUE ~ "unclassified"
)
table(vivo_atac_subset[vivo_atac_subset$class=="Sertoli"]$ovary_peak)


# Regions of Interest
roi<-swl(c("chr6:119,522,263-119,531,826", # Wnt5b
           "chr12:76,130,814-76,134,590", #Esr2
            "chr12:76,116,825-76,119,045", #Esr2_3p
          "chr1:136,912,616-136,917,861", #Nr5a2
              "chr4:130,800,511-130,803,585",#Sdc3
          "chr13:100,481,504-100,625,768", #Ocln
          "chr15:98,890,811-98,899,565", #Dhh
          "chr2:148,832,886-148,834,489" #Cst9
      #    "chr13:64,063,297-64,065,858", #HSD17b3
          #  "chr15:43,134,055-43,138,423" #Rspo2
                  ))

table(vivo_atac_subset$peak)
vivo_atac_subset[!vivo_atac_subset$peak %in% c("D","DS","S","F","FE","E"),]$peak <- "N"

#Numbers for Text
table(vivo_atac_subset[vivo_atac_subset$class=="Sertoli"]$peak)
round(table(vivo_atac_subset[vivo_atac_subset$class=="Sertoli"]$peak)/sum(table(vivo_atac_subset[vivo_atac_subset$class=="Sertoli"]$peak)),2)

table(vivo_atac_subset[vivo_atac_subset$class=="Granulosa"]$peak)
round(table(vivo_atac_subset[vivo_atac_subset$class=="Granulosa"]$peak)/sum(table(vivo_atac_subset[vivo_atac_subset$class=="Granulosa"]$peak)),2)


#roi<-c("vivo_atac029963","vivo_atac075082","vivo_atac078850","vivo_atac036384","vivo_atac007173")
#roi<-c("vivo_atac024838","vivo_atac061641","vivo_atac064754","vivo_atac005988","vivo_atac030065","vivo_atac081920")
#poi<-roi[roi %in% names(vivo_atac_subset)]
(df<-vivo_atac_subset[vivo_atac_subset %over% roi])
#df$label<-c("Esr2","Cytip","B2m","Nr5a2","Hsd17b3","Sdc3")


#13      64074603        64075302

length(vivo_atac_subset)
#  chr12:75,884,786-76,384,786 chr1:136,590,477-137,092,477 chr2:148,505,257-149,005,257

#t(log2(f_atac_vivo[roi,]))

svglite::svglite(paste0("somatic_atac_volcano_",ts,".svg"),width=3.5,height=3)
mcols(vivo_atac_subset) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
  mutate(peak=factor(peak,levels=c("DS","D","S","FE","F","E","N"))) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=peak)) + geom_point(size=0.75) + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(3,6,4,2,8,7,1)])) +
  annotate("text",x=df$lfc,y=(-1*log10(df$padj)),label=df$genes)
dev.off()


  #optionally color by Gran vs Sert gene expression
mcols(vivo_atac_subset) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=dimorph)) + geom_point() + theme_bw()  + scale_color_manual(values=c("red","purple","gray","blue")) +
  annotate("text",x=df$lfc,y=(-1*log10(df$padj)),label=df$genes)


```

# P-values for differentially ATACable regions in Figure 1
```{r}
as.data.frame(vivo_atac_subset[vivo_atac_subset %over% swl("chr2:148,651,621-148,909,512") & vivo_atac_subset$class!="Constitutive"])

as.data.frame(vivo_atac_subset[vivo_atac_subset %over% swl("chr12:76,086,000-76,206,000") & vivo_atac_subset$class!="Constitutive"])
```


# Figure 1B - Distance to Sert,Gran Genes
```{r distance_to_dimorphic_genes,eval=T}
s<-distanceToNearest(vivo_atac_subset,sertoli_transcripts,ignore.strand=TRUE)
stopifnot(all.equal(queryHits(s),1:length(vivo_atac_subset)))  
vivo_atac_subset$sertoli_distance<- NA 
vivo_atac_subset$nearest_sertoli_symbol<-NA
vivo_atac_subset$nearest_sertoli_gene_id<-NA
vivo_atac_subset[queryHits(s)]$sertoli_distance<-mcols(s)$distance 
vivo_atac_subset[queryHits(s)]$nearest_sertoli_symbol<-sertoli_transcripts[subjectHits(s)]$symbol
vivo_atac_subset[queryHits(s)]$nearest_sertoli_gene_id<-sertoli_transcripts[subjectHits(s)]$gene_id

g<-distanceToNearest(vivo_atac_subset,c(granulosa_transcripts),ignore.strand=TRUE)
stopifnot(all.equal(queryHits(g),1:length(vivo_atac_subset)))  
vivo_atac_subset$granulosa_distance<- NA 
vivo_atac_subset$nearest_granulosa_symbol<-NA
vivo_atac_subset$nearest_granulosa_gene_id<-NA
vivo_atac_subset[queryHits(g)]$granulosa_distance<-mcols(g)$distance 
vivo_atac_subset[queryHits(g)]$nearest_granulosa_symbol<-granulosa_transcripts[subjectHits(g)]$symbol
vivo_atac_subset[queryHits(g)]$nearest_granulosa_gene_id<-granulosa_transcripts[subjectHits(g)]$gene_id

head(vivo_atac_subset)


#FIGURE 1C  - Distance to dimorphic Genetable
temp<-vivo_atac_subset
cutoff<-c(5000,100000)
table(temp$close_to<-case_when (
  temp$sertoli_distance <= cutoff[1] & temp$granulosa_distance > cutoff[1]~ "sert_short", 
  temp$sertoli_distance > cutoff[1] & temp$granulosa_distance <= cutoff[1] ~ "gran_short", 
  temp$sertoli_distance <= cutoff[1] & temp$granulosa_distance <= cutoff[1] ~ "both_short", 
  temp$sertoli_distance > cutoff[1] & temp$sertoli_distance <= cutoff[2] & temp$granulosa_distance > cutoff[2] ~ "sert_long", 
  temp$granulosa_distance > cutoff[1] & temp$granulosa_distance <= cutoff[2] & temp$sertoli_distance > cutoff[2] ~ "gran_long", 
  temp$sertoli_distance > cutoff[1] & temp$sertoli_distance <= cutoff[2] & temp$granulosa_distance > cutoff[1] & temp$granulosa_distance <= cutoff[2] ~ "both_long", 
  temp$sertoli_distance > cutoff[2] & temp$granulosa_distance > cutoff[2] ~ "both_too_long", 
  TRUE ~ "unclassified"
))

temp[temp$close_to=="both_short"]

#Numbers for paper
table(temp[temp$class=="Sertoli"]$close_to)
table(temp[temp$class=="Granulosa"]$close_to)

g1<- mcols(temp) %>% as.data.frame() %>% 
  dplyr::filter(close_to=="sert_short" | close_to=="gran_short") %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
  ggplot(aes(x=lfc,y=(-1*log10(padj)),color=close_to)) + geom_point(size=0.7) + 
  xlim(c(-6,6)) + ylim(c(0,60)) + theme_bw()  + theme(legend.position = "None") +
  xlab("") + ylab("") +
  scale_color_manual(values=c("red","blue")) 
 
g2<- mcols(temp) %>% as.data.frame() %>% 
  dplyr::filter(close_to=="sert_long" | close_to=="gran_long") %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
  ggplot(aes(x=lfc,y=(-1*log10(padj)),color=close_to)) + geom_point(size=0.7) +
  xlim(c(-6,6)) + ylim(c(0,60)) + theme_bw()  + theme(legend.position = "None") +
    xlab("") + ylab("") +
  scale_color_manual(values=c("red","blue")) 

svglite::svglite(paste0("somatic_atac_degs_",ts,".svg"),width=3,height=4)
grid.arrange(g2,g1,ncol=1)
dev.off()


#how many genes go up vs dn near Sertoli Specfic genes
length(temp_close_to_Sertoli_DARs<-temp[temp$class=="Sertoli" & (temp$close_to=="sert_short" | temp$close_to=="gran_short")])
length(unique(subset(temp_close_to_Sertoli_DARs,close_to=="gran_short")$nearest_granulosa_symbol))
length(unique(subset(temp_close_to_Sertoli_DARs,close_to=="sert_short")$nearest_sertoli_symbol))

length(temp_close_to_granulosa_DARs<-temp[temp$class=="Granulosa" & (temp$close_to=="sert_short" | temp$close_to=="gran_short")])
length(unique(subset(temp_close_to_granulosa_DARs,close_to=="gran_short")$nearest_granulosa_symbol))
length(unique(subset(temp_close_to_granulosa_DARs,close_to=="sert_short")$nearest_sertoli_symbol))

#how many dense clusters are there?
sertoli_DAR_density<-distanceToNearest(vivo_atac_subset[vivo_atac_subset$class=="Sertoli"],ignore.strand=TRUE)
mean(mcols(sertoli_DAR_density)$distance)
sum(seqlengths(mm10))/length(vivo_atac_subset[vivo_atac_subset$class=="Sertoli"])
hist(mcols(sertoli_DAR_density)$distance,n=10000,xlim=c(0,100000))

granulosa_DAR_density<-distanceToNearest(vivo_atac_subset[vivo_atac_subset$class=="Granulosa"],ignore.strand=TRUE)
mean(mcols(granulosa_DAR_density)$distance)
sum(seqlengths(mm10))/length(vivo_atac_subset[vivo_atac_subset$class=="Granulosa"])
hist(mcols(granulosa_DAR_density)$distance,n=10000,xlim=c(0,100000))

```



### If a DAR is within 1kp of a gene is it mostly likely to be a sex-specific gene?


### invert thinking - find all atac peaks near differentially expressed genes
```{r mean_distance,eval=F}

cutoff<-list(c(0,1000),c(1000,10000),c(10000,100000))

stratify_distance<-function(co,d) {table(vivo_atac_subset[queryHits(d)[mcols(d)$distance >= co[1] & mcols(d)$distance < co[2]]]$class)}
#stratify_distance(c(0,100000),s)

#find nearest peaks for Sertoli Specific Genes
s<-distanceToNearest(vivo_atac_subset,c(sertoli_transcripts),ignore.strand=TRUE)
lst.s<-lapply(cutoff,stratify_distance,s)
indx.s <- sapply(lst.s, length)
res.s <- as.data.frame(do.call(rbind,lapply(lst.s, `length<-`,
                          max(indx.s))))
colnames(res.s) <- names(lst.s[[which.max(indx.s)]])
res.s$window<-as.character(cutoff)
res.s
  g1<-res.s %>% gather(celltype,n,-window) %>% 
    ggplot( aes(x=window, y=n, fill=celltype)) + 
    scale_fill_manual(values=c("gray","magenta","blue")) +
    geom_bar(stat="identity",position="dodge") + coord_flip() + 
    theme_bw() + theme(legend.position = "none") + ylim(c(0,1500)) +
    ggtitle("Binned Distances to Sertoli Specific Transcripts")
 
#find nearest peaks for Granulsa Specific Genes
g<-distanceToNearest(vivo_atac_subset,c(granulosa_transcripts),ignore.strand=TRUE)
lst.g<-lapply(cutoff,stratify_distance,g)
indx.g <- sapply(lst.g, length)
 res.g <- as.data.frame(do.call(rbind,lapply(lst.g, `length<-`,
                          max(indx.g))))
colnames(res.g) <- names(lst.g[[which.max(indx.g)]])
res.g$window<-as.character(cutoff)
res.g

g2<-res.g %>% gather(celltype,n,-window) %>% 
  ggplot( aes(x=window, y=n, fill=celltype)) + 
  scale_fill_manual(values=c("gray","magenta","blue")) +
  geom_bar(stat="identity",position="dodge") + 
  coord_flip() + scale_y_reverse(limits=c(1500,0)) + 
  theme_bw() + ggtitle("Binned Distances to Granulosa Specific Transcripts")
 
#plot back-to-back  
legend <- get_legend(g2)


ggdraw(plot_grid(plot_grid(g2 + theme(legend.position = "none"), g1 + theme(legend.position = "none"), ncol=2, align='h'),
                 plot_grid(NULL, legend, ncol=1),
                 rel_widths=c(1, 0.2)))

```



### Determine what fraction of peaks are clustered
```{r cluster_fraction,eval=F}
length(temp_s<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli"])
#length(gr_cst<-vivo_atac_subset[vivo_atac_subset %over% swl("chr2:148,651,621-148,909,512") & vivo_atac_subset$class!="Constitutive"])
#length(gr_er<-vivo_atac_subset[vivo_atac_subset %over% swl("chr12:76,086,000-76,206,000") & vivo_atac_subset$class!="Constitutive"])
#temp_s$density<-NA
#system.time(for (i in seq_along(1:length(temp_s))) {temp_s[i]$density<-countOverlaps(peak_center(temp_s[i])+125000,temp_s)})

summary(temp_s$density)
sum(temp_s$density > 8)
as.data.frame(temp_s[temp_s$density > 8])
```



### Heatmap for Chromatin Only
```{r chromatin_only_heatmap,eval=F}
seqlevelsStyle(temp_2)<-"NCBI"
g_p1<-rtracklayer::import("../data/input_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p2<-rtracklayer::import("../data/input_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p3<-rtracklayer::import("../data/k4me1_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p4<-rtracklayer::import("../data/k4me1_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p5<-rtracklayer::import("../data/k27ac_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p6<-rtracklayer::import("../data/k27ac_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p7<-rtracklayer::import("../data/k4me3_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p8<-rtracklayer::import("../data/k4me3_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

g_p9<-rtracklayer::import("../data/k27me3_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p10<-rtracklayer::import("../data/k27me3_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

mat_p1 = normalizeToMatrix(g_p1, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p2 = normalizeToMatrix(g_p2, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p3 = normalizeToMatrix(g_p3, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p4 = normalizeToMatrix(g_p4, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p5 = normalizeToMatrix(g_p5, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p6 = normalizeToMatrix(g_p6, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p7 = normalizeToMatrix(g_p7, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p8 = normalizeToMatrix(g_p8, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p9 = normalizeToMatrix(g_p9, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p10 = normalizeToMatrix(g_p10, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)

col_fun_p1= colorRamp2(quantile(mat_p1, c(0, 0.99)), c("white", "black"))
col_fun_p2= colorRamp2(quantile(mat_p2, c(0, 0.99)), c("white", "black"))
col_fun_p3= colorRamp2(quantile(mat_p3, c(0, 0.99)), c("white", "magenta"))
col_fun_p4= colorRamp2(quantile(mat_p4, c(0, 0.99)), c("white", "magenta"))
col_fun_p5= colorRamp2(quantile(mat_p5, c(0, 0.99)), c("white", "blue"))
col_fun_p6= colorRamp2(quantile(mat_p6, c(0, 0.99)), c("white", "blue"))
col_fun_p7= colorRamp2(quantile(mat_p7, c(0, 0.99)), c("white", "green"))
col_fun_p8= colorRamp2(quantile(mat_p8, c(0, 0.99)), c("white", "green"))
col_fun_p9= colorRamp2(quantile(mat_p9, c(0, 0.99)), c("white", "red"))
col_fun_p10= colorRamp2(quantile(mat_p10, c(0, 0.99)), c("white", "red"))

col=cbPalette[c(3,5,7,4,6,8)]
#ggplot(data.frame(x=rep(10,length(col)),y=letters[1:length(col)]),aes(x=x,fill=y)) + geom_bar() + scale_fill_manual(values=col) + theme_bw()

EnrichedHeatmap(mat_p1, col = col_fun_p1, name = "input_noCre",column_title = "input noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p2, col = col_fun_p2, name = "input_dhhCre",column_title = "input dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p3, col = col_fun_p3, name = "k4me1_noCre",column_title = "k4me1 noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p4, col = col_fun_p4, name = "k4me1_dhhCre",column_title = "k4me1 dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,3),yaxis = F)) ) +
EnrichedHeatmap(mat_p5, col = col_fun_p5, name = "k27ac_noCre",column_title = "k27ac noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,7.5),yaxis = F)) ) +
EnrichedHeatmap(mat_p6, col = col_fun_p6, name = "k27ac_dhhCre",column_title = "k27ac dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,7.5),yaxis = F)) ) +
EnrichedHeatmap(mat_p7, col = col_fun_p7, name = "k4me3_noCre",column_title = "k4me3 noCre",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,12),yaxis = F)) ) +
EnrichedHeatmap(mat_p8, col = col_fun_p8, name = "k4me3_dhhCre",column_title = "k4me3_dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,12),yaxis = F)) ) +
EnrichedHeatmap(mat_p9, col = col_fun_p9, name = "k27me3_noCre",column_title = "k27me3 noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,2),yaxis = F)) ) +
EnrichedHeatmap(mat_p10, col = col_fun_p10, name = "k27me3_dhhCre",column_title = "k27me3_dhhCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,2),yaxis = F)) ) 

```

### Tn5 footprinting **** DELETE?
```{r tn5_footprinting,eval=FALSE}
bamfile<-"../data/sertoli_fresh_ATAC_rep1_N704_GW052319.dedup.unique.noMT.bam"
bamfile.labels <- gsub(".bam", "", basename(bamfile))
seqlevelsStyle(Mmusculus)<-"NCBI"
seqlevel<-c(1:19,"X","Y")
which<-as(seqinfo(Mmusculus)[seqlevel],"GRanges")
possibleTag <- combn(LETTERS, 2)
possibleTag <- c(paste0(possibleTag[1, ], possibleTag[2, ]),
                 paste0(possibleTag[2, ], possibleTag[1, ]))
bamTop100 <- scanBam(BamFile(bamfile, yieldSize = 10000),
                     param = ScanBamParam(tag=possibleTag))[[1]]$tag
(tags <- names(bamTop100)[lengths(bamTop100)>0])

gal <- readBamFile(bamfile, tag=tags, which=which, asMates=TRUE, bigFile=TRUE)

outPath <- "splited"
dir.create(outPath)
shiftedBamfile <- file.path(outPath, "shifted.bam")
gal1 <- shiftGAlignmentsList(gal, outbam=shiftedBamfile)

pt <- PTscore(gal1, transcripts)
plot(pt$log2meanCoverage, pt$PT_score, cex=0.5,pch=16,
     xlab="log2 mean coverage",
     ylab="Promoter vs Transcript")

nfr <- NFRscore(gal1, transcripts)
  plot(nfr$log2meanCoverage, nfr$NFR_score, 
     xlab="log2 mean coverage",cex=0.5,pch=16,
     ylab="Nucleosome Free Regions score",
     main="NFRscore for 200bp flanking TSSs",
     xlim=c(-10, 10), ylim=c(-8, 8))

  fa<-readDNAStringSet("murphy_2007_site_selection.fa")
fa<-subseq(fa,3,15)
dmrt1_pfm<-consensusMatrix(fa)[1:4,]
dmrt1_pfm<-t(t(dmrt1_pfm[1:4,])*1/colSums(dmrt1_pfm[1:4,]))
  
  sigs <- factorFootprints(gal1, pfm=dmrt1_pfm, 
                         genome=mm10,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)
  
```


