---
title: "Figure 2 - Motifs"
output: html_document
editor_options: 
  chunk_output_type: console
creative_commons: CC BY
---
# Identify Motifs under ChIP-seq and ATAC-seq In Vivo Peaks

## Export DNA sequences under ChIP Peaks for MEME   
```{r export_sequences_for_meme,eval=F}
nseq<-553 #Meme can process more than this but but changes its output format so that it won't be processed by parseMEMEOutput412

length(ovary_foxl2)
length(sertoli_sox9)
length(ovary_esr2)
length(sertoli_dmrt1)

length(ovary_esr2_top<-topify(ovary_esr2))
length(sertoli_dmrt1_top<-topify(sertoli_dmrt1))


ovary_foxl2.center<-GRanges(seqnames=seqnames(ovary_foxl2),IRanges(start=start(ovary_foxl2)+ovary_foxl2$summit,width=1),strand="*")+75
ovary_foxl2.dna<-getSeq(mm10,sample(ovary_foxl2.center,nseq))
names(ovary_foxl2.dna)<-paste0("ovary_foxl2_",1:length(ovary_foxl2.dna))
export(ovary_foxl2.dna,"ovary_foxl2_sampled_151bp.fasta")
length(readDNAStringSet("ovary_foxl2_sampled_151bp.fasta"))

ovary_esr2_top.center<-GRanges(seqnames=seqnames(ovary_esr2_top),IRanges(start=start(ovary_esr2_top)+ovary_esr2_top$summit,width=1),strand="*")+75
ovary_esr2_top.dna<-getSeq(mm10,sample(ovary_esr2_top.center,nseq))
names(ovary_esr2_top.dna)<-paste0("ovary_esr2_top_",1:length(ovary_esr2_top.dna))
export(ovary_esr2_top.dna,"ovary_esr2_top_sampled_151bp.fasta")
length(readDNAStringSet("ovary_esr2_top_sampled_151bp.fasta"))

sertoli_sox9.center<-GRanges(seqnames=seqnames(sertoli_sox9),IRanges(start=sertoli_sox9$summit,width=1),strand="*")+75
sertoli_sox9.dna<-getSeq(mm10,sample(sertoli_sox9.center,nseq))
names(sertoli_sox9.dna)<-paste0("sertoli_sox9_",1:length(sertoli_sox9.dna))
export(sertoli_sox9.dna,"sertoli_sox9_sampled_151bp.fasta")
length(readDNAStringSet("sertoli_sox9_sampled_151bp.fasta"))

sertoli_dmrt1_top.center<-GRanges(seqnames=seqnames(sertoli_dmrt1_top),IRanges(start=sertoli_dmrt1_top$summit,width=1),strand="*")+75
sertoli_dmrt1_top.dna<-getSeq(mm10,sample(sertoli_dmrt1_top.center,nseq))
names(sertoli_dmrt1_top.dna)<-paste0("sertoli_dmrt1_top_",1:length(sertoli_dmrt1_top.dna))
export(sertoli_dmrt1_top.dna,"sertoli_dmrt1_top_sampled_151bp.fasta")
length(readDNAStringSet("sertoli_dmrt1_top_sampled_151bp.fasta"))

```

## Import MEME analysis for ChIP Peaks
```{r import_meme,eval=T}


#foxl2
ovary_foxl2.meme<-TFBSTools:::parseMEMEOutput412("ovary_foxl2_sampled_151bp.txt")
x<-lengths(ovary_foxl2.meme_dna<-relist(getSeq(readDNAStringSet("ovary_foxl2_sampled_151bp.fasta"),unlist(ovary_foxl2.meme$motifList)),ovary_foxl2.meme$motifList))
names(ovary_foxl2.meme_dna)<-paste0("Motif ",1:length(ovary_foxl2.meme$motifList),", p-value: ",ovary_foxl2.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(ovary_foxl2.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Foxl2 ChIP Peaks")

ovary_foxl2.pfm<-consensusMatrix(reverseComplement(ovary_foxl2.meme_dna[[1]]))
ovary_foxl2.motif<-new("pfm",mat=t(t(ovary_foxl2.pfm[1:4,])*1/colSums(ovary_foxl2.pfm[1:4,])), name="Ovary FOXL2 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(foxl2_jaspar_motif,ovary_foxl2.motif),revcomp=c(F,F)))


#esr2
ovary_esr2.meme<-TFBSTools:::parseMEMEOutput412("ovary_esr2_top_sampled_151bp.txt")
x<-lengths(ovary_esr2.meme_dna<-relist(getSeq(readDNAStringSet("ovary_esr2_top_sampled_151bp.fasta"),unlist(ovary_esr2.meme$motifList)),ovary_esr2.meme$motifList))
names(ovary_esr2.meme_dna)<-paste0("Motif ",1:length(ovary_esr2.meme$motifList),", p-value: ",ovary_esr2.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(ovary_esr2.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("esr2 ChIP Peaks")

ovary_esr2.pfm<-consensusMatrix(reverseComplement(ovary_esr2.meme_dna[[1]]))
ovary_esr2.motif<-new("pfm",mat=t(t(ovary_esr2.pfm[1:4,])*1/colSums(ovary_esr2.pfm[1:4,])), name="Ovary esr2 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(esr2_jaspar_motif,ovary_esr2.motif),revcomp=c(T,T)))


#dmrt1
sertoli_dmrt1.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dmrt1_top_sampled_151bp.txt")
x<-lengths(sertoli_dmrt1.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dmrt1_top_sampled_151bp.fasta"),unlist(sertoli_dmrt1.meme$motifList)),sertoli_dmrt1.meme$motifList))
names(sertoli_dmrt1.meme_dna)<-paste0("Motif ",1:length(sertoli_dmrt1.meme$motifList),", p-value: ",sertoli_dmrt1.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dmrt1.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Dmrt1 ChIP Peaks")

sertoli_dmrt1.pfm<-consensusMatrix(reverseComplement(sertoli_dmrt1.meme_dna[[1]]))
sertoli_dmrt1.motif<-new("pfm",mat=t(t(sertoli_dmrt1.pfm[1:4,])*1/colSums(sertoli_dmrt1.pfm[1:4,])), name="Sertoli Dmrt1 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(dmrt1_invitro_motif,sertoli_dmrt1.motif),revcomp=c(F,F)))

#sox9
sertoli_sox9.meme<-TFBSTools:::parseMEMEOutput412("sertoli_sox9_sampled_151bp.txt")
x<-lengths(sertoli_sox9.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_sox9_sampled_151bp.fasta"),unlist(sertoli_sox9.meme$motifList)),sertoli_sox9.meme$motifList))
names(sertoli_sox9.meme_dna)<-paste0("Motif ",1:length(sertoli_sox9.meme$motifList),", p-value: ",sertoli_sox9.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_sox9.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 ChIP Peaks")

sertoli_sox9.pfm<-consensusMatrix(sertoli_sox9.meme_dna[[1]])
sertoli_sox9.motif<-new("pfm",mat=t(t(sertoli_sox9.pfm[1:4,])*1/colSums(sertoli_sox9.pfm[1:4,])), name="Sertoli Sox9 Motif")
plotMotifLogoStack(DNAmotifAlignment(list(sox9_jaspar_motif,sertoli_sox9.motif),revcomp=c(F,F)))


#show similarity of Dmrt1 and Sox9 Motifs
plotMotifLogoStack(DNAmotifAlignment(list(sox9_jaspar_motif,sertoli_sox9.motif,dmrt1_invitro_motif,sertoli_dmrt1.motif),revcomp=c(F,F,F,F)))

```


## Export DNA sequences under ATAC Peaks for MEME  
```{r export_DAR_sequences_for_meme,eval=F}

# Dmrt
length(sertoli_dar_Dmrt1<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$peak =="D"])
sertoli_dar_Dmrt1<-sertoli_dar_Dmrt1[with(sertoli_dar_Dmrt1,order(padj)),]
sertoli_dar_Dmrt1$top<-c(rep("top",1000),rep("not_top",length(sertoli_dar_Dmrt1)-1000))

mcols(sertoli_dar_Dmrt1) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=top)) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)]))

sertoli_dar_Dmrt1<-peak_center(sertoli_dar_Dmrt1[1:1000])
sertoli_dar_Dmrt1.dna<-getSeq(mm10,sertoli_dar_Dmrt1+100)
names(sertoli_dar_Dmrt1.dna)<-paste0("sertoli_dar_Dmrt1_",1:length(sertoli_dar_Dmrt1.dna))
export(sertoli_dar_Dmrt1.dna,"sertoli_dar_Dmrt1_201bp.fasta")

# Sox9
length(sertoli_dar_Sox9<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$peak =="S"])
sertoli_dar_Sox9<-sertoli_dar_Sox9[with(sertoli_dar_Sox9,order(padj)),]
#sertoli_dar_Sox9$top<-c(rep("top",1000),rep("not_top",length(sertoli_dar_Sox9)-1000))

mcols(sertoli_dar_Sox9) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)))) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)]))

sertoli_dar_Sox9<-peak_center(sertoli_dar_Sox9)
sertoli_dar_Sox9.dna<-getSeq(mm10,sertoli_dar_Sox9+100)
names(sertoli_dar_Sox9.dna)<-paste0("sertoli_dar_Sox9_",1:length(sertoli_dar_Sox9.dna))
export(sertoli_dar_Sox9.dna,"sertoli_dar_Sox9_201bp.fasta")

# Not Dmrt1/Sox9
length(sertoli_dar_wo_DS<-vivo_atac_subset[vivo_atac_subset$class=="Sertoli" & vivo_atac_subset$peak =="N"])
sertoli_dar_wo_DS<-sertoli_dar_wo_DS[with(sertoli_dar_wo_DS,order(padj)),]
sertoli_dar_wo_DS$top<-c(rep("top",1000),rep("not_top",length(sertoli_dar_wo_DS)-1000))

mcols(sertoli_dar_wo_DS) %>% as.data.frame() %>% 
  dplyr::filter(!is.na(lfc) & !is.na(padj)) %>% 
ggplot(aes(x=lfc,y=(-1*log10(padj)),color=top)) + geom_point() + theme_bw()  + 
  scale_color_manual(values=c(cbPalette[c(6,2,3,8,1,4)]))

sertoli_dar_wo_DS<-peak_center(sertoli_dar_wo_DS[1:1000])
sertoli_dar_wo_DS.dna<-getSeq(mm10,sertoli_dar_wo_DS+100)
names(sertoli_dar_wo_DS.dna)<-paste0("sertoli_dar_wo_DS_",1:length(sertoli_dar_wo_DS.dna))
export(sertoli_dar_wo_DS.dna,"sertoli_dar_wo_DS_201bp.fasta")

```

## Import MEME analysis for ATAC Peaks
```{r load_DAR_motif,eval=T}
sertoli_dar_Dmrt1.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dar_Dmrt1_201bp.txt")
x<-lengths(sertoli_dar_Dmrt1.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dar_Dmrt1_201bp.fasta"),unlist(sertoli_dar_Dmrt1.meme$motifList)),sertoli_dar_Dmrt1.meme$motifList))
names(sertoli_dar_Dmrt1.meme_dna)<-paste0("Motif ",1:length(sertoli_dar_Dmrt1.meme$motifList),", p-value: ",sertoli_dar_Dmrt1.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dar_Dmrt1.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Dmrt1 Figure ")

sertoli_dar_Sox9.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dar_Sox9_201bp.txt")
x<-lengths(sertoli_dar_Sox9.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dar_Sox9_201bp.fasta"),unlist(sertoli_dar_Sox9.meme$motifList)),sertoli_dar_Sox9.meme$motifList))
names(sertoli_dar_Sox9.meme_dna)<-paste0("Motif ",1:length(sertoli_dar_Sox9.meme$motifList),", p-value: ",sertoli_dar_Sox9.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dar_Sox9.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 Figure ")

sertoli_dar_wo_DS.meme<-TFBSTools:::parseMEMEOutput412("sertoli_dar_wo_DS_201bp.txt")
x<-lengths(sertoli_dar_wo_DS.meme_dna<-relist(getSeq(readDNAStringSet("sertoli_dar_wo_DS_201bp.fasta"),unlist(sertoli_dar_wo_DS.meme$motifList)),sertoli_dar_wo_DS.meme$motifList))
names(sertoli_dar_wo_DS.meme_dna)<-paste0("Motif ",1:length(sertoli_dar_wo_DS.meme$motifList),", p-value: ",sertoli_dar_wo_DS.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(sertoli_dar_wo_DS.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("wo Dmrt1/Sox9 binding")

sertoli_dar_wo_DS.pfm<-consensusMatrix(sertoli_dar_wo_DS.meme_dna[[3]])
sertoli_dar_wo_DS.motif<-new("pfm",mat=t(t(sertoli_dar_wo_DS.pfm[1:4,])*1/colSums(sertoli_dar_wo_DS.pfm[1:4,])), name="sertoli_dar_wo_DS")

plotMotifLogoStack(DNAmotifAlignment(list(nr5a2_jaspar_motif,sertoli_dar_wo_DS.motif),revcomp=c(F,F)))

```


## Scan for ChIP motifs Genome-Wide
```{r scan_chip_motifs,eval=F}

system.time(sertoli_dmrt1.sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=sertoli_dmrt1.pfm[1:4,],score="85%")))
sapply(sertoli_dmrt1.sites,length)
sertoli_dmrt1.sites<-sort(unlist(GRangesList(sertoli_dmrt1.sites)))

system.time(sertoli_sox9.sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=sertoli_sox9.pfm[1:4,],score="85%")))
sapply(sertoli_sox9.sites,length)
sertoli_sox9.sites<-sort(unlist(GRangesList(sertoli_sox9.sites)))

#Scan for NR5A1 motif
system.time(sertoli_dar_wo_DS.sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=sertoli_dar_wo_DS.pfm[1:4,],score="90%")))
sapply(sertoli_dar_wo_DS.sites,length)
sertoli_dar_wo_DS.sites<-sort(unlist(GRangesList(sertoli_dar_wo_DS.sites)))

save(sertoli_sox9.sites,sertoli_dmrt1.sites,sertoli_dar_wo_DS.sites,file="genome_wide_motifs.rdata")
```




