---
title: "Figure 4"
output: html_document
editor_options: 
  chunk_output_type: console
---


# Figure

## Panel A & B - In Vitro RNA-seq
```{r figure4,eval=F}

as.data.frame(colData(dds_invitro<-dds[,!str_detect(colData(dds)$time,"ovary|testis") & !str_detect(colData(dds)$batch,"GW") ]))



#colors<-c("#FFD200","#FFBF00","#FDAE00","#E69F00",cbPalette[c(8,4,4,4,4,6,5,5,5,5)])
colors<-c(c1a,c1b,c1c,c1d,cbPalette[8],c2a,c2b,c2c,c2d,cbPalette[6],c3a,c3b,c3c,c3d,"pink","blue")
plotPCA(normTransform(dds_invitro),intgroup=c("gene","time"),ntop = 500)+
  scale_color_manual(values = colors)+theme_bw()

#subset to same genes in Figure 1 PCA
dim(dds_invitro_subset<-dds_invitro[rownames(dds_invitro) %in% rownames(dds_gonad_subset),])

plotPCA(normTransform(dds_invitro_subset),intgroup=c("gene","time"),ntop = Inf )+
  scale_color_manual(values = colors)+theme_bw()


gg_plotCounts("Dmrt1",d=dds_invitro)
gg_plotCounts("Sox9",d=dds_invitro)

#gg_plotCounts("Cyp19a1",d=dds_invitro)
#gg_plotCounts("Foxl2",d=dds_invitro)
#gg_plotCounts("Gstm6",d=dds_invitro)
#gg_plotCounts("Esr2",d=dds_invitro)
#gg_plotCounts("Cidea",d=dds_invitro)

```

## Day 7 Gene expression Differences
```{r}
as.data.frame(colData(dds_7d<-dds_invitro[,str_detect(colData(dds_invitro)$time,"7d")]))
#as.data.frame(colData(dds_invitro)) #fresh samples included
design(dds_7d)<- ~ gene
dds_7d$gene <- droplevels(dds_7d$gene)
dds_7d<-DESeq(dds_7d)
resultsNames(dds_7d)

summary(res_7d_full_dmrt1<-results(dds_7d, contrast=c("gene","dmrt1","control"),alpha=0.05))
res_7d_full_dmrt1$symbol<-symbols[match(rownames(res_7d_full_dmrt1),symbols$gene_id),]$gene_name
res_7d_full_dmrt1$chr<-symbols[match(rownames(res_7d_full_dmrt1),symbols$gene_id),]$chr


summary(res_7d_full_sox9<-results(dds_7d, contrast=c("gene","sox9","control"),alpha=0.05))
res_7d_full_sox9$symbol<-symbols[match(rownames(res_7d_full_sox9),symbols$gene_id),]$gene_name
res_7d_full_sox9$chr<-symbols[match(rownames(res_7d_full_sox9),symbols$gene_id),]$chr
```

## Panel C - 7 day GSEA
```{r}
somatic_gene_list<-list(sertoli=sertoli_up, granulosa=granulosa_up)
names(somatic_gene_list)

nm<-lapply(seq_along(somatic_gene_list), function(i) c(names(somatic_gene_list)[[i]],"NA", somatic_gene_list[[i]]))

sink("somatic_gene_list.gmt")
writeLines(unlist(lapply(nm, paste, collapse="\t")))
sink()

temp<-as.data.frame(subset(res_7d_full_dmrt1,baseMean > 200 & padj < 0.05) )
temp$ensembl<-rownames(temp)
temp<-temp[!is.na(temp$stat),]
write.table(temp[,c("ensembl","stat")],file="res_7d_full_dmrt1.rnk",sep="\t",quote=F,col.names=F,row.names = F)

temp<-as.data.frame(subset(res_7d_full_sox9,baseMean > 200 & padj < 0.05) )
temp$ensembl<-rownames(temp)
temp<-temp[!is.na(temp$stat),]
write.table(temp[,c("ensembl","stat")],file="res_7d_full_sox9.rnk",sep="\t",quote=F,col.names=F,row.names = F)

```

## Panel D - Plppr4

```{r}
as.data.frame(colData(dds_7dko<-dds[,!str_detect(colData(dds)$time,"ovary|testis") & str_detect(colData(dds)$time,"7d") ]))

gg_plotCounts("Plppr4",d=dds_invitro)
gg_plotCounts7d("Plppr4",d=dds_7dko)

```


