---
title: "Figure 3 - Motifs and DARs in vivo"
output: html_document
editor_options: 
  chunk_output_type: console
creative_commons: CC BY
---

# Heatmap of ChIP and ATAC Peaks and Motifs  
```{r figure3_heatmap_setup,eval=T}
load("genome_wide_motifs.rdata")

cutoff<-1000

#add fpkm for both gehes
vivo_atac_subset$sertoli_gene_sert_fpkm<-f_soma_mean[vivo_atac_subset$nearest_sertoli_gene_id,]$sertoli
vivo_atac_subset$sertoli_gene_gran_fpkm<-f_soma_mean[vivo_atac_subset$nearest_sertoli_gene_id,]$granulosa
vivo_atac_subset$sertoli_gene_stat<-results_soma[vivo_atac_subset$nearest_sertoli_gene_id,"stat"]
vivo_atac_subset$gran_gene_sert_fpkm<-f_soma_mean[vivo_atac_subset$nearest_granulosa_gene_id,]$sertoli
vivo_atac_subset$gran_gene_gran_fpkm<-f_soma_mean[vivo_atac_subset$nearest_granulosa_gene_id,]$granulosa
vivo_atac_subset$gran_gene_stat<-results_soma[vivo_atac_subset$nearest_granulosa_gene_id,"stat"]

length(s_regions<-vivo_atac_subset[queryHits(s)[mcols(s)$distance < cutoff]])

s_regions$sert_fpkm<-s_regions$sertoli_gene_sert_fpkm
s_regions$gran_fpkm<-s_regions$sertoli_gene_gran_fpkm
s_regions$stat<-s_regions$sertoli_gene_stat

length(g_regions<-vivo_atac_subset[queryHits(g)[mcols(g)$distance < cutoff]])

g_regions$sert_fpkm<-g_regions$gran_gene_sert_fpkm
g_regions$gran_fpkm<-g_regions$gran_gene_gran_fpkm
g_regions$stat<-g_regions$gran_gene_stat

head(g_regions)

#View(as.data.frame(g_regions[g_regions %over% s_regions]))
sum(g_regions %over% s_regions)

g_regions$deg<-"deg_gran"
s_regions$deg<-"deg_sert"
t_regions<-c(g_regions,s_regions)
t_regions<-t_regions[!t_regions %in% g_regions[g_regions %over% s_regions]]

length(t_regions)
table(t_regions$deg)
#View(as.data.frame((t_regions)))
head(sort(table(s_regions$nearest_sertoli_symbol),decreasing = TRUE),20)
head(sort(table(g_regions$nearest_granulosa_symbol),decreasing = TRUE),20)

```

# Heatmap of ChIP and ATAC Peaks and Motifs  
```{r figure3_heatmap,eval=T}

temp_1<-t_regions
temp_1<-temp_1[temp_1$class!="Granulosa"]  #only constitutive and Sertoli
temp_1$class<-paste0(temp_1$class,":",temp_1$deg) %>% gsub("oli","",.) %>% gsub("deg_","",.) %>% gsub("Constitutive","Con",.) %>% gsub("ulosa","", .)
table(temp_1$class)
temp_2<-peak_center(temp_1) 

df<-data.frame(gran=temp_1$gran_fpkm,sert=temp_1$sert_fpkm,stat=temp_1$stat)

#g_atac_dmrt1<-rtracklayer::import("../data/gran_p7CAGDmrt1_n2_P037_ATACseq_N702.bigWig",selection=temp_2+2000)
#g_atac_control<-rtracklayer::import("../data/gran_p7control_n1_P037_ATACseq_N703.bigWig",selection=temp_2+2000)
  
g_atac_dmrt1<-rtracklayer::import("../data/fresh_sertoli_noMT_dedup_unique_merge.bigWig",selection=temp_2+2000) #sertoli
g_atac_control<-rtracklayer::import("../data/fresh_gran_noMT_dedup_unique_merge.bigWig",selection=temp_2+2000) #granulosa
#g_p2<-rtracklayer::import("../data/sox9_Index12_DSG_PFA_062315_TACAAG.dedup.unique.bigWig",selection=temp_2+2000)
g_p2<-rtracklayer::import("../data/sox9_testis_idx512_Tue_Jan_28_2020_1800.bigWig",selection=temp_2+2000)
#g_p3<-rtracklayer::import("../data/ovaryav574_CAGATC.dedup.unique.bigWig",selection=temp_2+2000)

g_p4<-rtracklayer::import("../data/k27ac_noCre.dedup.unique.bigWig",selection=temp_2+2000)
g_p5<-rtracklayer::import("../data/k27ac_dhhCre.dedup.unique.bigWig",selection=temp_2+2000)

seqlevelsStyle(g_atac_dmrt1)<-"UCSC"
seqlevelsStyle(g_atac_control)<-"UCSC"
seqlevelsStyle(g_p2)<-"UCSC"
#seqlevelsStyle(g_p3)<-"UCSC"
seqlevelsStyle(g_p4)<-"UCSC"
seqlevelsStyle(g_p5)<-"UCSC"

mat_dmrt1=normalizeToMatrix(sertoli_dmrt1.sites, temp_2, value_column = "score", mean_mode = "absolute",
   extend = 2000, w = 50, smooth = FALSE)
#mat_dmrt1_open=normalizeToMatrix(group_a_sites, temp_2, value_column = "score", mean_mode = "absolute",
#   extend = 2000, w = 50, smooth = FALSE)
mat_sox9=normalizeToMatrix(testis_sox9.sites, temp_2, value_column = "score", mean_mode = "absolute",
   extend = 2000, w = 50, smooth = FALSE)
#mat_sox9_open=normalizeToMatrix(group_c_sites, temp_2, value_column = "score", mean_mode = "absolute",
#   extend = 2000, w = 50, smooth = FALSE)
mat_tea=normalizeToMatrix(sertoli_dar_wo_DS.sites, temp_2, value_column = "score", mean_mode = "absolute",
   extend = 2000, w = 50, smooth = FALSE)
#mat_foxl2=normalizeToMatrix(foxl2_pwm, temp_2, value_column = "score", mean_mode = "absolute",
#   extend = 2000, w = 50, smooth = FALSE)


seqlevelsStyle(temp_2)<-"UCSC"
g_p1<-rtracklayer::import("../data/covaris_6c_sertoli_dmrt1ChIP.R1.fastq.dedup.unique.ucsc.bigWig",selection=temp_2+2000)



mat_p1 = normalizeToMatrix(g_p1, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p2 = normalizeToMatrix(g_p2, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
#mat_p3 = normalizeToMatrix(g_p3, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p4 = normalizeToMatrix(g_p4, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_p5 = normalizeToMatrix(g_p5, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_ad = normalizeToMatrix(g_atac_dmrt1, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)
mat_ac = normalizeToMatrix(g_atac_control, temp_2, value_column = "score",  extend = 2000, mean_mode = "w0", w = 50)




col_fun_p1= colorRamp2(quantile(mat_p1, c(0, 0.99)), c("white", "darkblue"))
col_fun_p2= colorRamp2(quantile(mat_p2, c(0, 0.99)), c("white", "green"))
#col_fun_p3= colorRamp2(quantile(mat_p3, c(0, 0.99)), c("white", "magenta"))
col_fun_p4= colorRamp2(quantile(mat_p4, c(0, 0.99)), c("white", "black"))
col_fun_p5= colorRamp2(quantile(mat_p5, c(0, 0.99)), c("white", "black"))
col_fun_ac= colorRamp2(quantile(mat_ac, c(0, 0.99)), c("white", "red"))
col_fun_ad= colorRamp2(quantile(mat_ad, c(0, 0.99)), c("white", "blue"))
col_fun_df=colorRamp2(c(min(c(df$gran,df$sert)),2.5,max(c(df$gran,df$sert))),c("#008837","#f7f7f7","#7b3294"))

#mysplit<-ifelse(temp$is.sig,"DeNovo","PreExisting")
#mysplit<-ifelse(g_regions$class,"DeNovo","PreExisting")

#col=c("red","green","blue")
#col=cbPalette[c(3,5,7,4,6,8)]
col=cbPalette[c(6,7,4,1)]
  dmrt1_order<-row_order(  EnrichedHeatmap(mat_p1, col = col_fun_p1, name = "dmrt1",column_title = "Dmrt1 (Sertoli)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                  top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,18),yaxis = F)) ))
    
#svglite::svglite(paste0("Figure3_invivo_heatmap_legends_",ts,".svg"),width=6,height=7.5)
pdf(paste0("Figure3_invivo_heatmap_legends_",ts,".pdf"),width=6,height=7.5)
Heatmap(df$gran,name="FPKM Gran",col=col_fun_df, column_title = "Granulosa Expr",show_heatmap_legend = TRUE,column_title_rot=90,split=temp_1$class,width=20,row_order = as.integer(unlist(dmrt1_order)),cluster_rows = FALSE,use_raster=TRUE, raster_device="png", raster_quality=10) + 
Heatmap(df$sert,name="FPKM Sert",col=col_fun_df, column_title = "Sertoli Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20,use_raster=TRUE, raster_device="png", raster_quality=10)  +
  
EnrichedHeatmap(mat_ac, col = col_fun_ac, name = "atac_control",column_title = "Granulosa",axis_name_rot = 90,split=temp_1$class,pos_line=F,column_title_rot=90,show_heatmap_legend = FALSE,
              top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,13),yaxis = F)),use_raster=TRUE, raster_device="png",raster_quality=10) +
EnrichedHeatmap(mat_ad, col = col_fun_ad, name = "atac_dmrt1",column_title = "Sertoli",axis_name_rot = 90,split=temp_1$class,pos_line=F,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,25),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10) +
EnrichedHeatmap(mat_p1, col = col_fun_p1, name = "dmrt1",column_title = "Dmrt1 (Sertoli)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,9),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) +
EnrichedHeatmap(mat_p2, col = col_fun_p2, name = "Sox9",column_title = "Sox9 (Sertoli)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0.02,0.055),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) +
#EnrichedHeatmap(mat_p3, col = col_fun_p3, name = "Foxl2",column_title = "Foxl2 (Ovary)",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,1.5),yaxis = F)) ) +
EnrichedHeatmap(mat_p4, col = col_fun_p4, name = "k27nc",column_title = "k27 noCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,10),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) +
EnrichedHeatmap(mat_p5, col = col_fun_p5, name = "k27dhh",column_title = "k27 dHHCRE",axis_name_rot = 90,pos_line=F,split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,8),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) +
EnrichedHeatmap(mat_dmrt1, col = c("white", "darkblue"),pos_line=F, name = "dmrt1_sites", axis_name_rot = 90,column_title = "Dmrt1 Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) +
EnrichedHeatmap(mat_sox9, col = c("white", "darkgreen"),pos_line=F, name = "sox9_sites", axis_name_rot = 90,column_title = "Sox9 Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) +
#EnrichedHeatmap(mat_sox9_open, col = c("white", "darkgreen"),pos_line=F, name = "group_c_sites", axis_name_rot = 90,column_title = "group c Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) +
EnrichedHeatmap(mat_tea, col = c("white", "darkblue"),pos_line=F, name = "tea_sites", axis_name_rot = 90,column_title = "TEA Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)),use_raster=TRUE, raster_device="png", raster_quality=10 ) 
dev.off()
    
#EnrichedHeatmap(mat_foxl2, col = c("white", "magenta"),pos_line=F, name = "foxl2_sites", axis_name_rot = 90,column_title = "Foxl2 Motifs",split=temp_1$class,column_title_rot=90,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = T)) ) 
  
  
  #Heatmap(df$gran,name="FPKM Gran",column_title = "Granulosa Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20) + 
  #Heatmap(df$sert,name="FPKM Sert",column_title = "Sertoli Expr",show_heatmap_legend = FALSE,column_title_rot=90,split=temp_1$class,width=20)


#  EnrichedHeatmap(mat_p3, col = col_fun_p3, name = "7d Sox9 - Sox9",column_title = "7d Sox9 - Sox9",axis_name_rot = 90,pos_line=F,split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE) +
#  EnrichedHeatmap(mat_p2, col = col_fun_p2, name = "7d Dmrt1 - Sox9",column_title = "7d Dmrt1 - Sox9",pos_line=F,axis_name_rot = 90,split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE) +
#  EnrichedHeatmap(mat_dmrt1, col = c("white", "black"),pos_line=F, name = "dmrt1_sites", axis_name_rot = 90,column_title = "dmrt1_sites",split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE) +
#  EnrichedHeatmap(mat_sox9, col = c("white", "black"),pos_line=F, name = "sox9_sites", axis_name_rot = 90,column_title = "sox9_sites",split=mysplit,column_title_rot=90,show_heatmap_legend = FALSE)
  
```

