---
title: "Figure 4 - CAG-Dmrt1 Ovary"
output: html_document
editor_options: 
  chunk_output_type: console
creative_commons: CC BY
---

# Figure 4A - Somatic Ovary PCA

## Create a DDS object of the whole gonad samples 
```{r gonad_pca,eval=T}
as.data.frame(colData(dds_gonad<-dds[,str_detect(colData(dds)$time,"ovary|testis") & !str_detect(colData(dds)$filename,"DMEf|WTf")]))
colData(dds_gonad)$sox8<-as.numeric(substr(colData(dds_gonad)$gene,2,2))
colData(dds_gonad)$sox9<-as.numeric(substr(colData(dds_gonad)$gene,3,3))
colData(dds_gonad)$sox<-colData(dds_gonad)$sox8+colData(dds_gonad)$sox9
colData(dds_gonad)$gene<-as.factor(sapply(str_split(colData(dds_gonad)$gene,"_"), function(x) x[2]))
colData(dds_gonad)$batch<-as.factor(colData(dds_gonad)$batch)
as.data.frame(colData(dds_gonad))
plotPCA(normTransform(dds_gonad),intgroup=c("userid","time","gene","sox")) + 
  theme_bw() + ggtitle("Gonad Data - Full Gene Set ")

```

### Make Table of FPKMs averaged by condition
```{r fpkm_gonad,eval=T}
#get fpkm
dim(f_gonad<-as.data.frame(log2(fpkm(dds_gonad)+0.1)))

#colnames(f_gonad)<-paste(colData(dds_gonad)$time,colData(dds_gonad)$gene,colData(dds_gonad)$sox,1:nrow(colData(dds_gonad)),sep="_")
#f_gonad$mgi_symbol<-mcols(dds_gonad)$mgi_symbol
#View(f_gonad)

f_gonad_mean <- f_gonad %>% 
#  dplyr::filter(mgi_symbol %in% c(goi,goi2)) %>% 
  tibble::rownames_to_column(var="ensembl") %>%
  tidyr::gather(sample,fpkm,-ensembl) %>%
  dplyr::mutate(time=colData(dds_gonad)[sample,]$time) %>%
  dplyr::mutate(gene=colData(dds_gonad)[sample,]$gene) %>%
  dplyr::mutate(sox=colData(dds_gonad)[sample,]$sox) %>%
  dplyr::mutate(sample=paste(time,gene,sox,sep="_")) %>% 
  dplyr::group_by(sample,ensembl) %>% 
    dplyr::summarize(mean_fpkm=round(mean(fpkm),3)) %>% 
    dplyr::select(sample,ensembl,mean_fpkm) %>% 
    ungroup() %>% 
  tidyr::spread(sample,mean_fpkm) %>% as.data.frame()

  rownames(f_gonad_mean) <- f_gonad_mean$ensembl
  f_gonad_mean<-f_gonad_mean[,-1]

colnames(f_gonad_mean)<-c("Ovary Sox Null","Ovary WT","Cag-Dmrt1 Sox Null","Cag-Dmrt1 Sox=1","Cag-Dmrt1 Sox=2","Cag-Dmrt1 Sox=4", "Testis WT")
f_gonad_mean<-f_gonad_mean[,c("Ovary WT","Ovary Sox Null","Cag-Dmrt1 Sox Null","Cag-Dmrt1 Sox=1","Cag-Dmrt1 Sox=2","Cag-Dmrt1 Sox=4", "Testis WT")]

#Export version without Log2 for Table
f_gonad_mean_symbol <- as.data.frame(fpkm(dds_gonad[,colData(dds_gonad)$batch!="P006"])) %>% 
#  dplyr::filter(mgi_symbol %in% c(goi,goi2)) %>% 
  tibble::rownames_to_column(var="ensembl") %>%
  tidyr::gather(sample,fpkm,-ensembl) %>%
  dplyr::mutate(sox8=colData(dds_gonad)[sample,]$sox8) %>%
  dplyr::mutate(sox9=colData(dds_gonad)[sample,]$sox9) %>%
  dplyr::mutate(gene=colData(dds_gonad)[sample,]$gene) %>%
  dplyr::mutate(sample=paste(gene,sox8,sox9,sep="_")) %>% 
  dplyr::group_by(sample,ensembl) %>% 
    dplyr::summarize(mean_fpkm=round(mean(fpkm),3)) %>% 
    dplyr::select(sample,ensembl,mean_fpkm) %>% 
    ungroup() %>% 
  tidyr::spread(sample,mean_fpkm) %>% as.data.frame()


f_gonad_mean_symbol<-f_gonad_mean_symbol[apply(f_gonad_mean_symbol[,2:9],1,sum) > 0,]
f_gonad_mean_symbol$Symbol<-symbols[match(f_gonad_mean_symbol$ensembl,symbols$gene_id),]$gene_name
f_gonad_mean_symbol<-dplyr::rename(f_gonad_mean_symbol,
                                   "Ensembl GeneID" = "ensembl",
                                   "Wild Type Ovary" = "control_2_2",
                                   "CAG-Dmrt1 Ovary" = "dmrt1_2_2",
                                   "Sox8-/-;Sox9-/- Ovary" = "control_0_0",
                                   "Sox8-/-;Sox9-/-;CAG-Dmrt1 Ovary" = "dmrt1_0_0",
                                   "Sox8-/-;Sox9+/-;CAG-Dmrt1 Ovary" = "dmrt1_0_1",
                                   "Sox8+/-;Sox9-/-;CAG-Dmrt1 Ovary" = "dmrt1_1_0",
                                   "Sox8+/-;Sox9+/-;CAG-Dmrt1 Ovary" = "dmrt1_1_1",
                                   "Sox8+/+;Sox9-/-;CAG-Dmrt1 Ovary" = "dmrt1_2_0") %>%
  dplyr::select("Ensembl GeneID","Symbol","Wild Type Ovary","Sox8-/-;Sox9-/- Ovary","CAG-Dmrt1 Ovary","Sox8-/-;Sox9-/-;CAG-Dmrt1 Ovary","Sox8-/-;Sox9+/-;CAG-Dmrt1 Ovary","Sox8+/-;Sox9-/-;CAG-Dmrt1 Ovary", "Sox8+/-;Sox9+/-;CAG-Dmrt1 Ovary","Sox8+/+;Sox9-/-;CAG-Dmrt1 Ovary") 

#View(f_gonad_mean_symbol)
write.csv(f_gonad_mean_symbol,"Lindeman_et_al_Whole_Gonad_Mean_FPKMs.csv",row.names=F,quote=F)
dim(f_gonad)
#sum(high_fpkm<-apply(f_gonad_mean,1,max) > log2(2.5))
length(high_gonad_fpkm<-rownames(f_gonad_mean[apply(f_gonad_mean,1,max) > log2(2.5),]))
table(symbols[symbols$gene_id %in% high_gonad_fpkm,"chr"])
```


## Subset to soma DE & fpkm_high & autosomes
```{r gonad_subset_pca,eval=T}

# get autosomal genes
#length(autosomal<-mgi[mgi$chromosome_name %in% as.character(1:19),]$ensembl_gene_id)

#subset to ~soma
dim(dds_gonad)
dim(dds_gonad_subset<-dds_gonad[(rownames(dds_gonad) %in% rownames(results_soma)),])

#length(autosomal<-symbols[symbols$chr %in% as.character(c(1:19)),]$gene_id)
#dim(dds_gonad_subset<-dds_gonad[(rownames(dds_gonad) %in% rownames(results_soma)) & 
#                                 (rownames(dds_gonad) %in% high_gonad_fpkm) & 
#                                  (rownames(dds_gonad) %in% autosomal) &
#                                  (rownames(dds_gonad) %in% high_soma_fpkm),])

#as.data.frame(colData(dds_gonad_subset))

colors_RdBu<-rev(colorRampPalette(brewer.pal(11,"RdBu"))(10))
#plot(1:10,pch=16,col=colors,cex=4)


#plotPCA(normTransform(dds_gonad_subset),intgroup=c("time","gene","sox8","sox9"),ntop=Inf,returnData = F)

#Use rlog for PCA
#dds_gonad_subset_rlog<-rlog(dds_gonad_subset,blind=T)

figure_4a<-plotPCA(normTransform(dds_gonad_subset),intgroup=c("time","gene","sox8","sox9"),ntop=Inf,returnData = T) %>% 
  mutate(group=factor(group,levels=c("ovary:control:2:2","ovary:control:0:0","ovary:dmrt1:0:0","ovary:dmrt1:0:1",
                                     "ovary:dmrt1:1:0","ovary:dmrt1:1:1","ovary:dmrt1:2:0",
                                     "ovary:dmrt1:2:2","testis:control:2:2"))) %>% 
  ggplot(aes(x=PC1,y=PC2,fill=group)) + geom_point(size=10,shape=21) +
  #xlim(c(-50,60)) + ylim(c(-60,30)) +
  theme_bw() +  ggtitle("Gonad Data - Somatic Subset") +
  geom_text(aes(label=ifelse( (sox8+sox9 < 4 | sox8+sox9 > 0) ,as.character(paste0(sox8,sox9)),''),x=PC1,y=PC2),size=6,color="black") +
   scale_fill_manual(values=c("magenta",colors_RdBu[c(10,9,8,7,4,3,2)],"blue")) 
  

figure_4a

#Paper Image
svglite::svglite(paste0("somatic_gonad_PCA_",ts,".svg"),width=6,height=4.5)
figure_4a
dev.off()
```


## Eigenvectors for PC2
```{r eigenvectors_for_pca,eval=T}
#nt<-log2(counts(dds_gonad_subset,normalized=TRUE)+1)
#colnames(nt)<-paste(colData(dds_gonad_subset)$time,colData(dds_gonad_subset)$sox,colData(dds_gonad_subset)$gene,sep="_")
#nt.pca<-prcomp(t(nt))
dds_gonad_subset.pca<-plotPCA.DESeqTransform(normTransform(dds_gonad_subset),intgroup=c("userid","time","gene","sox"),returnData=T,ntop=Inf)
#plot(dds_gonad_subset_rlog.pca[,c(1,2)],col=ifelse(stringr::str_detect(rownames(dds_gonad_subset_rlog.pca),"WTm"),"red","black"),pch=16)
topN <- 20

nt.rot<-attr(dds_gonad_subset.pca,"rotation")
#pc1
nt.rot1<-as.data.frame(nt.rot[order(abs(nt.rot[,1]),decreasing=TRUE),])[,1,drop=F]
head(left_join(tibble::rownames_to_column(nt.rot1,var="gene_id"),symbols,by="gene_id"),topN) 

#pc2
nt.rot2<-as.data.frame(nt.rot[order(abs(nt.rot[,2]),decreasing=TRUE),])[,2,drop=F]
head(left_join(tibble::rownames_to_column(nt.rot2,var="gene_id"),symbols,by="gene_id"),topN)

#pc3
nt.rot3<-as.data.frame(nt.rot[order(abs(nt.rot[,3]),decreasing=TRUE),])[,3,drop=F]
head(left_join(tibble::rownames_to_column(nt.rot3,var="gene_id"),symbols,by="gene_id"),topN)


#nt.rot <- nt.pca$rotation
#pc1
#symbols[symbols$gene_id %in% names(nt.rot[,1][order(abs(nt.rot[,1]),decreasing=TRUE)][1:topN]),]
#pc2
#symbols[symbols$gene_id %in% names(nt.rot[,2][order(abs(nt.rot[,2]),decreasing=TRUE)][1:topN]),]

```


## Figure 4B - Foxl2 and Defb36 timecourse
```{r foxl2_defb32_timecourse,eval=T}

gg_sox_fpkm("Defb36",d=dds_gonad_subset,bar=T)+theme(axis.text.x=element_blank())+ylab("")
svglite::svglite(paste0("gonad_Defb36_expression_",ts,".svg"),width=5,height=3)
gg_sox_fpkm("Defb36",d=dds_gonad_subset,bar=T)+theme(axis.text.x=element_blank())+ylab("")
dev.off()

gg_sox_fpkm("Foxl2",d=dds_gonad_subset,bar=T)+theme(axis.text.x=element_blank())+ylab("")

svglite::svglite(paste0("gonad_Foxl2_expression_",ts,".svg"),width=5,height=3)
gg_sox_fpkm("Foxl2",d=dds_gonad_subset,bar=T)+theme(axis.text.x=element_blank())+ylab("")
dev.off()

gg_sox_fpkm("Dmrt1",d=dds_gonad_subset,bar=T)+theme(axis.text.x=element_blank())+ylab("")

svglite::svglite(paste0("gonad_Dmrt1_expression_",ts,".svg"),width=5,height=3)
gg_sox_fpkm("Dmrt1",d=dds_gonad_subset,bar=T)+theme(axis.text.x=element_blank())+ylab("")
dev.off()

nrow(temp<-f_gonad_mean[rownames(dds_gonad_subset),])




#Look at PC3
plotPCA(normTransform(dds_gonad_subset),intgroup=c("userid","time","gene","sox"),ntop=Inf) + 
  theme_bw() +  ggtitle("Gonad Data - Somatic Subset")

g1r<-plotPCA.DESeqTransform(normTransform(dds_gonad_subset),intgroup=c("userid","time","gene","sox"),ntop=Inf,dims=c(1,2)) + theme_bw() + theme(legend.position = "none")
g2r<-plotPCA.DESeqTransform(normTransform(dds_gonad_subset),intgroup=c("userid","time","gene","sox"),ntop=Inf,dims=c(3,1)) + theme_bw() + theme(legend.position = "none")
g3r<-plotPCA.DESeqTransform(normTransform(dds_gonad_subset),intgroup=c("userid","time","gene","sox"),ntop=Inf,dims=c(3,2)) + theme_bw() + theme(legend.position = "none")
g4r<-plotPCA.DESeqTransform(normTransform(dds_gonad_subset),intgroup=c("userid","time","gene","sox"),ntop=Inf,dims=c(2,3)) + theme_bw() 
grid.arrange(g1r,g3r,g_legend(g4r),g2r, ncol=2,top="Gonad Data - Somatic Subset")

#pheatmap(sweep(temp,1,apply(temp, 1, mean),"-"),cluster_cols=F,show_rownames = F,scale="none",col=colors)

```


## Figure 4C - Venn Diagram
```{r figure_4C_venn,eval=T}

# Compare Sox89 wt vs null as a categorial variable and look for an interaction term
colData(dds_ovary_cat<-dds_gonad[,colData(dds_gonad)$time=="ovary" & (colData(dds_gonad)$sox==0 | colData(dds_gonad)$sox==4)])
colData(dds_ovary_cat)$sox<-factor(ifelse(colData(dds_ovary_cat)$sox==4,"sox89wt","sox89null"),levels=c("sox89wt","sox89null"))
design(dds_ovary_cat)<- ~ gene + sox + gene:sox
as.data.frame(colData(dds_ovary_cat))


#as.data.frame(colData(dds_ovary))[,c(1,3:9)]
as.data.frame(colData(dds_ovary_wt<-dds_ovary_cat[,str_detect(colData(dds_ovary_cat)$sox,"sox89wt")]))
design(dds_ovary_wt)<-~gene
dds_ovary_wt<-estimateSizeFactors(dds_ovary_wt)
dds_ovary_wt<-DESeq(dds_ovary_wt)
summary(res_dds_ovary_wt<-results(dds_ovary_wt,alpha=0.05))

as.data.frame(colData(dds_ovary_null<-dds_ovary_cat[,str_detect(colData(dds_ovary_cat)$sox,"sox89null")]))
design(dds_ovary_null)<-~gene
dds_ovary_null<-estimateSizeFactors(dds_ovary_null)
dds_ovary_null<-DESeq(dds_ovary_null)
summary(res_dds_ovary_null<-results(dds_ovary_null,alpha=0.05))


length(x1<-rownames(subset(res_dds_ovary_wt, abs(log2FoldChange) > 1 & padj< 0.05 & baseMean > 50)))
length(y1<-rownames(subset(res_dds_ovary_null, abs(log2FoldChange) > 1 & padj< 0.05 & baseMean > 50)))
length(z1<- x1[x1 %in% y1])
#grid.newpage()
#vennplot <- draw.pairwise.venn(length(x1),length(y1),length(z1), c("Sox8/9 WT", "Sox8/9 Null"))

VennDiag <- euler(c("Sox8/9 WT" = length(x1)-length(z1), "Sox8/9 Null" = length(y1)-length(z1),
                    "Sox8/9 WT&Sox8/9 Null" = length(z1)))
plot(VennDiag, quantities = TRUE, font=12, cex=10, alpha=0.5,
     fill=c("grey", "lightgrey", "darkgrey"))


#Export for Supplementary data
temp1<-left_join(rownames_to_column(as.data.frame(res_dds_ovary_wt[x1,]),var="gene_id"),
                rownames_to_column(f_gonad_mean[,c("Ovary WT","Cag-Dmrt1 Sox=4")],var="gene_id")) %>% 
  left_join(symbols[,c("gene_id","gene_name")]) %>% 
  arrange(padj) %>% 
  dplyr::rename( "Ensembl GeneID" = "gene_id",
                 "Gene Symbol" = "gene_name",
                 "Mean Counts" = "baseMean",
                 "Log2FoldChange" = "log2FoldChange",
                 "BH padj" = "padj",
                 "FPKM WT Ovary" = "Ovary WT",
                 "FPKM CAG-Dmrt1 XX Gonad" = "Cag-Dmrt1 Sox=4") %>%
  dplyr::select("Ensembl GeneID","Gene Symbol","Mean Counts","Log2FoldChange","BH padj","FPKM WT Ovary","FPKM CAG-Dmrt1 XX Gonad") 

dim(temp1)
write.csv(temp1,"Lindeman_et_al_SoxWT_CagDmrt1_XXGonad.csv",row.names=F,quote=F)

temp2<-left_join(rownames_to_column(as.data.frame(res_dds_ovary_null[y1,]),var="gene_id"),
                rownames_to_column(f_gonad_mean[,c("Ovary Sox Null","Cag-Dmrt1 Sox Null")],var="gene_id")) %>% 
  left_join(symbols[,c("gene_id","gene_name")]) %>% 
  arrange(padj) %>% 
  dplyr::rename( "Ensembl GeneID" = "gene_id",
                 "Gene Symbol" = "gene_name",
                 "Mean Counts" = "baseMean",
                 "Log2FoldChange" = "log2FoldChange",
                 "BH padj" = "padj",
                 "FPKM Sox8/9 Null Ovary" = "Ovary Sox Null",
                 "FPKM CAG-Dmrt1;Sox8/9 Null XX Gonad" = "Cag-Dmrt1 Sox Null") %>%
  dplyr::select("Ensembl GeneID","Gene Symbol","Mean Counts","Log2FoldChange","BH padj","FPKM Sox8/9 Null Ovary","FPKM CAG-Dmrt1;Sox8/9 Null XX Gonad") 
                  
dim(temp2)
write.csv(temp2,"Lindeman_et_al_SoxNull_CagDmrt1_XXGonad.csv",row.names=F,quote=F)


```

# Check for compensating SOX proteins in absence of 8/9
```{r}
as.data.frame(colData(dds_ovary_comp<-dds_ovary_cat[,str_detect(colData(dds_ovary_cat)$gene,"dmrt1")]))
design(dds_ovary_comp)<-~sox
dds_ovary_comp<-estimateSizeFactors(dds_ovary_comp)
dds_ovary_comp<-DESeq(dds_ovary_comp)
summary(res_dds_ovary_comp<-results(dds_ovary_comp,alpha=0.05))
res_dds_ovary_comp_df<-left_join(rownames_to_column(as.data.frame(res_dds_ovary_comp),var="gene_id"),symbols) %>% 
  arrange(padj)
res_dds_ovary_comp_df[str_which(res_dds_ovary_comp_df$gene_name,"Sox"),]
f_gonad_mean_symbol[f_gonad_mean_symbol$`Ensembl GeneID`=="ENSMUSG00000041287",]
```



## Figure 4D - Sex Determination Heatmap
### Define list of sex determination genes based on embryonic microarray data and literature/ontology sources
```{r figure4D_sexdetermination_genes,eval=T}
length(embryonic_network_full <- tT[tT$adj.P.Val < 0.05 & tT$chr %in% 1:19,"ensembl"])

length(en_list<-symbols[symbols$gene_name %in% read_table("../granulosa_culture/network.txt",col_names = F)$X1,"gene_id"])

length(go_list<-symbols[symbols$gene_name %in% read_table("../granulosa_culture/go_sex_determination.txt",col_names = F)$X1,"gene_id"])
#vb_list <-stringr::str_to_title(read_table("../granulosa_culture/vivian_list.txt",col_names = F)$X1)
length(vb_list<-symbols[symbols$gene_name %in% stringr::str_to_title(read_table("../granulosa_culture/vivian_list.txt",col_names = F)$X1),"gene_id"])

length(sex_det_list<-unique(c(embryonic_network_full,en_list,go_list,vb_list)))




```


## Figure 4D - Heatmap of genes that are Sox8/9 dependent that are implicated in sex determination
```{r heatmap_embryonic_genes,eval=T}
length(x1<-rownames(subset(res_dds_ovary_wt, abs(log2FoldChange) > 1 & padj< 0.05 & baseMean > 50)))
length(y1<-rownames(subset(res_dds_ovary_null, abs(log2FoldChange) > 1 & padj< 0.05 & baseMean > 50)))
length(z1<- x1[x1 %in% y1])


length(x1sd<-x1[x1 %in% sex_det_list & !(x1 %in% z1)])
length(y1sd<-y1[y1 %in% sex_det_list & !(y1 %in% z1)])
length(z1sd<-z1[z1 %in% sex_det_list & !(z1 %in% y1sd) & !(z1 %in% x1sd)])

temp<-c(x1sd,y1sd,z1sd)
temp<-f_gonad_mean[temp,]
temp<-temp[,c("Ovary WT","Cag-Dmrt1 Sox=4","Ovary Sox Null","Cag-Dmrt1 Sox Null")]
nrow(temp<-temp[(apply(temp,1,function(x) max(x) > 5)),])  # remove low expressed genes
temp2<-sweep(temp,1,apply(temp, 1, mean),"-")
summary(apply(temp2,1,mean))

table(temp2$class <- case_when(
  rownames(temp2) %in% x1sd ~ "Sox89 WT Only",
  rownames(temp2) %in% y1sd ~ "Sox89 Null Only",
  rownames(temp2) %in% z1sd ~ "Sox89 Independent",
  TRUE ~ "blah"
))


temp2$class<-factor(temp2$class, levels=c("Sox89 WT Only","Sox89 Null Only","Sox89 Independent"))
rownames(temp2)<-symbols[match(rownames(temp2),symbols$gene_id),]$gene_name
#rownames(temp)<-symbols[match(rownames(temp),symbols$gene_id),]$gene_name

col_fun = circlize::colorRamp2(c(min(temp2[,1:4]),0, max(temp2[,1:4])), c("blue","white", "red"))

H1<-Heatmap(as.matrix(temp2[,1:4]), 
        column_title = "Sox Dependent Sex Determination Genes", rect_gp = gpar(col = "black", lwd = 0.5),
        cluster_rows = TRUE, cluster_columns=FALSE, split = temp2$class,  
        row_dend_reorder=temp2$`Cag-Dmrt1 Sox=4`,
        row_names_gp = gpar(fontsize = 8), row_title_gp = gpar(fontsize = 10),
        row_names_side = "left",col = col_fun, show_heatmap_legend = TRUE)



svglite::svglite(paste0("sox_dependent_sex_determination_",ts,".svg"),width=3,height=7)
H1
dev.off()

#View(temp2)

```


# Look at ChIP Data for all
```{r}

length(wt_only<-x1[!(x1 %in% z1)])
length(null_only<-y1[!(y1 %in% z1)])
length(independent<-z1[!(z1 %in% wt_only) & !(z1 %in% null_only)])

#sanity check
symbols[grep("Foxl2$",symbols$gene_name),"gene_id"] %in% wt_only
symbols[grep("Foxl2$",symbols$gene_name),"gene_id"] %in% independent
symbols[grep("Esr2$",symbols$gene_name),"gene_id"] %in% wt_only
symbols[grep("Esr2$",symbols$gene_name),"gene_id"] %in% independent
symbols[grep("Cyp17a1$",symbols$gene_name),"gene_id"] %in% null_only

transcripts$size<-width(transcripts)
transcripts[grep("Cyp19a1",transcripts$symbol)]
transcripts<-transcripts[with(transcripts,order(-size))]
length(transcripts_dedup<-transcripts[!duplicated(transcripts$gene_id)])
transcripts_dedup[grep("Cyp19a1",transcripts_dedup$symbol)]
strand(transcripts_dedup)<-"*"
length(transcripts_dedup<-transcripts_dedup[transcripts_dedup$gene_id %in% unique(c(x1,y1))])

length(ovary8w_ctvdmrt1_dmrt1)
table(transcripts_dedup$dmrt1_overlap <- (transcripts_dedup + 10000 ) %over% ovary8w_ctvdmrt1_dmrt1)  
table(transcripts_dedup$sox9_overlap <- (transcripts_dedup + 10000 ) %over% ovary_ctvdmrt1_sox9) 

(a<-table(transcripts_dedup[transcripts_dedup$gene_id %in% wt_only]$dmrt1_overlap)) #1652 sox8/9 dependent
(b<-table(transcripts_dedup[transcripts_dedup$gene_id %in% null_only]$dmrt1_overlap) ) #1343 only in null
(c<-table(transcripts_dedup[transcripts_dedup$gene_id %in% independent]$dmrt1_overlap) ) #1005 sox8/9 independent
matrix(c(a,c),nrow=2)

chisq.test(matrix(c(a,c),nrow=2))

(d<-table(transcripts_dedup[transcripts_dedup$gene_id %in% wt_only]$sox9_overlap)) #1652 sox8/9 dependent
(e<-table(transcripts_dedup[transcripts_dedup$gene_id %in% null_only]$sox9_overlap)) #1343 only in null
(f<-table(transcripts_dedup[transcripts_dedup$gene_id %in% independent]$sox9_overlap)) #1005 sox8/9 independent
matrix(c(d,f),nrow=2)
chisq.test(matrix(c(d,f),nrow=2))

```




## Figure 4E - In Vivo ATAC/ChIP/Motifs
### ChIP peaklist
```{r Figure_4E_invivo_motifs,eval=T}

length(ovary8w_ctvdmrt1_dmrt1)

(qs<-quantile(ovary8w_ctvdmrt1_dmrt1$score, c(0, 0.25, 0.5, 0.75, 0.99)))
plot(sort(ovary8w_ctvdmrt1_dmrt1$score,decreasing=T),pch=16,cex=0.5,col="red")
abline(h=qs,col="black")

length(ovary8w_ctvdmrt1_dmrt1_top<-ovary8w_ctvdmrt1_dmrt1[ovary8w_ctvdmrt1_dmrt1$score > qs[3] & ovary8w_ctvdmrt1_dmrt1$score < qs[5]])

length(ovary_ctvdmrt1_sox9) # also 8w
(qs<-quantile(ovary_ctvdmrt1_sox9$score, c(0, 0.25, 0.5, 0.75, 0.99)))
plot(sort(ovary_ctvdmrt1_sox9$score,decreasing=T),pch=16,cex=0.5,col="blue")
abline(h=qs,col="black")
length(ovary_ctvdmrt1_sox9_top<-ovary_ctvdmrt1_sox9[ovary_ctvdmrt1_sox9$score > qs[3] & ovary_ctvdmrt1_sox9$score < qs[5]])



length(temp_dmrt1<-keepSeqlevels(ovary8w_ctvdmrt1_dmrt1_top[!ovary8w_ctvdmrt1_dmrt1_top %over% el_mm10],value=c(1:19,"X"),pruning.mode = "coarse"))
length(temp_sox9<-keepSeqlevels(ovary_ctvdmrt1_sox9_top[!ovary_ctvdmrt1_sox9_top %over% el_mm10],value=c(1:19,"X"),pruning.mode = "coarse"))



#Focus on differentially ATAC-able
length(temp_dmrt1<-temp_dmrt1[temp_dmrt1 %over% vivo_atac_subset[vivo_atac_subset$class=="Sertoli"]])
length(temp_sox9<-temp_sox9[temp_sox9 %over% vivo_atac_subset[vivo_atac_subset$class=="Sertoli"]])


```

## Heatmap
```{r heatmap_atac_chip_motifs,eval=T}
# Remove regions that have mega-copy Dmrt1 sites
length(temp_dmrt1<-temp_dmrt1[countOverlaps(peak_center(temp_dmrt1)+100,sertoli_dmrt1.sites) < 5 & countOverlaps(peak_center(temp_dmrt1)+100,sertoli_sox9.sites) < 10])
length(temp_sox9<-temp_sox9[countOverlaps(peak_center(temp_sox9)+100,sertoli_dmrt1.sites) < 5 & countOverlaps(peak_center(temp_sox9)+100,sertoli_sox9.sites) < 10])

#Venn diagram for top peaks

length(a<-temp_dmrt1[!temp_dmrt1 %over% ovary_ctvdmrt1_sox9])
length(b<-temp_dmrt1[temp_dmrt1 %over% temp_sox9])
length(c<-temp_sox9[!temp_sox9 %over% ovary8w_ctvdmrt1_dmrt1 ])
#grid.newpage()
#vennplot <- draw.pairwise.venn(length(a),length(c),length(b), c("ovary8w_ctvdmrt1_dmrt1", "ovary_ctvdmrt1_sox9"))
#cross section area too large 

(sample_size<-min(c(length(a),length(b),length(c))))
length(vivo_atac_subset[vivo_atac_subset$class=="Constitutive" & vivo_atac_subset$Female > 3.3 & vivo_atac_subset$Male > 3.3])
length(d<-sample(vivo_atac_subset[vivo_atac_subset$class=="Constitutive" & vivo_atac_subset$Female > 3.3 & vivo_atac_subset$Male > 3.3],sample_size))

a<-sample(a,sample_size)
b<-sample(b,sample_size)

#length(temp<-c(sample(a,sample_size),
#               sample(b,sample_size),
#               sample(c,sample_size)))

temp<-c(a,b,c,d)



length(temp<-peak_center(temp) )
table(temp$level<-factor(c(rep("Dmrt1 Alone",length(a)),rep("Overlap",length(b)),rep("Sox9 Alone",length(c)),rep("Constitutive",sample_size)),levels=c("Dmrt1 Alone","Overlap","Sox9 Alone","Constitutive")))
seqlevelsStyle(temp)
```

## Meme on Each group
```{r export_fasta,eval=F}
#export for meme
group_a.dna<-getSeq(mm10,a+75)
names(group_a.dna)<-paste0("group_a_",1:length(group_a.dna))
export(group_a.dna,"group_a_151bp.fasta")

group_b.dna<-getSeq(mm10,b+75)
names(group_b.dna)<-paste0("group_b_",1:length(group_b.dna))
export(group_b.dna,"group_b_151bp.fasta")

group_c.dna<-getSeq(mm10,c+75)
names(group_c.dna)<-paste0("group_c_",1:length(group_c.dna))
export(group_c.dna,"group_c_151bp.fasta")

#group2.dna<-getSeq(mm10,temp[temp$level=="Overlap"]+100)
#names(group2.dna)<-paste0("group2_",1:length(group2.dna))    
#export(group2.dna,"b_open_201bp.fasta")

#group3.dna<-getSeq(mm10,temp[temp$level=="Sox9 Alone"]+100)
#names(group3.dna)<-paste0("group3_",1:length(group3.dna))


#x<-"group_c_201bp.txt" #doesn't work
#  x<-"group1_201bp.txt" #works
#filename.fasta<-paste0(tools::file_path_sans_ext(x),".fasta")
#filename.meme<-TFBSTools:::parseMEMEOutput412(x)
#filename.meme

```


```{r import_meme_groups,eval=F}

group_a.meme<-TFBSTools:::parseMEMEOutput412("group_a_sampled_151bp.txt")
x<-lengths(group_a.meme_dna<-relist(getSeq(readDNAStringSet("group_a_sampled_151bp.fasta"),unlist(group_a.meme$motifList)),group_a.meme$motifList))
names(group_a.meme_dna)<-paste0("Motif ",1:length(group_a.meme$motifList),", p-value: ",group_a.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(group_a.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Dmrt1 Alone")

group_b.meme<-TFBSTools:::parseMEMEOutput412("group_b_151bp.txt")
x<-lengths(group_b.meme_dna<-relist(getSeq(readDNAStringSet("group_b_151bp.fasta"),unlist(group_b.meme$motifList)),group_b.meme$motifList))
names(group_b.meme_dna)<-paste0("Motif ",1:length(group_b.meme$motifList),", p-value: ",group_b.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(group_b.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Dmrt1+Sox9")

group_c.meme<-TFBSTools:::parseMEMEOutput412("group_c_151bp.txt")
  x<-lengths(group_c.meme_dna<-relist(getSeq(readDNAStringSet("group_c_151bp.fasta"),unlist(group_c.meme$motifList)),group_c.meme$motifList))
names(group_c.meme_dna)<-paste0("Motif ",1:length(group_c.meme$motifList),", p-value: ",group_c.meme$motifEvalues," ( ",x," sites )")
ggplot() + geom_logo(lapply(group_c.meme_dna,as.character),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 Alone")

ggplot() + geom_logo(list(as.character(reverseComplement(group_c.meme_dna[[1]]))),seq_type='dna') + theme_logo() +
  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 Alone Site 1 revcomp")

#group2.meme<-TFBSTools:::parseMEMEOutput412("b_open_201bp.txt")
#x<-lengths(group2.meme_dna<-relist(getSeq(readDNAStringSet("b_open_201bp.fasta"),unlist(group2.meme$motifList)),group2.meme$motifList))
#names(group2.meme_dna)<-paste0("Motif ",1:length(group2.meme$motifList),", p-value: ",group2.meme$motifEvalues," ( ",x," sites )")
#ggplot() + geom_logo(lapply(group2.meme_dna,as.character),seq_type='dna') + theme_logo() +
#  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Overlap")
#
#ggplot() + geom_logo(as.character(reverseComplement(group2.meme_dna[[1]])),seq_type='dna') + theme_logo() +
#  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("revcomp")
#
#  group3.meme<-TFBSTools:::parseMEMEOutput412("group3_201bp.txt")
#x<-lengths(group3.meme_dna<-relist(getSeq(readDNAStringSet("group3_201bp.fasta"),unlist(group3.meme$motifList)),group3.meme$motifList))
#names(group3.meme_dna)<-paste0("Motif ",1:length(group3.meme$motifList),", p-value: ",group3.meme$motifEvalues," ( ",x," sites )")
#ggplot() + geom_logo(lapply(group3.meme_dna,as.character),seq_type='dna') + theme_logo() +
#  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("Sox9 Alone")

#ggplot() + geom_logo(list(as.character(group3.meme_dna[[1]]),as.character(reverseComplement(group3.meme_dna[[1]]))),seq_type='dna') + theme_logo() +
#  facet_wrap(~seq_group, ncol=1, scales='free_x') + ggtitle("revcomp")


#scan genome for "open" sox9 motif
#align with canonical



#GroupA motif
group_a_pfm<-consensusMatrix(reverseComplement(group_a.meme_dna[[1]]))
group_a_pfm<-group_a_pfm[1:4,]
group_a_motif<-new("pfm",mat=t(t(group_a_pfm[1:4,])*1/colSums(group_a_pfm[1:4,])), name="Group_A_Motif")

#GroupB motif
group_b_pfm<-consensusMatrix(group_b.meme_dna[[3]])
group_b_pfm<-group_b_pfm[1:4,]
group_b_motif<-new("pfm",mat=t(t(group_b_pfm[1:4,])*1/colSums(group_b_pfm[1:4,])), name="Group_B_Motif")

#Scan for GroupC motif
group_c_pfm<-consensusMatrix(reverseComplement(group_c.meme_dna[[1]]))
group_c_pfm<-group_c_pfm[1:4,]
group_c_motif<-new("pfm",mat=t(t(group_c_pfm[1:4,])*1/colSums(group_c_pfm[1:4,])), name="Group_C_Motif")


plotMotifLogoStack(DNAmotifAlignment(list(group_c_motif,group_b_motif,group_a_motif),revcomp=c(F,F,F)))



plotMotifLogoStack(DNAmotifAlignment(list(sox9_jaspar_motif,group_c_motif,group_b_motif,group_a_motif,dmrt1_invitro_motif),revcomp=c(F,F,F,F,F)))

```

## Scan for Motifs
```{r scan_for motifs,eval=F}
system.time(group_a_sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=group_a_pfm,score="92%")))
sapply(group_a_sites,length)
group_a_sites<-sort(unlist(GRangesList(group_a_sites)))

system.time(group_b_sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=group_b_pfm,score="90%")))
sapply(group_b_sites,length)
group_b_sites<-sort(unlist(GRangesList(group_b_sites)))

system.time(group_c_sites<-mclapply(c(1:19,"X","Y"), function(x) find_sites(x,pfm=group_c_pfm,score="85%")))
sapply(group_c_sites,length)
group_c_sites<-sort(unlist(GRangesList(group_c_sites)))

save(group_a_sites,group_b_sites,group_c_sites,file="group_abc_sites.rdata")
```

## Heatmap
```{r figure_4_heatmap,eval=T}
#load("group_abc_sites.rdata")
#g_p1<-rtracklayer::import("../data/dmrt1_chip_from_dmrt1_expressing_ovary_idx4_Project033_041317.fastq.dedup.unique.ucsc.bigWig",
#                       selection=temp+2000)
width<-2000
#Ovary  


seqlevelsStyle(temp)<-"UCSC"
g_8w<-rtracklayer::import("../hic_data/ovary82_ctvdmrt1_Dmrt1_Project032_122716_Sat_Feb_20_2021_1135.bigWig",
                       selection=temp+width)
g_sox9_ovary<-rtracklayer::import("../hic_data/ovary8w_ctvdmrt1_Sox9_Project032_122716_Sat_Feb_20_2021_1135.bigWig",
                       selection=temp+width)

#Testis
g_dmrt1_sertoli<-rtracklayer::import("../hic_data/DMRT1_Sertoli_p7_ChIP_Tue_Jan_05_2021_1150.bigWig",
                       selection=temp+width)
#g_sox9_sertoli<-rtracklayer::import("../data/sertoli_sox9_ChIP_index1_S1.dedup_pe_q55.bigWig",
#                       selection=temp+width)
#g_dmrt1_testis<-rtracklayer::import("../data/m8w_chip6_111223.dedup.unique.bigWig",
#                       selection=temp+width)
g_sox9_sertoli<-rtracklayer::import("../hic_data/SOX9_Sertoli_p7_ChIP_Tue_Jan_05_2021_1150.bigWig",
                       selection=temp+width)
#Fresh ATAC
#g_atac_sertoli<-rtracklayer::import("../data/N702.R1_trimmed.fastq.bigWig",selection=temp+with) #sertoli
#g_atac_granulosa<-rtracklayer::import("../data/N704.R1_trimmed.fastq.bigWig",selection=temp+width) #granulosa
#g_atac_sertoli<-rtracklayer::import("../data/P7_Sertoli_Rep1.bigWig",selection=temp+width) #sertoli
#g_atac_granulosa<-rtracklayer::import("../data/Adult_Granulosa_Rep1.bigWig",selection=temp+width) #granulosa
g_atac_sertoli<-rtracklayer::import("../HiC/sert_two_Mon_Jan_04_2021_1758.bigWig",selection=temp+width) #sertoli
g_atac_granulosa<-rtracklayer::import("../HiC/gran_two_Mon_Jan_04_2021_1758.bigWig",selection=temp+width) #granulosa

#seqlevelsStyle(temp)<-"NCBI"
#seqlevelsStyle(g_dmrt1_sertoli)<-"NCBI"
#seqlevelsStyle(g_sox9_sertoli)<-"NCBI"
#seqlevelsStyle(g_atac_sertoli)<-"NCBI"
#seqlevelsStyle(g_atac_granulosa)<-"NCBI"

#length(dmrt1_pwm<-rtracklayer::import("../data/dmrt1_invitro_pwm_score85.bed"))
#seqlevelsStyle(dmrt1_pwm)<-"NCBI"
#length(sox9_pwm<-rtracklayer::import("../data/sox9_pwm_score85.bed"))
#seqlevelsStyle(sox9_pwm)<-"NCBI"
#length(sox9_open_pwm<-rtracklayer::import("sox9_open_pwm_score85.bed"))
#seqlevelsStyle(sox9_open_pwm)<-"NCBI"

mat_8w = normalizeToMatrix(g_8w, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
mat_sox9_ovary = normalizeToMatrix(g_sox9_ovary, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
      
#mat_dmrt1_testis = normalizeToMatrix(g_dmrt1_testis, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
mat_dmrt1_sertoli = normalizeToMatrix(g_dmrt1_sertoli, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
mat_sox9_sertoli = normalizeToMatrix(g_sox9_sertoli, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
#mat_sox9_sertoli = normalizeToMatrix(g_sox9_sertoli, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
      
#atac
mat_atac_granulosa = normalizeToMatrix(g_atac_granulosa, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
mat_atac_sertoli = normalizeToMatrix(g_atac_sertoli, temp, value_column = "score",  extend = width, mean_mode = "w0", w = 50)
      
seqlevelsStyle(sertoli_dmrt1.sites)<-"UCSC"
seqlevelsStyle(sertoli_sox9.sites)<-"UCSC"
mat_a_pwm=normalizeToMatrix(sertoli_dmrt1.sites, temp, value_column = "score", mean_mode = "absolute",
         extend = width, w = 50, smooth = FALSE)
#mat_klf4_pwm=normalizeToMatrix(klf4_pwm, temp, value_column = "score", mean_mode = "absolute",
#         extend = width, w = 50, smooth = FALSE)
mat_b_pwm=normalizeToMatrix(sertoli_sox9.sites, temp, value_column = "score", mean_mode = "absolute",
         extend = width, w = 50, smooth = FALSE)
#mat_c_pwm=normalizeToMatrix(group_c_sites, temp, value_column = "score", mean_mode = "absolute",
#         extend = width, w = 50, smooth = FALSE)


col_fun_8w= colorRamp2(quantile(mat_8w, c(0, 0.99)), c("white", "blue"))
col_fun_sox9_ovary= colorRamp2(quantile(mat_8w, c(0, 0.99)), c("white", "green"))

#col_fun_dmrt1_testis= colorRamp2(quantile(mat_dmrt1_testis, c(0, 0.99)), c("white", "blue"))
col_fun_dmrt1_sertoli= colorRamp2(quantile(mat_dmrt1_sertoli, c(0, 0.99)), c("white", "blue"))
col_fun_sox9_sertoli= colorRamp2(quantile(mat_sox9_sertoli, c(0, 0.99)), c("white", "green"))
#col_fun_sox9_sertoli= colorRamp2(quantile(mat_sox9_sertoli, c(0, 0.99)), c("white", "green"))

col_fun_atac_granulosa= colorRamp2(quantile(mat_atac_granulosa, c(0, 0.99)), c("white", "red"))
col_fun_atac_sertoli= colorRamp2(quantile(mat_atac_sertoli, c(0, 0.99)), c("white", "blue"))

#col=c("red","green","blue","white")
col=cbPalette[c(6,7,4,1)]
#svglite::svglite(paste0("Figure4E_invivo_chip_heatmap_",ts,".svg"),width=8,height=7.5)
pdf(paste0("Figure4E_invivo_chip_heatmap_",ts,".pdf"),width=8,height=7.5)
EnrichedHeatmap(mat_atac_granulosa, col = col_fun_atac_granulosa, name = "atac granulosa",column_title = "atac granulsa",column_title_rot=90, axis_name_rot = 90,split=temp$level,pos_line=F,show_heatmap_legend = TRUE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,2.2),yaxis = F)) ,use_raster=TRUE) +
EnrichedHeatmap(mat_atac_sertoli, col = col_fun_atac_sertoli, name = "Atac sertoli",column_title = "atac sertoli",column_title_rot=90, axis_name_rot = 90,split=temp$level,pos_line=F,show_heatmap_legend = TRUE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0,2.2),yaxis = F)) ,use_raster=TRUE) +
EnrichedHeatmap(mat_8w, col = col_fun_8w, name = "8w OvaryDmrt1",column_title = "8w Ovary Dmrt1",column_title_rot=90, axis_name_rot = 90,split=temp$level,pos_line=F,show_heatmap_legend = TRUE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ,use_raster=TRUE) +
EnrichedHeatmap(mat_sox9_ovary, col = col_fun_sox9_ovary, name = "8w OvarySox9",column_title = "8w Ovary Sox9",column_title_rot=90, axis_name_rot = 90,split=temp$level,pos_line=F,show_heatmap_legend = TRUE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ,use_raster=TRUE) +
EnrichedHeatmap(mat_dmrt1_sertoli, col = col_fun_dmrt1_sertoli, name = "p7 Sertoli Dmrt1",column_title = "p7 Sertoli Dmrt1",column_title_rot=90, axis_name_rot = 90,split=temp$level,pos_line=F,show_heatmap_legend = TRUE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0.05,0.75),yaxis = F)),use_raster=TRUE ) +

EnrichedHeatmap(mat_sox9_sertoli, col = col_fun_sox9_sertoli, name = "p7 Sertoli Sox9",column_title = "p7 Sertoli Sox9",column_title_rot=90, axis_name_rot = 90,split=temp$level,pos_line=F,show_heatmap_legend = TRUE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),ylim=c(0.05,0.75),yaxis = F)),use_raster=TRUE ) +

EnrichedHeatmap(mat_a_pwm, col = c("white", "darkblue"),pos_line=F, name = "Dmrt1 Sites", axis_name_rot = 90,column_title = "Dmrt1 Motifs",column_title_rot=90,split=temp$level,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)),use_raster=TRUE ) +
#EnrichedHeatmap(mat_b_pwm, col = c("white", "black"),pos_line=F, name = "Klf4 PWM", axis_name_rot = 90,column_title = "KLF4 Motifs",column_title_rot=90,split=temp$level,show_heatmap_legend = FALSE,
#                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) +
EnrichedHeatmap(mat_b_pwm, col = c("white", "darkgreen"),pos_line=F, name = "Sox9 Sites", axis_name_rot = 90,column_title = "Sox9 Motifs",column_title_rot=90,split=temp$level,show_heatmap_legend = FALSE,
                top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)),use_raster=TRUE ) 
#EnrichedHeatmap(mat_c_pwm, col = c("white", "black"),pos_line=F, name = "Group C PWM", axis_name_rot = 90,column_title = "Group C Motifs",column_title_rot=90,split=temp$level,show_heatmap_legend = FALSE,
#               top_annotation = HeatmapAnnotation(lines = anno_enriched(gp = gpar(col = col),yaxis = F)) ) 
dev.off()



```
